// @refinedev/mui/dist/index.mjs
"use client"
var __defProp = Object.defineProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });

// src/index.tsx
import {
  SnackbarProvider as SnackbarProvider2,
  withSnackbar,
  SnackbarContent,
  useSnackbar as useSnackbar2
} from "notistack";

// src/components/themedLayout/index.tsx
import React5 from "react";
import Box2 from "@mui/material/Box";

// src/contexts/themedLayoutContext/index.tsx
import React, { useState } from "react";
var ThemedLayoutContext = React.createContext({
  siderCollapsed: false,
  mobileSiderOpen: false,
  setSiderCollapsed: () => void 0,
  setMobileSiderOpen: () => void 0
});
var ThemedLayoutContextProvider = /* @__PURE__ */ __name(({ children, initialSiderCollapsed, onSiderCollapsed }) => {
  const [siderCollapsed, setSiderCollapsedState] = useState(
    initialSiderCollapsed ?? false
  );
  const [mobileSiderOpen, setMobileSiderOpen] = useState(false);
  const setSiderCollapsed = /* @__PURE__ */ __name((collapsed) => {
    setSiderCollapsedState(collapsed);
    if (onSiderCollapsed) {
      onSiderCollapsed(collapsed);
    }
  }, "setSiderCollapsed");
  return /* @__PURE__ */ React.createElement(
    ThemedLayoutContext.Provider,
    {
      value: {
        siderCollapsed,
        mobileSiderOpen,
        setSiderCollapsed,
        setMobileSiderOpen,
        onSiderCollapsed
      }
    },
    children
  );
}, "ThemedLayoutContextProvider");

// src/components/themedLayout/sider/index.tsx
import React2, { useState as useState3 } from "react";
import Box from "@mui/material/Box";
import Drawer from "@mui/material/Drawer";
import List from "@mui/material/List";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";
import Collapse from "@mui/material/Collapse";
import Tooltip from "@mui/material/Tooltip";
import IconButton from "@mui/material/IconButton";
import Paper from "@mui/material/Paper";
import ListOutlined from "@mui/icons-material/esm/ListOutlined";
import Logout from "@mui/icons-material/esm/Logout";
import ExpandLess from "@mui/icons-material/esm/ExpandLess";
import ExpandMore from "@mui/icons-material/esm/ExpandMore";
import ChevronLeft from "@mui/icons-material/esm/ChevronLeft";
import Dashboard from "@mui/icons-material/esm/Dashboard";
import {
  CanAccess,
  useIsExistAuthentication,
  useLogout,
  useTranslate,
  useLink,
  useMenu,
  useActiveAuthProvider,
  useWarnAboutChange
} from "@refinedev/core";

// src/hooks/useAutocomplete/index.ts
import {
  useSelect as useSelectCore
} from "@refinedev/core";
import isEqual from "lodash-es/isEqual.js";
import unionWith from "lodash-es/unionWith.js";
var useAutocomplete = /* @__PURE__ */ __name((props) => {
  var _a, _b, _c, _d;
  const { query, defaultValueQuery, onSearch, overtime } = useSelectCore(props);
  return {
    autocompleteProps: {
      options: props.selectedOptionsOrder === "selected-first" ? unionWith(
        ((_a = defaultValueQuery.result) == null ? void 0 : _a.data) || [],
        ((_b = query.data) == null ? void 0 : _b.data) || [],
        isEqual
      ) : unionWith(
        ((_c = query.data) == null ? void 0 : _c.data) || [],
        ((_d = defaultValueQuery.result) == null ? void 0 : _d.data) || [],
        isEqual
      ),
      loading: query.isFetching || defaultValueQuery.query.isFetching,
      onInputChange: (event, value) => {
        if ((event == null ? void 0 : event.type) === "change") {
          onSearch(value);
        } else if ((event == null ? void 0 : event.type) === "click") {
          onSearch("");
        }
      },
      filterOptions: (x) => x
    },
    onSearch,
    query,
    defaultValueQuery,
    overtime
  };
}, "useAutocomplete");

// src/hooks/useDataGrid/index.ts
import {
  useUpdate,
  useLiveMode,
  useTable as useTableCore,
  useResourceParams
} from "@refinedev/core";
import { useEffect, useMemo, useRef, useState as useState2 } from "react";
import differenceWith from "lodash-es/differenceWith.js";
import isEqual2 from "lodash-es/isEqual.js";

// src/definitions/dataGrid/index.ts
import { GridLogicOperator } from "@mui/x-data-grid";
var transformSortModelToCrudSorting = /* @__PURE__ */ __name((sortModel) => {
  const sorter = sortModel.map(({ field, sort }) => ({
    field,
    order: sort || "asc"
  }));
  return sorter;
}, "transformSortModelToCrudSorting");
var transformCrudSortingToSortModel = /* @__PURE__ */ __name((crudSorting) => {
  const sortModel = crudSorting.map(({ field, order }) => ({
    field,
    sort: order
  }));
  return sortModel;
}, "transformCrudSortingToSortModel");
var transformMuiOperatorToCrudOperator = /* @__PURE__ */ __name((operatorValue) => {
  if (!operatorValue) {
    return "eq";
  }
  switch (operatorValue) {
    case "equals":
    case "is":
    case "=":
      return "eq";
    case "!=":
    case "not":
    case "doesNotEqual":
      return "ne";
    case "contains":
      return "contains";
    case "doesNotContain":
      return "ncontains";
    case "isAnyOf":
      return "in";
    case ">":
    case "after":
      return "gt";
    case ">=":
    case "onOrAfter":
      return "gte";
    case "<":
    case "before":
      return "lt";
    case "<=":
    case "onOrBefore":
      return "lte";
    case "startsWith":
      return "startswith";
    case "endsWith":
      return "endswith";
    case "isEmpty":
      return "null";
    case "isNotEmpty":
      return "nnull";
    default:
      return operatorValue;
  }
}, "transformMuiOperatorToCrudOperator");
var transformFilterModelToCrudFilters = /* @__PURE__ */ __name(({
  items,
  logicOperator
}) => {
  const filters = items.map(({ field, value, operator }) => {
    const filter = {
      field,
      value: ["isEmpty", "isNotEmpty"].includes(operator) ? true : value ?? "",
      operator: transformMuiOperatorToCrudOperator(operator)
    };
    return filter;
  });
  if (logicOperator === GridLogicOperator.Or) {
    return [{ operator: "or", value: filters }];
  }
  return filters;
}, "transformFilterModelToCrudFilters");
var transformCrudOperatorToMuiOperator = /* @__PURE__ */ __name((operator, columnType) => {
  switch (columnType) {
    case "number":
      switch (operator) {
        case "eq":
          return "=";
        case "ne":
          return "!=";
        case "gt":
          return ">";
        case "gte":
          return ">=";
        case "lt":
          return "<";
        case "lte":
          return "<=";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        case "in":
          return "isAnyOf";
        default:
          return operator;
      }
    case "singleSelect":
    case "boolean":
      switch (operator) {
        case "eq":
          return "is";
        default:
          return operator;
      }
    case void 0:
    case "string":
      switch (operator) {
        case "eq":
          return "equals";
        case "ne":
          return "doesNotEqual";
        case "contains":
          return "contains";
        case "ncontains":
          return "doesNotContain";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        case "startswith":
          return "startsWith";
        case "endswith":
          return "endsWith";
        case "in":
          return "isAnyOf";
        default:
          return operator;
      }
    case "date":
    case "dateTime":
      switch (operator) {
        case "eq":
          return "is";
        case "ne":
          return "not";
        case "gt":
          return "after";
        case "gte":
          return "onOrAfter";
        case "lt":
          return "before";
        case "lte":
          return "onOrBefore";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        default:
          return operator;
      }
    default:
      return operator;
  }
}, "transformCrudOperatorToMuiOperator");
var transformCrudFiltersToFilterModel = /* @__PURE__ */ __name((crudFilters, columnsType) => {
  var _a;
  const gridFilterItems = [];
  const fieldOperatorCount = {};
  const isExistOrFilter = crudFilters.some(
    (filter) => filter.operator === "or"
  );
  if (columnsType) {
    if (isExistOrFilter) {
      const orLogicalFilters = (_a = crudFilters.find(
        (filter) => filter.operator === "or"
      )) == null ? void 0 : _a.value;
      orLogicalFilters.map(({ field, value, operator }) => {
        const columnType = columnsType[field];
        const id = field + operator;
        fieldOperatorCount[id] = (fieldOperatorCount[id] || 0) + 1;
        const uniqueId = id + String(fieldOperatorCount[id]);
        gridFilterItems.push({
          field,
          operator: transformCrudOperatorToMuiOperator(operator, columnType),
          value: value === "" ? void 0 : value,
          id: fieldOperatorCount[id] > 1 ? uniqueId : id
        });
      });
    } else {
      crudFilters.map(({ field, value, operator }) => {
        const columnType = columnsType[field];
        const id = field + operator;
        fieldOperatorCount[id] = (fieldOperatorCount[id] || 0) + 1;
        const uniqueId = id + String(fieldOperatorCount[id]);
        gridFilterItems.push({
          field,
          operator: transformCrudOperatorToMuiOperator(operator, columnType),
          value: value === "" ? void 0 : value,
          id: fieldOperatorCount[id] > 1 ? uniqueId : id
        });
      });
    }
  }
  return {
    items: gridFilterItems,
    // If there is "or" filter, default link operator is "or"
    logicOperator: isExistOrFilter ? GridLogicOperator.Or : GridLogicOperator.And
  };
}, "transformCrudFiltersToFilterModel");

// src/hooks/useDataGrid/index.ts
var defaultPermanentFilter = [];
var defaultPermanentSort = [];
var DEFAULT_FILTER_DEBOUNCE_MS = 300;
function useDataGrid({
  onSearch: onSearchProp,
  pagination = { pageSize: 25 },
  filters: filtersFromProp,
  sorters: sortersFromProp,
  syncWithLocation: syncWithLocationProp,
  resource: resourceFromProp,
  successNotification,
  errorNotification,
  queryOptions,
  liveMode: liveModeFromProp,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName,
  overtimeOptions,
  editable = false,
  updateMutationOptions
} = {}) {
  const liveMode = useLiveMode(liveModeFromProp);
  const columnsTypes = useRef({});
  const filterDebounceRef = useRef(null);
  const { identifier } = useResourceParams({ resource: resourceFromProp });
  const {
    tableQuery,
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    filters,
    setFilters,
    sorters,
    setSorters,
    pageCount,
    createLinkForSyncWithLocation,
    overtime,
    result
  } = useTableCore({
    pagination: {
      ...pagination,
      pageSize: (pagination == null ? void 0 : pagination.pageSize) ?? 25
    },
    filters: { ...filtersFromProp, defaultBehavior: "replace" },
    sorters: sortersFromProp,
    syncWithLocation: syncWithLocationProp,
    resource: resourceFromProp,
    successNotification,
    errorNotification,
    queryOptions,
    liveMode: liveModeFromProp,
    onLiveEvent,
    liveParams,
    meta,
    dataProviderName,
    overtimeOptions
  });
  const [muiCrudFilters, setMuiCrudFilters] = useState2(filters);
  const { data, isFetched, isLoading } = tableQuery;
  const rowCountRef = useRef((data == null ? void 0 : data.total) || 0);
  const rowCount = useMemo(() => {
    if (data == null ? void 0 : data.total) {
      rowCountRef.current = data.total;
    }
    return rowCountRef.current;
  }, [data]);
  const isServerSideFilteringEnabled = ((filtersFromProp == null ? void 0 : filtersFromProp.mode) || "server") === "server";
  const isServerSideSortingEnabled = ((sortersFromProp == null ? void 0 : sortersFromProp.mode) || "server") === "server";
  const isPaginationEnabled = ((pagination == null ? void 0 : pagination.mode) ?? "server") !== "off";
  const preferredPermanentSorters = (sortersFromProp == null ? void 0 : sortersFromProp.permanent) ?? defaultPermanentSort;
  const preferredPermanentFilters = (filtersFromProp == null ? void 0 : filtersFromProp.permanent) ?? defaultPermanentFilter;
  const handlePageChange = /* @__PURE__ */ __name((page) => {
    if (isPaginationEnabled) {
      setCurrentPage(page + 1);
    }
  }, "handlePageChange");
  const handlePageSizeChange = /* @__PURE__ */ __name((pageSize2) => {
    if (isPaginationEnabled) {
      setPageSize(pageSize2);
    }
  }, "handlePageSizeChange");
  const clearFilterDebounce = /* @__PURE__ */ __name(() => {
    if (filterDebounceRef.current) {
      clearTimeout(filterDebounceRef.current);
      filterDebounceRef.current = null;
    }
  }, "clearFilterDebounce");
  useEffect(() => {
    return () => {
      clearFilterDebounce();
    };
  }, []);
  const applyFilters = /* @__PURE__ */ __name((crudFilters) => {
    setFilters(crudFilters.filter((f) => f.value !== ""));
    if (isPaginationEnabled) {
      setCurrentPage(1);
    }
  }, "applyFilters");
  const handleSortModelChange = /* @__PURE__ */ __name((sortModel) => {
    const crudSorting = transformSortModelToCrudSorting(sortModel);
    setSorters(crudSorting);
  }, "handleSortModelChange");
  const handleFilterModelChange = /* @__PURE__ */ __name((filterModel) => {
    const crudFilters = transformFilterModelToCrudFilters(filterModel);
    setMuiCrudFilters(crudFilters);
    if (isServerSideFilteringEnabled) {
      clearFilterDebounce();
      filterDebounceRef.current = setTimeout(() => {
        applyFilters(crudFilters);
      }, DEFAULT_FILTER_DEBOUNCE_MS);
      return;
    }
    applyFilters(crudFilters);
  }, "handleFilterModelChange");
  const search = /* @__PURE__ */ __name(async (value) => {
    if (onSearchProp) {
      const searchFilters = await onSearchProp(value);
      clearFilterDebounce();
      setMuiCrudFilters(searchFilters);
      applyFilters(searchFilters);
    }
  }, "search");
  const dataGridPaginationValues = /* @__PURE__ */ __name(() => {
    if (isPaginationEnabled) {
      return {
        paginationMode: "server",
        paginationModel: {
          page: currentPage - 1,
          pageSize
        },
        onPaginationModelChange: (model) => {
          handlePageChange(model.page);
          handlePageSizeChange(model.pageSize);
        }
      };
    }
    return {
      paginationMode: "client"
    };
  }, "dataGridPaginationValues");
  const { mutate } = useUpdate({
    mutationOptions: updateMutationOptions
  });
  const processRowUpdate = /* @__PURE__ */ __name(async (newRow, oldRow) => {
    if (!editable) {
      return Promise.resolve(oldRow);
    }
    if (!identifier) {
      return Promise.reject(new Error("Resource is not defined"));
    }
    return new Promise((resolve, reject) => {
      mutate(
        {
          resource: identifier,
          id: newRow.id,
          values: newRow,
          meta: updateMutationOptions == null ? void 0 : updateMutationOptions.meta
        },
        {
          onError: (error) => {
            reject(error);
          },
          onSuccess: () => {
            resolve(newRow);
          }
        }
      );
    });
  }, "processRowUpdate");
  const transformedSortModel = useMemo(
    () => transformCrudSortingToSortModel(
      differenceWith(sorters, preferredPermanentSorters, isEqual2)
    ),
    [sorters, preferredPermanentSorters]
  );
  return {
    tableQuery,
    dataGridProps: {
      disableRowSelectionOnClick: true,
      rows: (data == null ? void 0 : data.data) || [],
      loading: liveMode === "auto" ? isLoading : !isFetched,
      rowCount,
      ...dataGridPaginationValues(),
      sortingMode: isServerSideSortingEnabled ? "server" : "client",
      sortModel: transformedSortModel,
      onSortModelChange: handleSortModelChange,
      filterMode: isServerSideFilteringEnabled ? "server" : "client",
      // Disable DataGrid's debounce for server filtering to prevent input resets.
      filterDebounceMs: isServerSideFilteringEnabled ? 0 : void 0,
      filterModel: transformCrudFiltersToFilterModel(
        differenceWith(muiCrudFilters, preferredPermanentFilters, isEqual2),
        columnsTypes.current
      ),
      onFilterModelChange: handleFilterModelChange,
      onStateChange: (state) => {
        const newColumnsTypes = Object.fromEntries(
          Object.entries(state.columns.lookup).map(([key, value]) => {
            return [key, value.type];
          })
        );
        const isStateChanged = !isEqual2(newColumnsTypes, columnsTypes.current);
        if (isStateChanged) {
          columnsTypes.current = newColumnsTypes;
        }
      },
      processRowUpdate: editable ? processRowUpdate : void 0
    },
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    pageCount,
    sorters,
    setSorters,
    filters,
    setFilters,
    search,
    createLinkForSyncWithLocation,
    overtime,
    result
  };
}
__name(useDataGrid, "useDataGrid");

// src/hooks/useThemedLayoutContext/index.ts
import { useContext } from "react";
var useThemedLayoutContext = /* @__PURE__ */ __name(() => {
  const {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  } = useContext(ThemedLayoutContext);
  return {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  };
}, "useThemedLayoutContext");

// src/components/themedLayout/sider/index.tsx
var ThemedSider = /* @__PURE__ */ __name(({
  Title: TitleFromProps,
  render,
  meta,
  activeItemDisabled = false,
  siderItemsAreCollapsed = true
}) => {
  const {
    siderCollapsed,
    setSiderCollapsed,
    mobileSiderOpen,
    setMobileSiderOpen
  } = useThemedLayoutContext();
  const drawerWidth = /* @__PURE__ */ __name(() => {
    if (siderCollapsed)
      return 56;
    return 240;
  }, "drawerWidth");
  const t = useTranslate();
  const Link3 = useLink();
  const translate = useTranslate();
  const { menuItems, selectedKey, defaultOpenKeys } = useMenu({ meta });
  const isExistAuthentication = useIsExistAuthentication();
  const authProvider = useActiveAuthProvider();
  const { warnWhen, setWarnWhen } = useWarnAboutChange();
  const { mutate: mutateLogout } = useLogout();
  const defaultExpandMenuItems = (() => {
    const defaultOpenKeys2 = {};
    if (siderItemsAreCollapsed)
      return defaultOpenKeys2;
    return menuItems.reduce((prev, curr) => {
      const { key } = curr;
      return {
        ...prev,
        [key]: true
      };
    }, {});
  })();
  const [open, setOpen] = useState3(
    defaultExpandMenuItems
  );
  React2.useEffect(() => {
    setOpen((previous) => {
      const previousKeys = Object.keys(previous);
      const previousOpenKeys = previousKeys.filter((key) => previous[key]);
      const uniqueKeys = /* @__PURE__ */ new Set([...previousOpenKeys, ...defaultOpenKeys]);
      const uniqueKeysRecord = Object.fromEntries(
        Array.from(uniqueKeys.values()).map((key) => [key, true])
      );
      return uniqueKeysRecord;
    });
  }, [defaultOpenKeys]);
  const RenderToTitle = TitleFromProps ?? ThemedTitle;
  const handleClick = /* @__PURE__ */ __name((key) => {
    setOpen({ ...open, [key]: !open[key] });
  }, "handleClick");
  const renderTreeView = /* @__PURE__ */ __name((tree, selectedKey2) => {
    return tree.map((item) => {
      const { icon, label, route, name, children, meta: meta2 } = item;
      const isOpen = open[item.key || ""] || false;
      const isSelected = item.key === selectedKey2;
      const isNested = !((meta2 == null ? void 0 : meta2.parent) === void 0);
      if (children.length > 0) {
        return /* @__PURE__ */ React2.createElement(
          CanAccess,
          {
            key: item.key,
            resource: name,
            action: "list",
            params: {
              resource: item
            }
          },
          /* @__PURE__ */ React2.createElement("div", { key: item.key }, /* @__PURE__ */ React2.createElement(
            Tooltip,
            {
              title: label ?? name,
              placement: "right",
              disableHoverListener: !siderCollapsed,
              arrow: true
            },
            /* @__PURE__ */ React2.createElement(
              ListItemButton,
              {
                onClick: () => {
                  if (siderCollapsed) {
                    setSiderCollapsed(false);
                    if (!isOpen) {
                      handleClick(item.key || "");
                    }
                  } else {
                    handleClick(item.key || "");
                  }
                },
                sx: {
                  pl: isNested ? 4 : 2,
                  justifyContent: "center"
                }
              },
              /* @__PURE__ */ React2.createElement(
                ListItemIcon,
                {
                  sx: {
                    justifyContent: "center",
                    minWidth: "24px",
                    transition: "margin-right 0.3s",
                    marginRight: siderCollapsed ? "0px" : "12px",
                    color: "currentColor"
                  }
                },
                icon ?? /* @__PURE__ */ React2.createElement(ListOutlined, null)
              ),
              /* @__PURE__ */ React2.createElement(
                ListItemText,
                {
                  primary: label,
                  primaryTypographyProps: {
                    noWrap: true,
                    fontSize: "14px"
                  }
                }
              ),
              isOpen ? /* @__PURE__ */ React2.createElement(
                ExpandLess,
                {
                  sx: {
                    color: "text.icon"
                  }
                }
              ) : /* @__PURE__ */ React2.createElement(
                ExpandMore,
                {
                  sx: {
                    color: "text.icon"
                  }
                }
              )
            )
          ), !siderCollapsed && /* @__PURE__ */ React2.createElement(
            Collapse,
            {
              in: open[item.key || ""],
              timeout: "auto",
              unmountOnExit: true
            },
            /* @__PURE__ */ React2.createElement(List, { component: "div", disablePadding: true }, renderTreeView(children, selectedKey2))
          ))
        );
      }
      const linkStyle = activeItemDisabled && isSelected ? { pointerEvents: "none" } : {};
      return /* @__PURE__ */ React2.createElement(
        CanAccess,
        {
          key: item.key,
          resource: name,
          action: "list",
          params: { resource: item }
        },
        /* @__PURE__ */ React2.createElement(
          Tooltip,
          {
            title: label ?? name,
            placement: "right",
            disableHoverListener: !siderCollapsed,
            arrow: true
          },
          /* @__PURE__ */ React2.createElement(
            ListItemButton,
            {
              component: Link3,
              to: route,
              selected: isSelected,
              style: linkStyle,
              onClick: () => {
                setMobileSiderOpen(false);
              },
              sx: {
                pl: isNested ? 4 : 2,
                py: isNested ? 1.25 : 1,
                justifyContent: "center",
                color: isSelected ? "primary.main" : "text.primary"
              }
            },
            /* @__PURE__ */ React2.createElement(
              ListItemIcon,
              {
                sx: {
                  justifyContent: "center",
                  transition: "margin-right 0.3s",
                  marginRight: siderCollapsed ? "0px" : "12px",
                  minWidth: "24px",
                  color: "currentColor"
                }
              },
              icon ?? /* @__PURE__ */ React2.createElement(ListOutlined, null)
            ),
            /* @__PURE__ */ React2.createElement(
              ListItemText,
              {
                primary: label,
                primaryTypographyProps: {
                  noWrap: true,
                  fontSize: "14px"
                }
              }
            )
          )
        )
      );
    });
  }, "renderTreeView");
  const dashboard = /* @__PURE__ */ React2.createElement(CanAccess, { resource: "dashboard", action: "list" }, /* @__PURE__ */ React2.createElement(
    Tooltip,
    {
      title: translate("dashboard.title", "Dashboard"),
      placement: "right",
      disableHoverListener: !siderCollapsed,
      arrow: true
    },
    /* @__PURE__ */ React2.createElement(
      Link3,
      {
        to: "/",
        style: { textDecoration: "none" },
        onClick: () => {
          setMobileSiderOpen(false);
        }
      },
      /* @__PURE__ */ React2.createElement(
        ListItemButton,
        {
          selected: selectedKey === "/",
          sx: {
            pl: 2,
            py: 1,
            justifyContent: "center",
            color: selectedKey === "/" ? "primary.main" : "text.primary"
          }
        },
        /* @__PURE__ */ React2.createElement(
          ListItemIcon,
          {
            sx: {
              justifyContent: "center",
              minWidth: "24px",
              transition: "margin-right 0.3s",
              marginRight: siderCollapsed ? "0px" : "12px",
              color: "currentColor",
              fontSize: "14px"
            }
          },
          /* @__PURE__ */ React2.createElement(Dashboard, null)
        ),
        /* @__PURE__ */ React2.createElement(
          ListItemText,
          {
            primary: translate("dashboard.title", "Dashboard"),
            primaryTypographyProps: {
              noWrap: true,
              fontSize: "14px"
            }
          }
        )
      )
    )
  ));
  const handleLogout = /* @__PURE__ */ __name(() => {
    if (warnWhen) {
      const confirm = window.confirm(
        t(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (confirm) {
        setWarnWhen(false);
        mutateLogout();
      }
    } else {
      mutateLogout();
    }
  }, "handleLogout");
  const logout = isExistAuthentication && /* @__PURE__ */ React2.createElement(
    Tooltip,
    {
      title: t("buttons.logout", "Logout"),
      placement: "right",
      disableHoverListener: !siderCollapsed,
      arrow: true
    },
    /* @__PURE__ */ React2.createElement(
      ListItemButton,
      {
        key: "logout",
        onClick: () => handleLogout(),
        sx: {
          justifyContent: "center"
        }
      },
      /* @__PURE__ */ React2.createElement(
        ListItemIcon,
        {
          sx: {
            justifyContent: "center",
            minWidth: "24px",
            transition: "margin-right 0.3s",
            marginRight: siderCollapsed ? "0px" : "12px",
            color: "currentColor"
          }
        },
        /* @__PURE__ */ React2.createElement(Logout, null)
      ),
      /* @__PURE__ */ React2.createElement(
        ListItemText,
        {
          primary: t("buttons.logout", "Logout"),
          primaryTypographyProps: {
            noWrap: true,
            fontSize: "14px"
          }
        }
      )
    )
  );
  const items = renderTreeView(menuItems, selectedKey);
  const renderSider = /* @__PURE__ */ __name(() => {
    if (render) {
      return render({
        logout,
        items,
        collapsed: siderCollapsed
      });
    }
    return /* @__PURE__ */ React2.createElement(React2.Fragment, null, dashboard, items, logout);
  }, "renderSider");
  const drawer = /* @__PURE__ */ React2.createElement(
    List,
    {
      disablePadding: true,
      sx: {
        flexGrow: 1,
        paddingTop: "16px"
      }
    },
    renderSider()
  );
  return /* @__PURE__ */ React2.createElement(React2.Fragment, null, /* @__PURE__ */ React2.createElement(
    Box,
    {
      sx: {
        width: { xs: drawerWidth() },
        display: {
          xs: "none",
          md: "block"
        },
        transition: "width 0.3s ease"
      }
    }
  ), /* @__PURE__ */ React2.createElement(
    Box,
    {
      component: "nav",
      sx: {
        position: "fixed",
        zIndex: 1101,
        width: { sm: drawerWidth() },
        display: "flex"
      }
    },
    /* @__PURE__ */ React2.createElement(
      Drawer,
      {
        variant: "temporary",
        elevation: 2,
        open: mobileSiderOpen,
        onClose: () => setMobileSiderOpen(false),
        ModalProps: {
          keepMounted: true
          // Better open performance on mobile.
        },
        sx: {
          display: {
            sm: "block",
            md: "none"
          }
        }
      },
      /* @__PURE__ */ React2.createElement(
        Box,
        {
          sx: {
            width: drawerWidth()
          }
        },
        /* @__PURE__ */ React2.createElement(
          Box,
          {
            sx: {
              height: 64,
              display: "flex",
              alignItems: "center",
              paddingLeft: "16px",
              fontSize: "14px"
            }
          },
          /* @__PURE__ */ React2.createElement(RenderToTitle, { collapsed: false })
        ),
        drawer
      )
    ),
    /* @__PURE__ */ React2.createElement(
      Drawer,
      {
        variant: "permanent",
        sx: {
          display: { xs: "none", md: "block" },
          "& .MuiDrawer-paper": {
            width: drawerWidth(),
            overflow: "hidden",
            transition: "width 200ms cubic-bezier(0.4, 0, 0.6, 1) 0ms"
          }
        },
        open: true
      },
      /* @__PURE__ */ React2.createElement(
        Paper,
        {
          elevation: 0,
          sx: {
            fontSize: "14px",
            width: "100%",
            height: 64,
            display: "flex",
            flexShrink: 0,
            alignItems: "center",
            justifyContent: siderCollapsed ? "center" : "space-between",
            paddingLeft: siderCollapsed ? 0 : "16px",
            paddingRight: siderCollapsed ? 0 : "8px",
            variant: "outlined",
            borderRadius: 0,
            borderBottom: (theme) => `1px solid ${theme.palette.action.focus}`
          }
        },
        /* @__PURE__ */ React2.createElement(RenderToTitle, { collapsed: siderCollapsed }),
        !siderCollapsed && /* @__PURE__ */ React2.createElement(IconButton, { size: "small", onClick: () => setSiderCollapsed(true) }, /* @__PURE__ */ React2.createElement(ChevronLeft, null))
      ),
      /* @__PURE__ */ React2.createElement(
        Box,
        {
          sx: {
            flexGrow: 1,
            overflowX: "hidden",
            overflowY: "auto"
          }
        },
        drawer
      )
    )
  ));
}, "ThemedSider");

// src/components/themedLayout/header/index.tsx
import React4 from "react";
import { useGetIdentity } from "@refinedev/core";
import AppBar from "@mui/material/AppBar";
import Stack from "@mui/material/Stack";
import Toolbar from "@mui/material/Toolbar";
import Typography from "@mui/material/Typography";
import Avatar from "@mui/material/Avatar";

// src/components/themedLayout/hamburgerMenu/index.tsx
import React3 from "react";
import IconButton2 from "@mui/material/IconButton";
import Menu from "@mui/icons-material/esm/Menu";
var HamburgerIcon = /* @__PURE__ */ __name((props) => /* @__PURE__ */ React3.createElement(IconButton2, { color: "inherit", "aria-label": "open drawer", edge: "start", ...props }, /* @__PURE__ */ React3.createElement(Menu, null)), "HamburgerIcon");
var HamburgerMenu = /* @__PURE__ */ __name(() => {
  const {
    siderCollapsed,
    setSiderCollapsed,
    mobileSiderOpen,
    setMobileSiderOpen
  } = useThemedLayoutContext();
  return /* @__PURE__ */ React3.createElement(React3.Fragment, null, /* @__PURE__ */ React3.createElement(
    HamburgerIcon,
    {
      onClick: () => setSiderCollapsed(!siderCollapsed),
      sx: {
        mr: 2,
        display: { xs: "none", md: "flex" },
        ...!siderCollapsed && { display: "none" }
      }
    }
  ), /* @__PURE__ */ React3.createElement(
    HamburgerIcon,
    {
      onClick: () => setMobileSiderOpen(!mobileSiderOpen),
      sx: {
        mr: 2,
        display: { xs: "flex", md: "none" },
        ...mobileSiderOpen && { display: "none" }
      }
    }
  ));
}, "HamburgerMenu");

// src/components/themedLayout/header/index.tsx
var ThemedHeader = /* @__PURE__ */ __name(({
  sticky
}) => {
  const { data: user } = useGetIdentity();
  const prefferedSticky = sticky ?? true;
  return /* @__PURE__ */ React4.createElement(AppBar, { position: prefferedSticky ? "sticky" : "relative" }, /* @__PURE__ */ React4.createElement(Toolbar, null, /* @__PURE__ */ React4.createElement(HamburgerMenu, null), /* @__PURE__ */ React4.createElement(
    Stack,
    {
      direction: "row",
      width: "100%",
      justifyContent: "flex-end",
      alignItems: "center"
    },
    /* @__PURE__ */ React4.createElement(
      Stack,
      {
        direction: "row",
        gap: "16px",
        alignItems: "center",
        justifyContent: "center"
      },
      (user == null ? void 0 : user.name) && /* @__PURE__ */ React4.createElement(Typography, { variant: "subtitle2", }, user == null ? void 0 : user.name),
      (user == null ? void 0 : user.avatar) && /* @__PURE__ */ React4.createElement(Avatar, { src: user == null ? void 0 : user.avatar, alt: user == null ? void 0 : user.name })
    )
  )));
}, "ThemedHeader");

// src/components/themedLayout/index.tsx
var ThemedLayout = /* @__PURE__ */ __name(({
  Sider,
  Header,
  Title,
  Footer,
  OffLayoutArea,
  children,
  initialSiderCollapsed,
  onSiderCollapsed,
  childrenBoxProps = {},
  containerBoxProps = {}
}) => {
  const SiderToRender = Sider ?? ThemedSider;
  const HeaderToRender = Header ?? ThemedHeader;
  const { sx: childrenSx, ...restChildrenProps } = childrenBoxProps;
  const { sx: containerSx, ...restContainerProps } = containerBoxProps;
  return /* @__PURE__ */ React5.createElement(
    ThemedLayoutContextProvider,
    {
      initialSiderCollapsed,
      onSiderCollapsed
    },
    /* @__PURE__ */ React5.createElement(
      Box2,
      {
        sx: { display: "flex", flexDirection: "row", ...containerSx },
        ...restContainerProps
      },
      /* @__PURE__ */ React5.createElement(SiderToRender, { Title }),
      /* @__PURE__ */ React5.createElement(
        Box2,
        {
          sx: [
            {
              display: "flex",
              flexDirection: "column",
              flex: 1,
              minWidth: "1px",
              minHeight: "1px"
            }
          ]
        },
        /* @__PURE__ */ React5.createElement(HeaderToRender, null),
        /* @__PURE__ */ React5.createElement(
          Box2,
          {
            component: "main",
            sx: {
              p: { xs: 1, md: 2, lg: 3 },
              flexGrow: 1,
              bgcolor: (theme) => theme.palette.background.default,
              ...childrenSx
            },
            ...restChildrenProps
          },
          children
        ),
        Footer && /* @__PURE__ */ React5.createElement(Footer, null)
      ),
      OffLayoutArea && /* @__PURE__ */ React5.createElement(OffLayoutArea, null)
    )
  );
}, "ThemedLayout");

// src/components/themedLayout/title/index.tsx
import React6 from "react";
import { useLink as useLink2, useRefineOptions } from "@refinedev/core";
import MuiLink from "@mui/material/Link";
import SvgIcon from "@mui/material/SvgIcon";
import Typography2 from "@mui/material/Typography";
var ThemedTitle = /* @__PURE__ */ __name(({
  collapsed,
  wrapperStyles,
  icon: iconFromProps,
  text: textFromProps
}) => {
  const {
    title: { icon: defaultIcon, text: defaultText } = {}
  } = useRefineOptions();
  const icon = typeof iconFromProps === "undefined" ? defaultIcon : iconFromProps;
  const text = typeof textFromProps === "undefined" ? defaultText : textFromProps;
  const Link3 = useLink2();
  return /* @__PURE__ */ React6.createElement(Link3, { to: "/", style: { textDecoration: "none" } }, /* @__PURE__ */ React6.createElement(
    MuiLink,
    {
      underline: "none",
      sx: {
        display: "flex",
        alignItems: "center",
        gap: "12px",
        ...wrapperStyles
      }
    },
    /* @__PURE__ */ React6.createElement(SvgIcon, { height: "24px", width: "24px", color: "primary" }, icon),
    !collapsed && /* @__PURE__ */ React6.createElement(
      Typography2,
      {
        variant: "h6",
        fontWeight: 700,
        color: "text.primary",
        fontSize: "inherit",
        textOverflow: "ellipsis",
        overflow: "hidden"
      },
      text
    )
  ));
}, "ThemedTitle");

// src/components/pages/error/index.tsx
import React7, { useEffect as useEffect2, useState as useState4 } from "react";
import { useGo, useResourceParams as useResourceParams2, useTranslate as useTranslate2 } from "@refinedev/core";
import Stack2 from "@mui/material/Stack";
import Button from "@mui/material/Button";
import Tooltip2 from "@mui/material/Tooltip";
import Typography3 from "@mui/material/Typography";
import Grid from "@mui/material/Grid2";
import Info from "@mui/icons-material/esm/Info";
var ErrorComponent = /* @__PURE__ */ __name(() => {
  const [errorMessage, setErrorMessage] = useState4();
  const go = useGo();
  const { resource, action } = useResourceParams2();
  const translate = useTranslate2();
  useEffect2(() => {
    if (resource && action) {
      setErrorMessage(
        translate(
          "pages.error.info",
          {
            action,
            resource: resource == null ? void 0 : resource.name
          },
          `You may have forgotten to add the "${action}" component to "${resource == null ? void 0 : resource.name}" resource.`
        )
      );
    }
  }, [action, resource]);
  return /* @__PURE__ */ React7.createElement(Grid, { display: "flex", justifyContent: "center", alignItems: "center", mt: 20 }, /* @__PURE__ */ React7.createElement(Grid, { container: true, direction: "column", display: "flex", alignItems: "center" }, /* @__PURE__ */ React7.createElement(Typography3, { variant: "h1" }, "404"), /* @__PURE__ */ React7.createElement(Stack2, { direction: "row", spacing: "2" }, /* @__PURE__ */ React7.createElement(Typography3, null, translate(
    "pages.error.404",
    "Sorry, the page you visited does not exist."
  )), errorMessage && /* @__PURE__ */ React7.createElement(Tooltip2, { title: errorMessage }, /* @__PURE__ */ React7.createElement(Info, { }))), /* @__PURE__ */ React7.createElement(
    Button,
    {
      onClick: () => {
        go({ to: "/" });
      }
    },
    translate("pages.error.backHome", "Back Home")
  )));
}, "ErrorComponent");

// src/components/pages/auth/index.tsx
import React12 from "react";

// src/components/pages/auth/components/login/index.tsx
import * as React8 from "react";
import { useForm } from "@refinedev/react-hook-form";
import { FormProvider } from "react-hook-form";
import Button2 from "@mui/material/Button";
import TextField from "@mui/material/TextField";
import Box3 from "@mui/material/Box";
import Typography4 from "@mui/material/Typography";
import Container from "@mui/material/Container";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import Checkbox from "@mui/material/Checkbox";
import FormControlLabel from "@mui/material/FormControlLabel";
import Divider from "@mui/material/Divider";
import MuiLink2 from "@mui/material/Link";
import Stack3 from "@mui/material/Stack";
import {
  useLogin,
  useTranslate as useTranslate3,
  useLink as useLink3
} from "@refinedev/core";

// src/components/pages/auth/components/styles.ts
var layoutStyles = {};
var titleStyles = {
  textAlign: "center",
  fontSize: "24px",
  marginBottom: "24px",
  overflowWrap: "break-word",
  hyphens: "manual",
  textOverflow: "unset",
  whiteSpace: "pre-wrap"
};

// src/components/pages/auth/components/login/index.tsx
var LoginPage = /* @__PURE__ */ __name(({
  providers,
  registerLink,
  forgotPasswordLink,
  rememberMe,
  contentProps,
  wrapperProps,
  renderContent,
  formProps,
  title,
  hideForm,
  mutationVariables
}) => {
  var _a;
  const { onSubmit, ...useFormProps } = formProps || {};
  const methods = useForm({
    ...useFormProps
  });
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = methods;
  const { mutate: login, isPending } = useLogin();
  const translate = useTranslate3();
  const Link3 = useLink3();
  const PageTitle = title === false ? null : /* @__PURE__ */ React8.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React8.createElement(
      ThemedTitle,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ React8.createElement(React8.Fragment, null, /* @__PURE__ */ React8.createElement(Stack3, { spacing: 1 }, providers.map((provider) => {
        return /* @__PURE__ */ React8.createElement(
          Button2,
          {
            key: provider.name,
            variant: "outlined",
            fullWidth: true,
            sx: {
              color: "primary.light",
              borderColor: "primary.light",
              textTransform: "none"
            },
            onClick: () => login({ ...mutationVariables, providerName: provider.name }),
            startIcon: provider.icon
          },
          provider.label
        );
      })), !hideForm && /* @__PURE__ */ React8.createElement(
        Divider,
        {
          sx: {
            fontSize: "12px",
            marginY: "16px"
          }
        },
        translate("pages.login.divider", "or")
      ));
    }
    return null;
  }, "renderProviders");
  const Content = /* @__PURE__ */ React8.createElement(Card, { ...contentProps ?? {} }, /* @__PURE__ */ React8.createElement(CardContent, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React8.createElement(
    Typography4,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.login.title", "Sign in to your account")
  ), renderProviders(), !hideForm && /* @__PURE__ */ React8.createElement(
    Box3,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return login({ ...mutationVariables, ...data });
      })
    },
    /* @__PURE__ */ React8.createElement(
      TextField,
      {
        ...register("email", {
          required: translate(
            "pages.login.errors.requiredEmail",
            "Email is required"
          )
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.login.fields.email", "Email"),
        error: !!errors.email,
        name: "email",
        type: "email",
        autoComplete: "email",
        sx: {
          mt: 0
        }
      }
    ),
    /* @__PURE__ */ React8.createElement(
      TextField,
      {
        ...register("password", {
          required: translate(
            "pages.login.errors.requiredPassword",
            "Password is required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate("pages.login.fields.password", "Password"),
        helperText: (_a = errors == null ? void 0 : errors.password) == null ? void 0 : _a.message,
        error: !!errors.password,
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React8.createElement(
      Box3,
      {
        component: "div",
        sx: {
          mt: "24px",
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center"
        }
      },
      rememberMe ?? /* @__PURE__ */ React8.createElement(
        FormControlLabel,
        {
          sx: {
            span: {
              fontSize: "14px",
              color: "text.secondary"
            }
          },
          color: "secondary",
          control: /* @__PURE__ */ React8.createElement(
            Checkbox,
            {
              size: "small",
              id: "remember",
              ...register("remember")
            }
          ),
          label: translate(
            "pages.login.buttons.rememberMe",
            "Remember me"
          )
        }
      ),
      forgotPasswordLink ?? /* @__PURE__ */ React8.createElement(
        MuiLink2,
        {
          variant: "body2",
          color: "primary",
          fontSize: "12px",
          component: Link3,
          underline: "none",
          to: "/forgot-password"
        },
        translate(
          "pages.login.buttons.forgotPassword",
          "Forgot password?"
        )
      )
    ),
    /* @__PURE__ */ React8.createElement(
      Button2,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        disabled: isPending,
        sx: { mt: "24px" }
      },
      translate("pages.login.signin", "Sign in")
    )
  ), registerLink ?? /* @__PURE__ */ React8.createElement(
    Box3,
    {
      sx: {
        mt: "24px",
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
      }
    },
    /* @__PURE__ */ React8.createElement(
      Typography4,
      {
        textAlign: "center",
        variant: "body2",
        component: "span",
        fontSize: "12px"
      },
      translate(
        "pages.login.buttons.noAccount",
        "Don\u2019t have an account?"
      )
    ),
    /* @__PURE__ */ React8.createElement(
      MuiLink2,
      {
        ml: "4px",
        fontSize: "12px",
        variant: "body2",
        color: "primary",
        component: Link3,
        underline: "none",
        to: "/register",
        fontWeight: "bold"
      },
      translate("pages.login.signup", "Sign up")
    )
  )));
  return /* @__PURE__ */ React8.createElement(FormProvider, { ...methods }, /* @__PURE__ */ React8.createElement(Box3, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React8.createElement(
    Container,
    {
      component: "main",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: hideForm ? "flex-start" : "center",
        alignItems: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    /* @__PURE__ */ React8.createElement(
      Box3,
      {
        sx: {
          width: "100%",
          maxWidth: "400px",
          display: "flex",
          flexDirection: "column",
          paddingTop: hideForm ? "15dvh" : 0
        }
      },
      renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React8.createElement(React8.Fragment, null, PageTitle, Content)
    )
  )));
}, "LoginPage");

// src/components/pages/auth/components/register/index.tsx
import * as React9 from "react";
import { useForm as useForm2 } from "@refinedev/react-hook-form";
import Button3 from "@mui/material/Button";
import TextField2 from "@mui/material/TextField";
import Box4 from "@mui/material/Box";
import Typography5 from "@mui/material/Typography";
import Container2 from "@mui/material/Container";
import Card2 from "@mui/material/Card";
import CardContent2 from "@mui/material/CardContent";
import Divider2 from "@mui/material/Divider";
import Stack4 from "@mui/material/Stack";
import MuiLink3 from "@mui/material/Link";
import {
  useTranslate as useTranslate4,
  useLink as useLink4,
  useRegister
} from "@refinedev/core";
var RegisterPage = /* @__PURE__ */ __name(({
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  providers,
  formProps,
  title,
  hideForm,
  mutationVariables
}) => {
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm2({
    ...useFormProps
  });
  const { mutate: registerMutate, isPending } = useRegister();
  const translate = useTranslate4();
  const Link3 = useLink4();
  const PageTitle = title === false ? null : /* @__PURE__ */ React9.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React9.createElement(
      ThemedTitle,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ React9.createElement(React9.Fragment, null, /* @__PURE__ */ React9.createElement(Stack4, { spacing: 1 }, providers.map((provider) => {
        return /* @__PURE__ */ React9.createElement(
          Button3,
          {
            key: provider.name,
            color: "secondary",
            fullWidth: true,
            variant: "outlined",
            sx: {
              color: "primary.light",
              borderColor: "primary.light",
              textTransform: "none"
            },
            onClick: () => registerMutate({
              ...mutationVariables,
              providerName: provider.name
            }),
            startIcon: provider.icon
          },
          provider.label
        );
      })), !hideForm && /* @__PURE__ */ React9.createElement(
        Divider2,
        {
          sx: {
            fontSize: "12px",
            marginY: "16px"
          }
        },
        translate(
          "pages.register.divider",
          translate("pages.login.divider", "or")
        )
      ));
    }
    return null;
  }, "renderProviders");
  const Content = /* @__PURE__ */ React9.createElement(Card2, { ...contentProps ?? {} }, /* @__PURE__ */ React9.createElement(CardContent2, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React9.createElement(
    Typography5,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.register.title", "Sign up for your account")
  ), renderProviders(), !hideForm && /* @__PURE__ */ React9.createElement(
    Box4,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return registerMutate({ ...mutationVariables, ...data });
      })
    },
    /* @__PURE__ */ React9.createElement(
      TextField2,
      {
        ...register("email", {
          required: translate(
            "pages.register.errors.requiredEmail",
            "Email is required"
          ),
          pattern: {
            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
            message: translate(
              "pages.register.errors.validEmail",
              "Invalid email address"
            )
          }
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.register.email", "Email"),
        error: !!errors.email,
        helperText: errors["email"] ? errors["email"].message : "",
        name: "email",
        autoComplete: "email",
        sx: {
          mt: 0
        }
      }
    ),
    /* @__PURE__ */ React9.createElement(
      TextField2,
      {
        ...register("password", {
          required: translate(
            "pages.register.errors.requiredPassword",
            "Password is required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate("pages.register.fields.password", "Password"),
        helperText: errors["password"] ? errors["password"].message : "",
        error: !!errors.password,
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React9.createElement(
      Button3,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        disabled: isPending,
        sx: {
          mt: "24px"
        }
      },
      translate("pages.register.signup", "Sign up")
    )
  ), loginLink ?? /* @__PURE__ */ React9.createElement(
    Box4,
    {
      display: "flex",
      justifyContent: "flex-end",
      alignItems: "center",
      sx: {
        mt: "24px",
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
      }
    },
    /* @__PURE__ */ React9.createElement(Typography5, { variant: "body2", component: "span", fontSize: "12px" }, translate(
      "pages.register.buttons.haveAccount",
      translate(
        "pages.login.buttons.haveAccount",
        "Have an account?"
      )
    )),
    /* @__PURE__ */ React9.createElement(
      MuiLink3,
      {
        ml: "4px",
        variant: "body2",
        color: "primary",
        component: Link3,
        underline: "none",
        to: "/login",
        fontSize: "12px",
        fontWeight: "bold"
      },
      translate(
        "pages.register.signin",
        translate("pages.login.signin", "Sign in")
      )
    )
  )));
  return /* @__PURE__ */ React9.createElement(Box4, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React9.createElement(
    Container2,
    {
      component: "main",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: hideForm ? "flex-start" : "center",
        alignItems: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    /* @__PURE__ */ React9.createElement(
      Box4,
      {
        sx: {
          width: "100%",
          maxWidth: "400px",
          display: "flex",
          flexDirection: "column",
          paddingTop: hideForm ? "15dvh" : 0
        }
      },
      renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React9.createElement(React9.Fragment, null, PageTitle, Content)
    )
  ));
}, "RegisterPage");

// src/components/pages/auth/components/forgotPassword/index.tsx
import Button4 from "@mui/material/Button";
import TextField3 from "@mui/material/TextField";
import Box5 from "@mui/material/Box";
import Typography6 from "@mui/material/Typography";
import Container3 from "@mui/material/Container";
import Card3 from "@mui/material/Card";
import CardContent3 from "@mui/material/CardContent";
import MuiLink4 from "@mui/material/Link";
import { useForm as useForm3 } from "@refinedev/react-hook-form";
import * as React10 from "react";
import {
  useForgotPassword,
  useLink as useLink5,
  useTranslate as useTranslate5
} from "@refinedev/core";
var ForgotPasswordPage = /* @__PURE__ */ __name(({
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title,
  mutationVariables
}) => {
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = useForm3({
    ...useFormProps
  });
  const { mutate, isPending } = useForgotPassword();
  const translate = useTranslate5();
  const Link3 = useLink5();
  const PageTitle = title === false ? null : /* @__PURE__ */ React10.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React10.createElement(
      ThemedTitle,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const Content = /* @__PURE__ */ React10.createElement(Card3, { ...contentProps ?? {} }, /* @__PURE__ */ React10.createElement(CardContent3, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React10.createElement(
    Typography6,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.forgotPassword.title", "Forgot your password?")
  ), /* @__PURE__ */ React10.createElement(
    Box5,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return mutate({ ...mutationVariables, ...data });
      })
    },
    /* @__PURE__ */ React10.createElement(
      TextField3,
      {
        ...register("email", {
          required: translate(
            "pages.forgotPassword.errors.requiredEmail",
            "Email is required"
          ),
          pattern: {
            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
            message: translate(
              "pages.register.errors.validEmail",
              "Invalid email address"
            )
          }
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.forgotPassword.fields.email", "Email"),
        name: "email",
        type: "email",
        error: !!errors.email,
        autoComplete: "email",
        sx: {
          m: 0
        }
      }
    ),
    loginLink ?? /* @__PURE__ */ React10.createElement(Box5, { textAlign: "right", sx: { mt: "24px" } }, /* @__PURE__ */ React10.createElement(Typography6, { variant: "body2", component: "span", fontSize: "12px" }, translate(
      "pages.forgotPassword.buttons.haveAccount",
      translate(
        "pages.register.buttons.haveAccount",
        "Have an account? "
      )
    )), " ", /* @__PURE__ */ React10.createElement(
      MuiLink4,
      {
        variant: "body2",
        component: Link3,
        underline: "none",
        to: "/login",
        fontWeight: "bold",
        fontSize: "12px",
        color: "primary.light"
      },
      translate(
        "pages.forgotPassword.signin",
        translate("pages.login.signin", "Sign in")
      )
    )),
    /* @__PURE__ */ React10.createElement(
      Button4,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        sx: { mt: "24px" },
        disabled: isPending
      },
      translate(
        "pages.forgotPassword.buttons.submit",
        "Send reset instructions"
      )
    )
  )));
  return /* @__PURE__ */ React10.createElement(React10.Fragment, null, /* @__PURE__ */ React10.createElement(Box5, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React10.createElement(
    Container3,
    {
      component: "main",
      maxWidth: "xs",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React10.createElement(React10.Fragment, null, PageTitle, Content)
  )));
}, "ForgotPasswordPage");

// src/components/pages/auth/components/updatePassword/index.tsx
import * as React11 from "react";
import { useForm as useForm4 } from "@refinedev/react-hook-form";
import Button5 from "@mui/material/Button";
import TextField4 from "@mui/material/TextField";
import Box6 from "@mui/material/Box";
import Typography7 from "@mui/material/Typography";
import Container4 from "@mui/material/Container";
import Card4 from "@mui/material/Card";
import CardContent4 from "@mui/material/CardContent";
import {
  useTranslate as useTranslate6,
  useUpdatePassword
} from "@refinedev/core";
var UpdatePasswordPage = /* @__PURE__ */ __name(({
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title = void 0,
  mutationVariables
}) => {
  var _a, _b;
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    watch,
    handleSubmit,
    formState: { errors }
  } = useForm4({
    ...useFormProps
  });
  const { mutate: update, isPending } = useUpdatePassword();
  const translate = useTranslate6();
  const PageTitle = title === false ? null : /* @__PURE__ */ React11.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React11.createElement(
      ThemedTitle,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const Content = /* @__PURE__ */ React11.createElement(Card4, { ...contentProps ?? {} }, /* @__PURE__ */ React11.createElement(CardContent4, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React11.createElement(
    Typography7,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.updatePassword.title", "Set New Password")
  ), /* @__PURE__ */ React11.createElement(
    Box6,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return update({ ...mutationVariables, ...data });
      })
    },
    /* @__PURE__ */ React11.createElement(
      TextField4,
      {
        ...register("password", {
          required: translate(
            "pages.updatePassword.errors.requiredPassword",
            "Password required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate(
          "pages.updatePassword.fields.password",
          "New Password"
        ),
        helperText: (_a = errors == null ? void 0 : errors.password) == null ? void 0 : _a.message,
        error: !!(errors == null ? void 0 : errors.password),
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          m: 0
        }
      }
    ),
    /* @__PURE__ */ React11.createElement(
      TextField4,
      {
        ...register("confirmPassword", {
          required: translate(
            "pages.updatePassword.errors.requiredConfirmPassword",
            "Confirm Password is required"
          ),
          validate: (value) => {
            if (watch("password") !== value) {
              return translate(
                "pages.updatePassword.errors.confirmPasswordNotMatch",
                "Passwords do not match"
              );
            }
            return true;
          }
        }),
        id: "confirmPassword",
        margin: "normal",
        fullWidth: true,
        name: "confirmPassword",
        label: translate(
          "pages.updatePassword.fields.confirmPassword",
          "Confirm New Password"
        ),
        helperText: (_b = errors == null ? void 0 : errors.confirmPassword) == null ? void 0 : _b.message,
        error: !!(errors == null ? void 0 : errors.confirmPassword),
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-confirm-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React11.createElement(
      Button5,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        sx: {
          mt: "24px"
        },
        disabled: isPending
      },
      translate("pages.updatePassword.buttons.submit", "Update")
    )
  )));
  return /* @__PURE__ */ React11.createElement(React11.Fragment, null, /* @__PURE__ */ React11.createElement(Box6, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React11.createElement(
    Container4,
    {
      component: "main",
      maxWidth: "xs",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React11.createElement(React11.Fragment, null, PageTitle, Content)
  )));
}, "UpdatePasswordPage");

// src/components/pages/auth/index.tsx
var AuthPage = /* @__PURE__ */ __name((props) => {
  const { type } = props;
  const renderView = /* @__PURE__ */ __name(() => {
    switch (type) {
      case "register":
        return /* @__PURE__ */ React12.createElement(RegisterPage, { ...props });
      case "forgotPassword":
        return /* @__PURE__ */ React12.createElement(ForgotPasswordPage, { ...props });
      case "updatePassword":
        return /* @__PURE__ */ React12.createElement(UpdatePasswordPage, { ...props });
      default:
        return /* @__PURE__ */ React12.createElement(LoginPage, { ...props });
    }
  }, "renderView");
  return /* @__PURE__ */ React12.createElement(React12.Fragment, null, renderView());
}, "AuthPage");

// src/components/crud/create/index.tsx
import React13 from "react";
import {
  useTranslate as useTranslate7,
  useUserFriendlyName,
  useBack,
  useResourceParams as useResourceParams3
} from "@refinedev/core";
import Card5 from "@mui/material/Card";
import CardHeader from "@mui/material/CardHeader";
import CardActions from "@mui/material/CardActions";
import CardContent5 from "@mui/material/CardContent";
import IconButton3 from "@mui/material/IconButton";
import Typography8 from "@mui/material/Typography";
import Box7 from "@mui/material/Box";
import CircularProgress from "@mui/material/CircularProgress";
import { alpha } from "@mui/system";
import ArrowBackIcon from "@mui/icons-material/esm/ArrowBack";
import { RefinePageHeaderClassNames } from "@refinedev/ui-types";
var Create = /* @__PURE__ */ __name(({
  title,
  children,
  saveButtonProps: saveButtonPropsFromProps,
  resource: resourceFromProps,
  isLoading = false,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps
}) => {
  var _a;
  const translate = useTranslate7();
  const back = useBack();
  const getUserFriendlyName = useUserFriendlyName();
  const { resource, action, identifier } = useResourceParams3({
    resource: resourceFromProps
  });
  const breadcrumb = breadcrumbFromProps;
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ React13.createElement(React13.Fragment, null, breadcrumb) : /* @__PURE__ */ React13.createElement(Breadcrumb, null);
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps
  };
  const defaultFooterButtons = /* @__PURE__ */ React13.createElement(SaveButton, { ...saveButtonProps });
  return /* @__PURE__ */ React13.createElement(
    Card5,
    {
      ...wrapperProps ?? {},
      sx: {
        position: "relative",
        ...wrapperProps == null ? void 0 : wrapperProps.sx
      }
    },
    isLoading && /* @__PURE__ */ React13.createElement(
      Box7,
      {
        sx: {
          position: "absolute",
          inset: 0,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          zIndex: (theme) => theme.zIndex.drawer + 1,
          // this is needed to support custom themes, dark mode etc.
          bgcolor: (theme) => alpha(theme.palette.background.paper, 0.4)
        }
      },
      /* @__PURE__ */ React13.createElement(CircularProgress, null)
    ),
    breadcrumbComponent,
    /* @__PURE__ */ React13.createElement(
      CardHeader,
      {
        sx: {
          display: "flex",
          flexWrap: "wrap",
          ".MuiCardHeader-action": {
            margin: 0,
            alignSelf: "center"
          }
        },
        title: title ?? /* @__PURE__ */ React13.createElement(
          Typography8,
          {
            variant: "h5",
            className: RefinePageHeaderClassNames.Title
          },
          translate(
            `${identifier}.titles.create`,
            `Create ${getUserFriendlyName(
              ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier,
              "singular"
            )}`
          )
        ),
        avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ React13.createElement(
          IconButton3,
          {
            onClick: action !== "list" || typeof action !== "undefined" ? back : void 0
          },
          /* @__PURE__ */ React13.createElement(ArrowBackIcon, null)
        ),
        action: headerButtons ? /* @__PURE__ */ React13.createElement(Box7, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
          defaultButtons: null
        }) : headerButtons : null) : void 0,
        ...headerProps ?? {}
      }
    ),
    /* @__PURE__ */ React13.createElement(CardContent5, { ...contentProps ?? {} }, children),
    /* @__PURE__ */ React13.createElement(
      CardActions,
      {
        sx: {
          display: "flex",
          justifyContent: "flex-end",
          gap: "16px",
          padding: "16px"
        },
        ...footerButtonProps ?? {}
      },
      footerButtons ? typeof footerButtons === "function" ? footerButtons({
        defaultButtons: defaultFooterButtons,
        saveButtonProps
      }) : footerButtons : defaultFooterButtons
    )
  );
}, "Create");

// src/components/crud/edit/index.tsx
import React14 from "react";
import {
  useMutationMode,
  useTranslate as useTranslate8,
  useUserFriendlyName as useUserFriendlyName2,
  useToPath,
  useResourceParams as useResourceParams4,
  useBack as useBack2,
  useGo as useGo2
} from "@refinedev/core";
import Card6 from "@mui/material/Card";
import CardHeader2 from "@mui/material/CardHeader";
import IconButton4 from "@mui/material/IconButton";
import CardContent6 from "@mui/material/CardContent";
import CardActions2 from "@mui/material/CardActions";
import Typography9 from "@mui/material/Typography";
import Box8 from "@mui/material/Box";
import CircularProgress2 from "@mui/material/CircularProgress";
import { alpha as alpha2 } from "@mui/system";
import ArrowBackIcon2 from "@mui/icons-material/esm/ArrowBack";
import { RefinePageHeaderClassNames as RefinePageHeaderClassNames2 } from "@refinedev/ui-types";
var Edit = /* @__PURE__ */ __name(({
  title,
  saveButtonProps: saveButtonPropsFromProps,
  mutationMode: mutationModeProp,
  recordItemId,
  children,
  deleteButtonProps: deleteButtonPropsFromProps,
  canDelete,
  resource: resourceFromProps,
  isLoading = false,
  breadcrumb: breadcrumbFromProps,
  dataProviderName,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps,
  autoSaveProps
}) => {
  var _a, _b;
  const translate = useTranslate8();
  const { mutationMode: mutationModeContext } = useMutationMode();
  const mutationMode = mutationModeProp ?? mutationModeContext;
  const back = useBack2();
  const go = useGo2();
  const getUserFriendlyName = useUserFriendlyName2();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = useResourceParams4({ resource: resourceFromProps, id: recordItemId });
  const goListPath = useToPath({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const isDeleteButtonVisible = canDelete ?? (((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) || deleteButtonPropsFromProps);
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ React14.createElement(React14.Fragment, null, breadcrumb) : /* @__PURE__ */ React14.createElement(Breadcrumb, null);
  const listButtonProps = hasList ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    recordItemId: id,
    dataProviderName
  };
  const defaultHeaderButtons = /* @__PURE__ */ React14.createElement(Box8, { display: "flex", flexDirection: "row", alignItems: "center" }, autoSaveProps && /* @__PURE__ */ React14.createElement(AutoSaveIndicator, { ...autoSaveProps }), hasList && /* @__PURE__ */ React14.createElement(ListButton, { ...listButtonProps }), /* @__PURE__ */ React14.createElement(RefreshButton, { ...refreshButtonProps }));
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    mutationMode,
    variant: "outlined",
    onSuccess: () => {
      go({ to: goListPath });
    },
    recordItemId: id,
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps
  };
  const defaultFooterButtons = /* @__PURE__ */ React14.createElement(React14.Fragment, null, isDeleteButtonVisible && /* @__PURE__ */ React14.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ React14.createElement(SaveButton, { ...saveButtonProps }));
  return /* @__PURE__ */ React14.createElement(
    Card6,
    {
      ...wrapperProps ?? {},
      sx: {
        position: "relative",
        ...wrapperProps == null ? void 0 : wrapperProps.sx
      }
    },
    isLoading && /* @__PURE__ */ React14.createElement(
      Box8,
      {
        sx: {
          position: "absolute",
          inset: 0,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          zIndex: (theme) => theme.zIndex.drawer + 1,
          // this is needed to support custom themes, dark mode etc.
          bgcolor: (theme) => alpha2(theme.palette.background.paper, 0.4)
        }
      },
      /* @__PURE__ */ React14.createElement(CircularProgress2, null)
    ),
    breadcrumbComponent,
    /* @__PURE__ */ React14.createElement(
      CardHeader2,
      {
        sx: {
          display: "flex",
          flexWrap: "wrap",
          ".MuiCardHeader-action": {
            margin: 0,
            alignSelf: "center"
          }
        },
        title: title ?? /* @__PURE__ */ React14.createElement(
          Typography9,
          {
            variant: "h5",
            className: RefinePageHeaderClassNames2.Title
          },
          translate(
            `${identifier}.titles.edit`,
            `Edit ${getUserFriendlyName(
              ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.label) ?? identifier,
              "singular"
            )}`
          )
        ),
        avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ React14.createElement(
          IconButton4,
          {
            onClick: action !== "list" && typeof action !== "undefined" ? back : void 0
          },
          /* @__PURE__ */ React14.createElement(ArrowBackIcon2, null)
        ),
        action: /* @__PURE__ */ React14.createElement(Box8, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
          defaultButtons: defaultHeaderButtons,
          listButtonProps,
          refreshButtonProps
        }) : headerButtons : defaultHeaderButtons),
        ...headerProps ?? {}
      }
    ),
    /* @__PURE__ */ React14.createElement(CardContent6, { ...contentProps ?? {} }, children),
    /* @__PURE__ */ React14.createElement(
      CardActions2,
      {
        sx: {
          display: "flex",
          justifyContent: "flex-end",
          gap: "16px",
          padding: "16px"
        },
        ...footerButtonProps ?? {}
      },
      footerButtons ? typeof footerButtons === "function" ? footerButtons({
        defaultButtons: defaultFooterButtons,
        deleteButtonProps,
        saveButtonProps
      }) : footerButtons : defaultFooterButtons
    )
  );
}, "Edit");

// src/components/crud/list/index.tsx
import React15 from "react";
import {
  useTranslate as useTranslate9,
  useUserFriendlyName as useUserFriendlyName3,
  useResourceParams as useResourceParams5
} from "@refinedev/core";
import Card7 from "@mui/material/Card";
import CardHeader3 from "@mui/material/CardHeader";
import CardContent7 from "@mui/material/CardContent";
import Typography10 from "@mui/material/Typography";
import Box9 from "@mui/material/Box";
import { RefinePageHeaderClassNames as RefinePageHeaderClassNames3 } from "@refinedev/ui-types";
var List2 = /* @__PURE__ */ __name(({
  title,
  canCreate,
  children,
  createButtonProps: createButtonPropsFromProps,
  resource: resourceFromProps,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons
}) => {
  var _a;
  const translate = useTranslate9();
  const getUserFriendlyName = useUserFriendlyName3();
  const { resource, identifier } = useResourceParams5({
    resource: resourceFromProps
  });
  const isCreateButtonVisible = canCreate ?? (!!(resource == null ? void 0 : resource.create) || createButtonPropsFromProps);
  const breadcrumb = breadcrumbFromProps;
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ React15.createElement(React15.Fragment, null, breadcrumb) : /* @__PURE__ */ React15.createElement(Breadcrumb, null);
  const createButtonProps = isCreateButtonVisible ? {
    resource: identifier,
    ...createButtonPropsFromProps
  } : void 0;
  const defaultHeaderButtons = isCreateButtonVisible ? /* @__PURE__ */ React15.createElement(CreateButton, { ...createButtonProps }) : null;
  return /* @__PURE__ */ React15.createElement(Card7, { ...wrapperProps ?? {} }, breadcrumbComponent, /* @__PURE__ */ React15.createElement(
    CardHeader3,
    {
      sx: {
        display: "flex",
        flexWrap: "wrap",
        ".MuiCardHeader-action": {
          margin: 0,
          alignSelf: "center"
        }
      },
      title: title ?? /* @__PURE__ */ React15.createElement(
        Typography10,
        {
          variant: "h5",
          className: RefinePageHeaderClassNames3.Title
        },
        translate(
          `${identifier}.titles.list`,
          getUserFriendlyName(
            ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier,
            "plural"
          )
        )
      ),
      action: /* @__PURE__ */ React15.createElement(Box9, { display: "flex", gap: "16px", ...headerButtonProps }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        createButtonProps
      }) : headerButtons : defaultHeaderButtons),
      ...headerProps ?? {}
    }
  ), /* @__PURE__ */ React15.createElement(CardContent7, { ...contentProps ?? {} }, children));
}, "List");

// src/components/crud/show/index.tsx
import React16 from "react";
import {
  useTranslate as useTranslate10,
  useUserFriendlyName as useUserFriendlyName4,
  useBack as useBack3,
  useGo as useGo3,
  useResourceParams as useResourceParams6,
  useToPath as useToPath2
} from "@refinedev/core";
import Card8 from "@mui/material/Card";
import CardActions3 from "@mui/material/CardActions";
import CardContent8 from "@mui/material/CardContent";
import CardHeader4 from "@mui/material/CardHeader";
import IconButton5 from "@mui/material/IconButton";
import Box10 from "@mui/material/Box";
import Typography11 from "@mui/material/Typography";
import CircularProgress3 from "@mui/material/CircularProgress";
import { alpha as alpha3 } from "@mui/system";
import ArrowBackIcon3 from "@mui/icons-material/esm/ArrowBack";
import { RefinePageHeaderClassNames as RefinePageHeaderClassNames4 } from "@refinedev/ui-types";
var Show = /* @__PURE__ */ __name(({
  title,
  canEdit,
  canDelete,
  deleteButtonProps: deleteButtonPropsFromProps,
  isLoading = false,
  children,
  resource: resourceFromProps,
  recordItemId,
  breadcrumb: breadcrumbFromProps,
  dataProviderName,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps
}) => {
  var _a, _b, _c;
  const translate = useTranslate10();
  const back = useBack3();
  const go = useGo3();
  const getUserFriendlyName = useUserFriendlyName4();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = useResourceParams6({ resource: resourceFromProps, id: recordItemId });
  const goListPath = useToPath2({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const hasDelete = canDelete ?? (((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) || deleteButtonPropsFromProps);
  const isDeleteButtonVisible = hasDelete && typeof id !== "undefined";
  const isEditButtonVisible = canEdit ?? ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.canEdit) ?? !!(resource == null ? void 0 : resource.edit);
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ React16.createElement(React16.Fragment, null, breadcrumb) : /* @__PURE__ */ React16.createElement(Breadcrumb, null);
  const listButtonProps = hasList ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier ?? (resource == null ? void 0 : resource.name)
  } : void 0;
  const editButtonProps = isEditButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier ?? (resource == null ? void 0 : resource.name),
    recordItemId: id
  } : void 0;
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier ?? (resource == null ? void 0 : resource.name),
    recordItemId: id,
    onSuccess: () => {
      go({ to: goListPath });
    },
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: identifier ?? (resource == null ? void 0 : resource.name),
    recordItemId: id,
    dataProviderName
  };
  const defaultHeaderButtons = /* @__PURE__ */ React16.createElement(React16.Fragment, null, hasList && /* @__PURE__ */ React16.createElement(ListButton, { ...listButtonProps }), isEditButtonVisible && /* @__PURE__ */ React16.createElement(EditButton, { ...editButtonProps }), isDeleteButtonVisible && /* @__PURE__ */ React16.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ React16.createElement(RefreshButton, { ...refreshButtonProps }));
  return /* @__PURE__ */ React16.createElement(
    Card8,
    {
      ...wrapperProps ?? {},
      sx: {
        position: "relative",
        ...wrapperProps == null ? void 0 : wrapperProps.sx
      }
    },
    isLoading && /* @__PURE__ */ React16.createElement(
      Box10,
      {
        sx: {
          position: "absolute",
          inset: 0,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          zIndex: (theme) => theme.zIndex.drawer + 1,
          // this is needed to support custom themes, dark mode etc.
          bgcolor: (theme) => alpha3(theme.palette.background.paper, 0.4)
        }
      },
      /* @__PURE__ */ React16.createElement(CircularProgress3, null)
    ),
    breadcrumbComponent,
    /* @__PURE__ */ React16.createElement(
      CardHeader4,
      {
        sx: {
          display: "flex",
          flexWrap: "wrap",
          ".MuiCardHeader-action": {
            margin: 0,
            alignSelf: "center"
          }
        },
        title: title ?? /* @__PURE__ */ React16.createElement(
          Typography11,
          {
            variant: "h5",
            className: RefinePageHeaderClassNames4.Title
          },
          translate(
            `${identifier}.titles.show`,
            `Show ${getUserFriendlyName(
              ((_c = resource == null ? void 0 : resource.meta) == null ? void 0 : _c.label) ?? identifier,
              "singular"
            )}`
          )
        ),
        avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ React16.createElement(
          IconButton5,
          {
            onClick: action !== "list" && typeof action !== "undefined" ? back : void 0
          },
          /* @__PURE__ */ React16.createElement(ArrowBackIcon3, null)
        ),
        action: /* @__PURE__ */ React16.createElement(Box10, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
          defaultButtons: defaultHeaderButtons,
          deleteButtonProps,
          editButtonProps,
          listButtonProps,
          refreshButtonProps
        }) : headerButtons : defaultHeaderButtons),
        ...headerProps ?? {}
      }
    ),
    /* @__PURE__ */ React16.createElement(CardContent8, { ...contentProps ?? {} }, children),
    /* @__PURE__ */ React16.createElement(CardActions3, { sx: { padding: "16px" }, ...footerButtonProps ?? {} }, footerButtons ? typeof footerButtons === "function" ? footerButtons({ defaultButtons: null }) : footerButtons : null)
  );
}, "Show");

// src/components/buttons/create/index.tsx
import React17 from "react";
import { useCreateButton } from "@refinedev/core";
import {
  RefineButtonClassNames,
  RefineButtonTestIds
} from "@refinedev/ui-types";
import Button6 from "@mui/material/Button";
import AddBoxOutlined from "@mui/icons-material/esm/AddBoxOutlined";
var CreateButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...props
}) => {
  const { to, label, title, disabled, hidden, LinkComponent } = useCreateButton(
    {
      resource: resourceNameFromProps,
      meta,
      accessControl
    }
  );
  const isDisabled = disabled || props.disabled;
  const isHidden = hidden || props.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = props;
  return /* @__PURE__ */ React17.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ React17.createElement(
      Button6,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ React17.createElement(AddBoxOutlined, { ...svgIconProps }),
        title,
        variant: "contained",
        sx: { minWidth: 0, ...sx },
        
        className: RefineButtonClassNames.CreateButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ React17.createElement(AddBoxOutlined, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "CreateButton");

// src/components/buttons/edit/index.tsx
import React18 from "react";
import { useEditButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames2,
  RefineButtonTestIds as RefineButtonTestIds2
} from "@refinedev/ui-types";
import Button7 from "@mui/material/Button";
import EditOutlined from "@mui/icons-material/esm/EditOutlined";
var EditButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = useEditButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    accessControl,
    meta
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ React18.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ React18.createElement(
      Button7,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ React18.createElement(EditOutlined, { sx: { selfAlign: "center" }, ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: RefineButtonClassNames2.EditButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ React18.createElement(EditOutlined, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "EditButton");

// src/components/buttons/delete/index.tsx
import React19 from "react";
import { useDeleteButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames3,
  RefineButtonTestIds as RefineButtonTestIds3
} from "@refinedev/ui-types";
import Button8 from "@mui/material/Button";
import Dialog from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogTitle from "@mui/material/DialogTitle";
import LoadingButton from "@mui/lab/LoadingButton";
import DeleteOutline from "@mui/icons-material/esm/DeleteOutline";
var DeleteButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  onSuccess,
  mutationMode,
  children,
  successNotification,
  errorNotification,
  hideText = false,
  accessControl,
  meta,
  dataProviderName,
  confirmTitle,
  confirmOkText,
  confirmCancelText,
  svgIconProps,
  invalidates,
  ...rest
}) => {
  const {
    onConfirm,
    title,
    label,
    hidden,
    disabled,
    loading,
    confirmTitle: defaultConfirmTitle,
    confirmOkLabel,
    cancelLabel
  } = useDeleteButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    dataProviderName,
    mutationMode,
    accessControl,
    invalidates,
    onSuccess,
    meta,
    successNotification,
    errorNotification
  });
  const [open, setOpen] = React19.useState(false);
  const { sx, ...restProps } = rest;
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ React19.createElement("div", null, /* @__PURE__ */ React19.createElement(
    LoadingButton,
    {
      color: "error",
      onClick: () => setOpen(true),
      disabled: isDisabled,
      loading,
      startIcon: !hideText && /* @__PURE__ */ React19.createElement(DeleteOutline, { ...svgIconProps }),
      title,
      sx: { minWidth: 0, ...sx },
      loadingPosition: hideText ? "center" : "start",
      
      className: RefineButtonClassNames3.DeleteButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ React19.createElement(DeleteOutline, { fontSize: "small", ...svgIconProps }) : children ?? label
  ), /* @__PURE__ */ React19.createElement(
    Dialog,
    {
      open,
      onClose: () => setOpen(false),
      "aria-labelledby": "alert-dialog-title",
      "aria-describedby": "alert-dialog-description"
    },
    /* @__PURE__ */ React19.createElement(DialogTitle, { id: "alert-dialog-title" }, confirmTitle ?? defaultConfirmTitle),
    /* @__PURE__ */ React19.createElement(DialogActions, { sx: { justifyContent: "center" } }, /* @__PURE__ */ React19.createElement(Button8, { onClick: () => setOpen(false) }, confirmCancelText ?? cancelLabel), /* @__PURE__ */ React19.createElement(
      Button8,
      {
        color: "error",
        onClick: () => {
          onConfirm();
          setOpen(false);
        },
        autoFocus: true
      },
      confirmOkText ?? confirmOkLabel
    ))
  ));
}, "DeleteButton");

// src/components/buttons/refresh/index.tsx
import React20 from "react";
import { useRefreshButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames4,
  RefineButtonTestIds as RefineButtonTestIds4
} from "@refinedev/ui-types";
import LoadingButton2 from "@mui/lab/LoadingButton";
import RefreshOutlined from "@mui/icons-material/esm/RefreshOutlined";
var RefreshButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  dataProviderName,
  svgIconProps,
  children,
  onClick,
  ...rest
}) => {
  const {
    onClick: onRefresh,
    loading,
    label
  } = useRefreshButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    dataProviderName
  });
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ React20.createElement(
    LoadingButton2,
    {
      startIcon: !hideText && /* @__PURE__ */ React20.createElement(RefreshOutlined, { ...svgIconProps }),
      loading,
      loadingPosition: hideText ? "center" : "start",
      onClick: onClick ? onClick : onRefresh,
      sx: { minWidth: 0, ...sx },
      
      className: RefineButtonClassNames4.RefreshButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ React20.createElement(RefreshOutlined, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "RefreshButton");

// src/components/buttons/show/index.tsx
import React21 from "react";
import { useShowButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames5,
  RefineButtonTestIds as RefineButtonTestIds5
} from "@refinedev/ui-types";
import Button9 from "@mui/material/Button";
import VisibilityOutlined from "@mui/icons-material/esm/VisibilityOutlined";
var ShowButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = useShowButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    accessControl,
    meta
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ React21.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ React21.createElement(
      Button9,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ React21.createElement(VisibilityOutlined, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: RefineButtonClassNames5.ShowButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ React21.createElement(VisibilityOutlined, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "ShowButton");

// src/components/buttons/list/index.tsx
import React22 from "react";
import { useListButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames6,
  RefineButtonTestIds as RefineButtonTestIds6
} from "@refinedev/ui-types";
import Button10 from "@mui/material/Button";
import ListOutlined2 from "@mui/icons-material/esm/ListOutlined";
var ListButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = useListButton({
    resource: resourceNameFromProps,
    meta,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ React22.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ React22.createElement(
      Button10,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ React22.createElement(ListOutlined2, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: RefineButtonClassNames6.ListButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ React22.createElement(ListOutlined2, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "ListButton");

// src/components/buttons/export/index.tsx
import React23 from "react";
import { useExportButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames7,
  RefineButtonTestIds as RefineButtonTestIds7
} from "@refinedev/ui-types";
import LoadingButton3 from "@mui/lab/LoadingButton";
import ImportExportOutlined from "@mui/icons-material/esm/ImportExportOutlined";
var ExportButton = /* @__PURE__ */ __name(({
  hideText = false,
  children,
  loading = false,
  svgIconProps,
  ...rest
}) => {
  const { label } = useExportButton();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ React23.createElement(
    LoadingButton3,
    {
      ...rest,
      loading,
      startIcon: !hideText && /* @__PURE__ */ React23.createElement(ImportExportOutlined, { ...svgIconProps }),
      loadingPosition: hideText ? "center" : "start",
      sx: { minWidth: 0, ...sx },
      
      className: RefineButtonClassNames7.ExportButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ React23.createElement(ImportExportOutlined, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "ExportButton");

// src/components/buttons/save/index.tsx
import React24 from "react";
import { useSaveButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames8,
  RefineButtonTestIds as RefineButtonTestIds8
} from "@refinedev/ui-types";
import LoadingButton4 from "@mui/lab/LoadingButton";
import SaveOutlined from "@mui/icons-material/esm/SaveOutlined";
var SaveButton = /* @__PURE__ */ __name(({
  hideText = false,
  svgIconProps,
  children,
  ...rest
}) => {
  const { label } = useSaveButton();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ React24.createElement(
    LoadingButton4,
    {
      startIcon: !hideText && /* @__PURE__ */ React24.createElement(SaveOutlined, { ...svgIconProps }),
      sx: { minWidth: 0, ...sx },
      variant: "contained",
      
      className: RefineButtonClassNames8.SaveButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ React24.createElement(SaveOutlined, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "SaveButton");

// src/components/buttons/clone/index.tsx
import React25 from "react";
import { useCloneButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames9,
  RefineButtonTestIds as RefineButtonTestIds9
} from "@refinedev/ui-types";
import Button11 from "@mui/material/Button";
import AddBoxOutlined2 from "@mui/icons-material/esm/AddBoxOutlined";
var CloneButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = useCloneButton({
    resource: resourceNameFromProps,
    id: recordItemId,
    meta,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ React25.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ React25.createElement(
      Button11,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ React25.createElement(AddBoxOutlined2, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: RefineButtonClassNames9.CloneButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ React25.createElement(AddBoxOutlined2, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "CloneButton");

// src/components/buttons/import/index.tsx
import React26 from "react";
import { useImportButton } from "@refinedev/core";
import {
  RefineButtonClassNames as RefineButtonClassNames10,
  RefineButtonTestIds as RefineButtonTestIds10
} from "@refinedev/ui-types";
import LoadingButton5 from "@mui/lab/LoadingButton";
import ImportExportOutlined2 from "@mui/icons-material/esm/ImportExportOutlined";
var ImportButton = /* @__PURE__ */ __name(({
  inputProps,
  hideText = false,
  loading = false,
  svgIconProps,
  children,
  ...rest
}) => {
  const { label } = useImportButton();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ React26.createElement("label", { htmlFor: "contained-button-file" }, /* @__PURE__ */ React26.createElement("input", { ...inputProps, id: "contained-button-file", multiple: true, hidden: true }), /* @__PURE__ */ React26.createElement(
    LoadingButton5,
    {
      component: "span",
      startIcon: !hideText && /* @__PURE__ */ React26.createElement(ImportExportOutlined2, { ...svgIconProps }),
      loadingPosition: hideText ? "center" : "start",
      loading,
      sx: { minWidth: 0, ...sx },
      
      className: RefineButtonClassNames10.ImportButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ React26.createElement(ImportExportOutlined2, { fontSize: "small", ...svgIconProps }) : children ?? label
  ));
}, "ImportButton");

// src/components/circularDeterminate/index.tsx
import React27, { useState as useState5, useEffect as useEffect3 } from "react";
import Box11 from "@mui/material/Box";
import CircularProgress4 from "@mui/material/CircularProgress";
import Typography12 from "@mui/material/Typography";
var CircularDeterminate = /* @__PURE__ */ __name(({
  undoableTimeout,
  message
}) => {
  const [progress, setProgress] = useState5(100);
  const [timeCount, setTimeCount] = useState5(undoableTimeout);
  useEffect3(() => {
    const increaseProgress = 100 / undoableTimeout;
    const timer = setInterval(() => {
      setTimeCount((prevProgress) => prevProgress - 1);
      setProgress((prevProgress) => prevProgress - increaseProgress);
    }, 1e3);
    if (timeCount === 0) {
      clearInterval(timer);
    }
    return () => {
      clearInterval(timer);
    };
  }, [timeCount]);
  return /* @__PURE__ */ React27.createElement(React27.Fragment, null, /* @__PURE__ */ React27.createElement(Box11, { sx: { position: "relative", display: "inline-flex" } }, /* @__PURE__ */ React27.createElement(
    CircularProgress4,
    {
      color: "inherit",
      variant: "determinate",
      value: progress
    }
  ), /* @__PURE__ */ React27.createElement(
    Box11,
    {
      sx: {
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
        position: "absolute",
        display: "flex",
        alignItems: "center",
        justifyContent: "center"
      }
    },
    /* @__PURE__ */ React27.createElement(Typography12, { component: "div" }, timeCount)
  )), /* @__PURE__ */ React27.createElement(
    Box11,
    {
      sx: {
        marginLeft: "10px",
        maxWidth: { xs: "150px", md: "100%" }
      }
    },
    /* @__PURE__ */ React27.createElement(Typography12, { variant: "subtitle2" }, message)
  ));
}, "CircularDeterminate");

// src/components/fields/text/index.tsx
import React28 from "react";
import Typography13 from "@mui/material/Typography";
var TextField5 = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ React28.createElement(Typography13, { variant: "body2", ...rest }, value);
}, "TextField");

// src/components/fields/tag/index.tsx
import React29 from "react";
import Chip from "@mui/material/Chip";
var TagField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ React29.createElement(Chip, { label: value == null ? void 0 : value.toString(), ...rest });
}, "TagField");

// src/components/fields/email/index.tsx
import React30 from "react";
import Typography14 from "@mui/material/Typography";
import Link from "@mui/material/Link";
var EmailField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ React30.createElement(Typography14, { variant: "body2" }, /* @__PURE__ */ React30.createElement(Link, { href: `mailto:${value}`, ...rest }, value));
}, "EmailField");

// src/components/fields/boolean/index.tsx
import React31 from "react";
import Tooltip3 from "@mui/material/Tooltip";
import CheckOutlined from "@mui/icons-material/esm/CheckOutlined";
import CloseOutlined from "@mui/icons-material/esm/CloseOutlined";
var BooleanField = /* @__PURE__ */ __name(({
  value,
  valueLabelTrue = "true",
  valueLabelFalse = "false",
  trueIcon,
  falseIcon,
  svgIconProps,
  ...rest
}) => {
  return /* @__PURE__ */ React31.createElement(Tooltip3, { title: value ? valueLabelTrue : valueLabelFalse, ...rest }, value ? /* @__PURE__ */ React31.createElement("span", null, trueIcon ?? /* @__PURE__ */ React31.createElement(CheckOutlined, { ...svgIconProps })) : /* @__PURE__ */ React31.createElement("span", null, falseIcon ?? /* @__PURE__ */ React31.createElement(CloseOutlined, { ...svgIconProps })));
}, "BooleanField");

// src/components/fields/date/index.tsx
import React32 from "react";
import dayjs from "dayjs";
import LocalizedFormat from "dayjs/plugin/localizedFormat.js";
import Typography15 from "@mui/material/Typography";
dayjs.extend(LocalizedFormat);
var defaultLocale = dayjs.locale();
var DateField = /* @__PURE__ */ __name(({
  value,
  locales,
  format: dateFormat = "L",
  ...rest
}) => {
  return /* @__PURE__ */ React32.createElement(Typography15, { variant: "body2", ...rest }, value ? dayjs(value).locale(locales || defaultLocale).format(dateFormat) : "");
}, "DateField");

// src/components/fields/file/index.tsx
import React33 from "react";
var FileField = /* @__PURE__ */ __name(({
  title,
  src,
  ...rest
}) => {
  return /* @__PURE__ */ React33.createElement(UrlField, { value: src, title, ...rest }, title ?? src);
}, "FileField");

// src/components/fields/url/index.tsx
import React34 from "react";
import Link2 from "@mui/material/Link";
import Typography16 from "@mui/material/Typography";
var UrlField = /* @__PURE__ */ __name(({
  children,
  value,
  ...rest
}) => {
  return /* @__PURE__ */ React34.createElement(Typography16, { variant: "body2" }, /* @__PURE__ */ React34.createElement(Link2, { href: value, ...rest }, children ?? value));
}, "UrlField");

// src/components/fields/number/index.tsx
import React35 from "react";
import Typography17 from "@mui/material/Typography";
function toLocaleStringSupportsOptions() {
  return !!(typeof Intl === "object" && Intl && typeof Intl.NumberFormat === "function");
}
__name(toLocaleStringSupportsOptions, "toLocaleStringSupportsOptions");
var NumberField = /* @__PURE__ */ __name(({
  value,
  locale,
  options,
  ...rest
}) => {
  const number = Number(value);
  return /* @__PURE__ */ React35.createElement(Typography17, { variant: "body2", ...rest }, toLocaleStringSupportsOptions() ? number.toLocaleString(locale, options) : number);
}, "NumberField");

// src/components/fields/markdown/index.tsx
import React36 from "react";
import ReactMarkdown from "react-markdown";
import gfm from "remark-gfm";
var MarkdownField = /* @__PURE__ */ __name(({ value = "" }) => {
  return /* @__PURE__ */ React36.createElement(
    ReactMarkdown,
    {
      remarkPlugins: [gfm]
    },
    value
  );
}, "MarkdownField");

// src/components/breadcrumb/index.tsx
import React37 from "react";
import {
  matchResourceFromRoute,
  useBreadcrumb,
  useLink as useLink6,
  useResourceParams as useResourceParams7
} from "@refinedev/core";
import Breadcrumbs from "@mui/material/Breadcrumbs";
import Typography18 from "@mui/material/Typography";
import Grid2 from "@mui/material/Grid2";
import HomeOutlined from "@mui/icons-material/esm/HomeOutlined";
var Breadcrumb = /* @__PURE__ */ __name(({
  breadcrumbProps,
  showHome = true,
  hideIcons = false,
  meta,
  minItems = 2
}) => {
  var _a, _b;
  const { breadcrumbs } = useBreadcrumb({ meta });
  const Link3 = useLink6();
  const { resources } = useResourceParams7();
  const rootRouteResource = matchResourceFromRoute("/", resources);
  if (breadcrumbs.length < minItems)
    return null;
  const LinkRouter = /* @__PURE__ */ __name((props) => {
    const { to, children, ...restProps } = props;
    return /* @__PURE__ */ React37.createElement(Link3, { to: to || "" }, /* @__PURE__ */ React37.createElement("span", { ...restProps }, children));
  }, "LinkRouter");
  return /* @__PURE__ */ React37.createElement(
    Breadcrumbs,
    {
      "aria-label": "breadcrumb",
      sx: {
        padding: 2,
        ...(breadcrumbProps == null ? void 0 : breadcrumbProps.sx) ?? {}
      },
      ...breadcrumbProps
    },
    showHome && rootRouteResource.found && /* @__PURE__ */ React37.createElement(
      LinkRouter,
      {
        underline: "hover",
        sx: {
          display: "flex",
          alignItems: "center"
        },
        color: "inherit",
        to: "/"
      },
      ((_b = (_a = rootRouteResource == null ? void 0 : rootRouteResource.resource) == null ? void 0 : _a.meta) == null ? void 0 : _b.icon) ?? /* @__PURE__ */ React37.createElement(
        HomeOutlined,
        {
          sx: {
            fontSize: "18px"
          }
        }
      )
    ),
    breadcrumbs.map(({ label, icon, href }) => {
      return /* @__PURE__ */ React37.createElement(
        Grid2,
        {
          key: label,
          sx: {
            display: "flex",
            alignItems: "center",
            "& .MuiSvgIcon-root": {
              fontSize: "16px"
            }
          }
        },
        !hideIcons && icon,
        href ? /* @__PURE__ */ React37.createElement(
          LinkRouter,
          {
            underline: "hover",
            sx: {
              display: "flex",
              alignItems: "center",
              fontSize: "14px",
              marginLeft: 0.5
            },
            color: "inherit",
            to: href,
            variant: "subtitle1"
          },
          label
        ) : /* @__PURE__ */ React37.createElement(Typography18, { fontSize: "14px" }, label)
      );
    })
  );
}, "Breadcrumb");

// src/components/autoSaveIndicator/index.tsx
import React38 from "react";
import {
  useTranslate as useTranslate11,
  AutoSaveIndicator as AutoSaveIndicatorCore
} from "@refinedev/core";
import Typography19 from "@mui/material/Typography";
import MoreHorizOutlinedIcon from "@mui/icons-material/esm/MoreHorizOutlined";
import SyncOutlinedIcon from "@mui/icons-material/esm/SyncOutlined";
import TaskAltOutlinedIcon from "@mui/icons-material/esm/TaskAltOutlined";
import ErrorOutlineOutlinedIcon from "@mui/icons-material/esm/ErrorOutlineOutlined";
var AutoSaveIndicator = /* @__PURE__ */ __name(({
  status,
  elements: {
    success = /* @__PURE__ */ React38.createElement(
      Message,
      {
        translationKey: "autoSave.success",
        defaultMessage: "saved",
        icon: /* @__PURE__ */ React38.createElement(TaskAltOutlinedIcon, { fontSize: "small" })
      }
    ),
    error = /* @__PURE__ */ React38.createElement(
      Message,
      {
        translationKey: "autoSave.error",
        defaultMessage: "auto save failure",
        icon: /* @__PURE__ */ React38.createElement(ErrorOutlineOutlinedIcon, { fontSize: "small" })
      }
    ),
    loading = /* @__PURE__ */ React38.createElement(
      Message,
      {
        translationKey: "autoSave.loading",
        defaultMessage: "saving...",
        icon: /* @__PURE__ */ React38.createElement(SyncOutlinedIcon, { fontSize: "small" })
      }
    ),
    idle = /* @__PURE__ */ React38.createElement(
      Message,
      {
        translationKey: "autoSave.idle",
        defaultMessage: "waiting for changes",
        icon: /* @__PURE__ */ React38.createElement(MoreHorizOutlinedIcon, { fontSize: "small" })
      }
    )
  } = {}
}) => {
  return /* @__PURE__ */ React38.createElement(
    AutoSaveIndicatorCore,
    {
      status,
      elements: {
        success,
        error,
        loading,
        idle
      }
    }
  );
}, "AutoSaveIndicator");
var Message = /* @__PURE__ */ __name(({
  translationKey,
  defaultMessage,
  icon
}) => {
  const translate = useTranslate11();
  return /* @__PURE__ */ React38.createElement(
    Typography19,
    {
      color: "gray",
      fontSize: "0.8rem",
      position: "relative",
      display: "flex",
      alignItems: "center",
      flexWrap: "wrap",
      marginRight: ".3rem"
    },
    translate(translationKey, defaultMessage),
    /* @__PURE__ */ React38.createElement("span", { style: { position: "relative", top: "3px", marginLeft: "3px" } }, icon)
  );
}, "Message");

// src/theme/index.ts
import { createTheme } from "@mui/material/styles";

// src/theme/palette/lightPalette.ts
var lightPalette = {
  mode: "light",
  primary: {
    main: "#67be23",
    contrastText: "#fff"
  },
  secondary: {
    main: "#2A132E",
    contrastText: "#fff"
  },
  background: {
    default: "#f0f0f0",
    paper: "#ffffff"
  },
  success: {
    main: "#67be23",
    contrastText: "#fff"
  },
  error: {
    main: "#fa541c",
    contrastText: "#fff"
  },
  warning: {
    main: "#fa8c16",
    contrastText: "#fff"
  },
  info: {
    main: "#0b82f0",
    contrastText: "#fff"
  },
  divider: "rgba(0,0,0,0)",
  text: {
    primary: "#626262",
    secondary: "#9f9f9f",
    disabled: "#c1c1c1"
  }
};

// src/theme/palette/darkPalette.ts
var darkPalette = {
  mode: "dark",
  primary: {
    main: "#67be23",
    contrastText: "#fff"
  },
  secondary: {
    main: "#2A132E",
    contrastText: "#fff"
  },
  background: {
    default: "#212121",
    paper: "#242424"
  },
  success: {
    main: "#67be23",
    contrastText: "#fff"
  },
  error: {
    main: "#ee2a1e",
    contrastText: "#fff"
  },
  warning: {
    main: "#fa8c16",
    contrastText: "#fff"
  },
  info: {
    main: "#1890ff",
    contrastText: "#fff"
  },
  divider: "rgba(0,0,0,0)",
  text: {
    primary: "#fff",
    secondary: "rgba(255,255,255,0.7)",
    disabled: "#d1d1d1"
  }
};

// src/theme/typography.ts
var typography = {
  fontFamily: [
    "Montserrat",
    "-apple-system",
    "BlinkMacSystemFont",
    '"Segoe UI"',
    "Roboto",
    '"Helvetica Neue"',
    "Arial",
    "sans-serif",
    '"Apple Color Emoji"',
    '"Segoe UI Emoji"',
    '"Segoe UI Symbol"'
  ].join(",")
};

// src/theme/palette/refinePalette.ts
var RefinePalettes = {
  Blue: {
    mode: "light",
    primary: {
      main: "#1976D2",
      light: "#4791db",
      dark: "#115293"
    }
  },
  BlueDark: {
    mode: "dark",
    primary: {
      main: "#67b7f7",
      light: "#85c5f8",
      dark: "#4880ac"
    }
  },
  Purple: {
    mode: "light",
    primary: {
      main: "#7B1FA2",
      light: "#954bb4",
      dark: "#561571"
    }
  },
  PurpleDark: {
    mode: "dark",
    primary: {
      main: "#AB47BC",
      light: "#bb6bc9",
      dark: "#773183"
    }
  },
  Magenta: {
    mode: "light",
    primary: {
      main: "#C2185B",
      light: "#ce467b",
      dark: "#87103f"
    }
  },
  MagentaDark: {
    mode: "dark",
    primary: {
      main: "#EC407A",
      light: "#ef6694",
      dark: "#a52c55"
    }
  },
  Red: {
    mode: "light",
    primary: {
      main: "#D32F2F",
      light: "#db5858",
      dark: "#932020"
    }
  },
  RedDark: {
    mode: "dark",
    primary: {
      main: "#EF5350",
      light: "#f27573",
      dark: "#a73a38"
    }
  },
  Orange: {
    mode: "light",
    primary: {
      main: "#F57C00",
      light: "#f79633",
      dark: "#ab5600"
    }
  },
  OrangeDark: {
    mode: "dark",
    primary: {
      main: "#FFA726",
      light: "#ffb851",
      dark: "#b2741a"
    }
  },
  Yellow: {
    mode: "light",
    primary: {
      main: "#FFA000",
      light: "#ffb333",
      dark: "#b27000"
    }
  },
  YellowDark: {
    mode: "dark",
    primary: {
      main: "#FFCA28",
      light: "#ffd453",
      dark: "#E87040"
    }
  },
  Green: {
    mode: "light",
    primary: {
      main: "#689F38",
      light: "#86b25f",
      dark: "#486f27"
    }
  },
  GreenDark: {
    mode: "dark",
    primary: {
      main: "#9CCC65",
      light: "#afd683",
      dark: "#6d8e46"
    }
  }
};

// src/theme/index.ts
var commonThemeProperties = {
  shape: {
    borderRadius: 6
  },
  typography: {
    ...typography
  }
};
var LightTheme = createTheme({
  ...commonThemeProperties,
  palette: lightPalette,
  components: {
    MuiAppBar: {
      styleOverrides: {
        colorDefault: {
          backgroundColor: "#fff"
        }
      }
    },
    MuiPaper: {
      styleOverrides: {
        root: {
          backgroundImage: "linear-gradient(rgba(255, 255, 255, 0.01), rgba(255, 255, 255, 0.01))"
        }
      }
    },
    MuiTypography: {
      styleOverrides: {
        h5: {
          fontWeight: 800,
          lineHeight: "2rem"
        }
      }
    }
  }
});
var DarkTheme = createTheme({
  ...commonThemeProperties,
  palette: darkPalette,
  components: {
    MuiPaper: {
      styleOverrides: {
        root: {
          backgroundImage: "linear-gradient(rgba(255, 255, 255, 0.025), rgba(255, 255, 255, 0.025))"
        }
      }
    },
    MuiAppBar: {
      defaultProps: {
        color: "transparent"
      }
    },
    MuiTypography: {
      styleOverrides: {
        h5: {
          fontWeight: 800,
          lineHeight: "2rem"
        }
      }
    }
  }
});
var RefineThemes = Object.keys(RefinePalettes).reduce((acc, key) => {
  const paletteName = key;
  return {
    ...acc,
    [key]: createTheme({
      palette: {
        ...RefinePalettes[paletteName]
      },
      components: {
        MuiButton: {
          styleOverrides: {
            root: ({ ownerState }) => ({
              ...ownerState.variant === "contained" && ownerState.color === "primary" && {
                color: "#fff"
              }
            })
          }
        }
      }
    })
  };
}, {});

// src/providers/notificationProvider/index.tsx
import React39 from "react";
import { useSnackbar } from "notistack";
import Box12 from "@mui/material/Box";
import IconButton6 from "@mui/material/IconButton";
import Typography20 from "@mui/material/Typography";
import UndoOutlined from "@mui/icons-material/esm/UndoOutlined";
var useNotificationProvider = /* @__PURE__ */ __name(() => {
  const { closeSnackbar, enqueueSnackbar } = useSnackbar();
  const notificationProvider = {
    open: ({
      message,
      type,
      undoableTimeout,
      key,
      cancelMutation,
      description
    }) => {
      if (type === "progress") {
        const action = /* @__PURE__ */ __name((key2) => /* @__PURE__ */ React39.createElement(
          IconButton6,
          {
            onClick: () => {
              cancelMutation == null ? void 0 : cancelMutation();
              closeSnackbar(key2);
            },
            color: "inherit"
          },
          /* @__PURE__ */ React39.createElement(UndoOutlined, null)
        ), "action");
        enqueueSnackbar(
          /* @__PURE__ */ React39.createElement(React39.Fragment, null, /* @__PURE__ */ React39.createElement(
            CircularDeterminate,
            {
              undoableTimeout: undoableTimeout ?? 0,
              message
            }
          )),
          {
            action,
            preventDuplicate: true,
            key,
            autoHideDuration: (undoableTimeout ?? 0) * 1e3
          }
        );
      } else {
        enqueueSnackbar(
          /* @__PURE__ */ React39.createElement(Box12, null, /* @__PURE__ */ React39.createElement(Typography20, { variant: "subtitle2", component: "h6" }, description), /* @__PURE__ */ React39.createElement(Typography20, { variant: "caption", component: "p" }, message)),
          {
            key,
            variant: type
          }
        );
      }
    },
    close: (key) => {
      closeSnackbar(key);
    }
  };
  return notificationProvider;
}, "useNotificationProvider");

// src/providers/refineSnackbarProvider/index.tsx
import React40 from "react";
import { styled } from "@mui/material/styles";
import { SnackbarProvider } from "notistack";
var SnackbarProviderWithDefaultValues = /* @__PURE__ */ __name(({
  anchorOrigin = {
    vertical: "top",
    horizontal: "right"
  },
  disableWindowBlurListener = true,
  ...rest
}) => {
  return /* @__PURE__ */ React40.createElement(
    SnackbarProvider,
    {
      anchorOrigin,
      disableWindowBlurListener,
      ...rest
    }
  );
}, "SnackbarProviderWithDefaultValues");
var RefineSnackbarProvider = styled(SnackbarProviderWithDefaultValues)`
&.SnackbarItem-contentRoot {
    background-color: ${(props) => props.theme.palette.background.default};
    color: ${(props) => props.theme.palette.primary.main};
}
&.SnackbarItem-variantSuccess {
    background-color: ${(props) => props.theme.palette.success.main};
    color: ${(props) => props.theme.palette.success.contrastText};
}
&.SnackbarItem-variantError {
    background-color: ${(props) => props.theme.palette.error.main};
    color: ${(props) => props.theme.palette.error.contrastText};
}
&.SnackbarItem-variantInfo {
    background-color: ${(props) => props.theme.palette.info.main};
    color: ${(props) => props.theme.palette.info.contrastText};
}
&.SnackbarItem-variantWarning {
    background-color: ${(props) => props.theme.palette.warning.main};
    color: ${(props) => props.theme.palette.warning.contrastText};
}
`;
export {
  AuthPage,
  AutoSaveIndicator,
  BooleanField,
  Breadcrumb,
  CircularDeterminate,
  CloneButton,
  Create,
  CreateButton,
  DarkTheme,
  DateField,
  DeleteButton,
  Edit,
  EditButton,
  EmailField,
  ErrorComponent,
  ExportButton,
  FileField,
  HamburgerMenu,
  ImportButton,
  LightTheme,
  List2 as List,
  ListButton,
  MarkdownField,
  SnackbarContent as NotistackSnackbarContent,
  NumberField,
  RefineSnackbarProvider,
  RefineThemes,
  RefreshButton,
  SaveButton,
  Show,
  ShowButton,
  SnackbarProvider2 as SnackbarProvider,
  TagField,
  TextField5 as TextFieldComponent,
  ThemedHeader,
  ThemedLayout,
  ThemedLayoutContext,
  ThemedLayoutContextProvider,
  ThemedSider,
  ThemedTitle,
  UrlField,
  useAutocomplete,
  useDataGrid,
  useNotificationProvider,
  useSnackbar2 as useSnackbar,
  useThemedLayoutContext,
  withSnackbar
};
