// @refinedev/mui/dist/index.cjs
"use client"
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __name = (target, value) => __defProp(target, "name", { value, configurable: true });
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.tsx
var src_exports = {};
__export(src_exports, {
  AuthPage: () => AuthPage,
  AutoSaveIndicator: () => AutoSaveIndicator,
  BooleanField: () => BooleanField,
  Breadcrumb: () => Breadcrumb,
  CircularDeterminate: () => CircularDeterminate,
  CloneButton: () => CloneButton,
  Create: () => Create,
  CreateButton: () => CreateButton,
  DarkTheme: () => DarkTheme,
  DateField: () => DateField,
  DeleteButton: () => DeleteButton,
  Edit: () => Edit,
  EditButton: () => EditButton,
  EmailField: () => EmailField,
  ErrorComponent: () => ErrorComponent,
  ExportButton: () => ExportButton,
  FileField: () => FileField,
  HamburgerMenu: () => HamburgerMenu,
  ImportButton: () => ImportButton,
  LightTheme: () => LightTheme,
  List: () => List2,
  ListButton: () => ListButton,
  MarkdownField: () => MarkdownField,
  NotistackSnackbarContent: () => import_notistack3.SnackbarContent,
  NumberField: () => NumberField,
  RefineSnackbarProvider: () => RefineSnackbarProvider,
  RefineThemes: () => RefineThemes,
  RefreshButton: () => RefreshButton,
  SaveButton: () => SaveButton,
  Show: () => Show,
  ShowButton: () => ShowButton,
  SnackbarProvider: () => import_notistack3.SnackbarProvider,
  TagField: () => TagField,
  TextFieldComponent: () => TextField5,
  ThemedHeader: () => ThemedHeader,
  ThemedLayout: () => ThemedLayout,
  ThemedLayoutContext: () => ThemedLayoutContext,
  ThemedLayoutContextProvider: () => ThemedLayoutContextProvider,
  ThemedSider: () => ThemedSider,
  ThemedTitle: () => ThemedTitle,
  UrlField: () => UrlField,
  useAutocomplete: () => useAutocomplete,
  useDataGrid: () => useDataGrid,
  useNotificationProvider: () => useNotificationProvider,
  useSnackbar: () => import_notistack3.useSnackbar,
  useThemedLayoutContext: () => useThemedLayoutContext,
  withSnackbar: () => import_notistack3.withSnackbar
});
module.exports = __toCommonJS(src_exports);
var import_notistack3 = require("notistack");

// src/components/themedLayout/index.tsx
var import_react7 = __toESM(require("react"));
var import_Box2 = __toESM(require("@mui/material/Box"));

// src/contexts/themedLayoutContext/index.tsx
var import_react = __toESM(require("react"));
var ThemedLayoutContext = import_react.default.createContext({
  siderCollapsed: false,
  mobileSiderOpen: false,
  setSiderCollapsed: () => void 0,
  setMobileSiderOpen: () => void 0
});
var ThemedLayoutContextProvider = /* @__PURE__ */ __name(({ children, initialSiderCollapsed, onSiderCollapsed }) => {
  const [siderCollapsed, setSiderCollapsedState] = (0, import_react.useState)(
    initialSiderCollapsed ?? false
  );
  const [mobileSiderOpen, setMobileSiderOpen] = (0, import_react.useState)(false);
  const setSiderCollapsed = /* @__PURE__ */ __name((collapsed) => {
    setSiderCollapsedState(collapsed);
    if (onSiderCollapsed) {
      onSiderCollapsed(collapsed);
    }
  }, "setSiderCollapsed");
  return /* @__PURE__ */ import_react.default.createElement(
    ThemedLayoutContext.Provider,
    {
      value: {
        siderCollapsed,
        mobileSiderOpen,
        setSiderCollapsed,
        setMobileSiderOpen,
        onSiderCollapsed
      }
    },
    children
  );
}, "ThemedLayoutContextProvider");

// src/components/themedLayout/sider/index.tsx
var import_react4 = __toESM(require("react"));
var import_Box = __toESM(require("@mui/material/Box"));
var import_Drawer = __toESM(require("@mui/material/Drawer"));
var import_List = __toESM(require("@mui/material/List"));
var import_ListItemButton = __toESM(require("@mui/material/ListItemButton"));
var import_ListItemIcon = __toESM(require("@mui/material/ListItemIcon"));
var import_ListItemText = __toESM(require("@mui/material/ListItemText"));
var import_Collapse = __toESM(require("@mui/material/Collapse"));
var import_Tooltip = __toESM(require("@mui/material/Tooltip"));
var import_IconButton = __toESM(require("@mui/material/IconButton"));
var import_Paper = __toESM(require("@mui/material/Paper"));
var import_ListOutlined = __toESM(require("@mui/icons-material/ListOutlined"));
var import_Logout = __toESM(require("@mui/icons-material/Logout"));
var import_ExpandLess = __toESM(require("@mui/icons-material/ExpandLess"));
var import_ExpandMore = __toESM(require("@mui/icons-material/ExpandMore"));
var import_ChevronLeft = __toESM(require("@mui/icons-material/ChevronLeft"));
var import_Dashboard = __toESM(require("@mui/icons-material/Dashboard"));
var import_core3 = require("@refinedev/core");

// src/hooks/useAutocomplete/index.ts
var import_core = require("@refinedev/core");
var import_isEqual = __toESM(require("lodash/isEqual"));
var import_unionWith = __toESM(require("lodash/unionWith"));
var useAutocomplete = /* @__PURE__ */ __name((props) => {
  var _a, _b, _c, _d;
  const { query, defaultValueQuery, onSearch, overtime } = (0, import_core.useSelect)(props);
  return {
    autocompleteProps: {
      options: props.selectedOptionsOrder === "selected-first" ? (0, import_unionWith.default)(
        ((_a = defaultValueQuery.result) == null ? void 0 : _a.data) || [],
        ((_b = query.data) == null ? void 0 : _b.data) || [],
        import_isEqual.default
      ) : (0, import_unionWith.default)(
        ((_c = query.data) == null ? void 0 : _c.data) || [],
        ((_d = defaultValueQuery.result) == null ? void 0 : _d.data) || [],
        import_isEqual.default
      ),
      loading: query.isFetching || defaultValueQuery.query.isFetching,
      onInputChange: (event, value) => {
        if ((event == null ? void 0 : event.type) === "change") {
          onSearch(value);
        } else if ((event == null ? void 0 : event.type) === "click") {
          onSearch("");
        }
      },
      filterOptions: (x) => x
    },
    onSearch,
    query,
    defaultValueQuery,
    overtime
  };
}, "useAutocomplete");

// src/hooks/useDataGrid/index.ts
var import_core2 = require("@refinedev/core");
var import_react2 = require("react");
var import_differenceWith = __toESM(require("lodash/differenceWith"));
var import_isEqual2 = __toESM(require("lodash/isEqual"));

// src/definitions/dataGrid/index.ts
var import_x_data_grid = require("@mui/x-data-grid");
var transformSortModelToCrudSorting = /* @__PURE__ */ __name((sortModel) => {
  const sorter = sortModel.map(({ field, sort }) => ({
    field,
    order: sort || "asc"
  }));
  return sorter;
}, "transformSortModelToCrudSorting");
var transformCrudSortingToSortModel = /* @__PURE__ */ __name((crudSorting) => {
  const sortModel = crudSorting.map(({ field, order }) => ({
    field,
    sort: order
  }));
  return sortModel;
}, "transformCrudSortingToSortModel");
var transformMuiOperatorToCrudOperator = /* @__PURE__ */ __name((operatorValue) => {
  if (!operatorValue) {
    return "eq";
  }
  switch (operatorValue) {
    case "equals":
    case "is":
    case "=":
      return "eq";
    case "!=":
    case "not":
    case "doesNotEqual":
      return "ne";
    case "contains":
      return "contains";
    case "doesNotContain":
      return "ncontains";
    case "isAnyOf":
      return "in";
    case ">":
    case "after":
      return "gt";
    case ">=":
    case "onOrAfter":
      return "gte";
    case "<":
    case "before":
      return "lt";
    case "<=":
    case "onOrBefore":
      return "lte";
    case "startsWith":
      return "startswith";
    case "endsWith":
      return "endswith";
    case "isEmpty":
      return "null";
    case "isNotEmpty":
      return "nnull";
    default:
      return operatorValue;
  }
}, "transformMuiOperatorToCrudOperator");
var transformFilterModelToCrudFilters = /* @__PURE__ */ __name(({
  items,
  logicOperator
}) => {
  const filters = items.map(({ field, value, operator }) => {
    const filter = {
      field,
      value: ["isEmpty", "isNotEmpty"].includes(operator) ? true : value ?? "",
      operator: transformMuiOperatorToCrudOperator(operator)
    };
    return filter;
  });
  if (logicOperator === import_x_data_grid.GridLogicOperator.Or) {
    return [{ operator: "or", value: filters }];
  }
  return filters;
}, "transformFilterModelToCrudFilters");
var transformCrudOperatorToMuiOperator = /* @__PURE__ */ __name((operator, columnType) => {
  switch (columnType) {
    case "number":
      switch (operator) {
        case "eq":
          return "=";
        case "ne":
          return "!=";
        case "gt":
          return ">";
        case "gte":
          return ">=";
        case "lt":
          return "<";
        case "lte":
          return "<=";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        case "in":
          return "isAnyOf";
        default:
          return operator;
      }
    case "singleSelect":
    case "boolean":
      switch (operator) {
        case "eq":
          return "is";
        default:
          return operator;
      }
    case void 0:
    case "string":
      switch (operator) {
        case "eq":
          return "equals";
        case "ne":
          return "doesNotEqual";
        case "contains":
          return "contains";
        case "ncontains":
          return "doesNotContain";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        case "startswith":
          return "startsWith";
        case "endswith":
          return "endsWith";
        case "in":
          return "isAnyOf";
        default:
          return operator;
      }
    case "date":
    case "dateTime":
      switch (operator) {
        case "eq":
          return "is";
        case "ne":
          return "not";
        case "gt":
          return "after";
        case "gte":
          return "onOrAfter";
        case "lt":
          return "before";
        case "lte":
          return "onOrBefore";
        case "null":
          return "isEmpty";
        case "nnull":
          return "isNotEmpty";
        default:
          return operator;
      }
    default:
      return operator;
  }
}, "transformCrudOperatorToMuiOperator");
var transformCrudFiltersToFilterModel = /* @__PURE__ */ __name((crudFilters, columnsType) => {
  var _a;
  const gridFilterItems = [];
  const fieldOperatorCount = {};
  const isExistOrFilter = crudFilters.some(
    (filter) => filter.operator === "or"
  );
  if (columnsType) {
    if (isExistOrFilter) {
      const orLogicalFilters = (_a = crudFilters.find(
        (filter) => filter.operator === "or"
      )) == null ? void 0 : _a.value;
      orLogicalFilters.map(({ field, value, operator }) => {
        const columnType = columnsType[field];
        const id = field + operator;
        fieldOperatorCount[id] = (fieldOperatorCount[id] || 0) + 1;
        const uniqueId = id + String(fieldOperatorCount[id]);
        gridFilterItems.push({
          field,
          operator: transformCrudOperatorToMuiOperator(operator, columnType),
          value: value === "" ? void 0 : value,
          id: fieldOperatorCount[id] > 1 ? uniqueId : id
        });
      });
    } else {
      crudFilters.map(({ field, value, operator }) => {
        const columnType = columnsType[field];
        const id = field + operator;
        fieldOperatorCount[id] = (fieldOperatorCount[id] || 0) + 1;
        const uniqueId = id + String(fieldOperatorCount[id]);
        gridFilterItems.push({
          field,
          operator: transformCrudOperatorToMuiOperator(operator, columnType),
          value: value === "" ? void 0 : value,
          id: fieldOperatorCount[id] > 1 ? uniqueId : id
        });
      });
    }
  }
  return {
    items: gridFilterItems,
    // If there is "or" filter, default link operator is "or"
    logicOperator: isExistOrFilter ? import_x_data_grid.GridLogicOperator.Or : import_x_data_grid.GridLogicOperator.And
  };
}, "transformCrudFiltersToFilterModel");

// src/hooks/useDataGrid/index.ts
var defaultPermanentFilter = [];
var defaultPermanentSort = [];
var DEFAULT_FILTER_DEBOUNCE_MS = 300;
function useDataGrid({
  onSearch: onSearchProp,
  pagination = { pageSize: 25 },
  filters: filtersFromProp,
  sorters: sortersFromProp,
  syncWithLocation: syncWithLocationProp,
  resource: resourceFromProp,
  successNotification,
  errorNotification,
  queryOptions,
  liveMode: liveModeFromProp,
  onLiveEvent,
  liveParams,
  meta,
  dataProviderName,
  overtimeOptions,
  editable = false,
  updateMutationOptions
} = {}) {
  const liveMode = (0, import_core2.useLiveMode)(liveModeFromProp);
  const columnsTypes = (0, import_react2.useRef)({});
  const filterDebounceRef = (0, import_react2.useRef)(null);
  const { identifier } = (0, import_core2.useResourceParams)({ resource: resourceFromProp });
  const {
    tableQuery,
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    filters,
    setFilters,
    sorters,
    setSorters,
    pageCount,
    createLinkForSyncWithLocation,
    overtime,
    result
  } = (0, import_core2.useTable)({
    pagination: {
      ...pagination,
      pageSize: (pagination == null ? void 0 : pagination.pageSize) ?? 25
    },
    filters: { ...filtersFromProp, defaultBehavior: "replace" },
    sorters: sortersFromProp,
    syncWithLocation: syncWithLocationProp,
    resource: resourceFromProp,
    successNotification,
    errorNotification,
    queryOptions,
    liveMode: liveModeFromProp,
    onLiveEvent,
    liveParams,
    meta,
    dataProviderName,
    overtimeOptions
  });
  const [muiCrudFilters, setMuiCrudFilters] = (0, import_react2.useState)(filters);
  const { data, isFetched, isLoading } = tableQuery;
  const rowCountRef = (0, import_react2.useRef)((data == null ? void 0 : data.total) || 0);
  const rowCount = (0, import_react2.useMemo)(() => {
    if (data == null ? void 0 : data.total) {
      rowCountRef.current = data.total;
    }
    return rowCountRef.current;
  }, [data]);
  const isServerSideFilteringEnabled = ((filtersFromProp == null ? void 0 : filtersFromProp.mode) || "server") === "server";
  const isServerSideSortingEnabled = ((sortersFromProp == null ? void 0 : sortersFromProp.mode) || "server") === "server";
  const isPaginationEnabled = ((pagination == null ? void 0 : pagination.mode) ?? "server") !== "off";
  const preferredPermanentSorters = (sortersFromProp == null ? void 0 : sortersFromProp.permanent) ?? defaultPermanentSort;
  const preferredPermanentFilters = (filtersFromProp == null ? void 0 : filtersFromProp.permanent) ?? defaultPermanentFilter;
  const handlePageChange = /* @__PURE__ */ __name((page) => {
    if (isPaginationEnabled) {
      setCurrentPage(page + 1);
    }
  }, "handlePageChange");
  const handlePageSizeChange = /* @__PURE__ */ __name((pageSize2) => {
    if (isPaginationEnabled) {
      setPageSize(pageSize2);
    }
  }, "handlePageSizeChange");
  const clearFilterDebounce = /* @__PURE__ */ __name(() => {
    if (filterDebounceRef.current) {
      clearTimeout(filterDebounceRef.current);
      filterDebounceRef.current = null;
    }
  }, "clearFilterDebounce");
  (0, import_react2.useEffect)(() => {
    return () => {
      clearFilterDebounce();
    };
  }, []);
  const applyFilters = /* @__PURE__ */ __name((crudFilters) => {
    setFilters(crudFilters.filter((f) => f.value !== ""));
    if (isPaginationEnabled) {
      setCurrentPage(1);
    }
  }, "applyFilters");
  const handleSortModelChange = /* @__PURE__ */ __name((sortModel) => {
    const crudSorting = transformSortModelToCrudSorting(sortModel);
    setSorters(crudSorting);
  }, "handleSortModelChange");
  const handleFilterModelChange = /* @__PURE__ */ __name((filterModel) => {
    const crudFilters = transformFilterModelToCrudFilters(filterModel);
    setMuiCrudFilters(crudFilters);
    if (isServerSideFilteringEnabled) {
      clearFilterDebounce();
      filterDebounceRef.current = setTimeout(() => {
        applyFilters(crudFilters);
      }, DEFAULT_FILTER_DEBOUNCE_MS);
      return;
    }
    applyFilters(crudFilters);
  }, "handleFilterModelChange");
  const search = /* @__PURE__ */ __name(async (value) => {
    if (onSearchProp) {
      const searchFilters = await onSearchProp(value);
      clearFilterDebounce();
      setMuiCrudFilters(searchFilters);
      applyFilters(searchFilters);
    }
  }, "search");
  const dataGridPaginationValues = /* @__PURE__ */ __name(() => {
    if (isPaginationEnabled) {
      return {
        paginationMode: "server",
        paginationModel: {
          page: currentPage - 1,
          pageSize
        },
        onPaginationModelChange: (model) => {
          handlePageChange(model.page);
          handlePageSizeChange(model.pageSize);
        }
      };
    }
    return {
      paginationMode: "client"
    };
  }, "dataGridPaginationValues");
  const { mutate } = (0, import_core2.useUpdate)({
    mutationOptions: updateMutationOptions
  });
  const processRowUpdate = /* @__PURE__ */ __name(async (newRow, oldRow) => {
    if (!editable) {
      return Promise.resolve(oldRow);
    }
    if (!identifier) {
      return Promise.reject(new Error("Resource is not defined"));
    }
    return new Promise((resolve, reject) => {
      mutate(
        {
          resource: identifier,
          id: newRow.id,
          values: newRow,
          meta: updateMutationOptions == null ? void 0 : updateMutationOptions.meta
        },
        {
          onError: (error) => {
            reject(error);
          },
          onSuccess: () => {
            resolve(newRow);
          }
        }
      );
    });
  }, "processRowUpdate");
  const transformedSortModel = (0, import_react2.useMemo)(
    () => transformCrudSortingToSortModel(
      (0, import_differenceWith.default)(sorters, preferredPermanentSorters, import_isEqual2.default)
    ),
    [sorters, preferredPermanentSorters]
  );
  return {
    tableQuery,
    dataGridProps: {
      disableRowSelectionOnClick: true,
      rows: (data == null ? void 0 : data.data) || [],
      loading: liveMode === "auto" ? isLoading : !isFetched,
      rowCount,
      ...dataGridPaginationValues(),
      sortingMode: isServerSideSortingEnabled ? "server" : "client",
      sortModel: transformedSortModel,
      onSortModelChange: handleSortModelChange,
      filterMode: isServerSideFilteringEnabled ? "server" : "client",
      // Disable DataGrid's debounce for server filtering to prevent input resets.
      filterDebounceMs: isServerSideFilteringEnabled ? 0 : void 0,
      filterModel: transformCrudFiltersToFilterModel(
        (0, import_differenceWith.default)(muiCrudFilters, preferredPermanentFilters, import_isEqual2.default),
        columnsTypes.current
      ),
      onFilterModelChange: handleFilterModelChange,
      onStateChange: (state) => {
        const newColumnsTypes = Object.fromEntries(
          Object.entries(state.columns.lookup).map(([key, value]) => {
            return [key, value.type];
          })
        );
        const isStateChanged = !(0, import_isEqual2.default)(newColumnsTypes, columnsTypes.current);
        if (isStateChanged) {
          columnsTypes.current = newColumnsTypes;
        }
      },
      processRowUpdate: editable ? processRowUpdate : void 0
    },
    currentPage,
    setCurrentPage,
    pageSize,
    setPageSize,
    pageCount,
    sorters,
    setSorters,
    filters,
    setFilters,
    search,
    createLinkForSyncWithLocation,
    overtime,
    result
  };
}
__name(useDataGrid, "useDataGrid");

// src/hooks/useThemedLayoutContext/index.ts
var import_react3 = require("react");
var useThemedLayoutContext = /* @__PURE__ */ __name(() => {
  const {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  } = (0, import_react3.useContext)(ThemedLayoutContext);
  return {
    mobileSiderOpen,
    siderCollapsed,
    setMobileSiderOpen,
    setSiderCollapsed
  };
}, "useThemedLayoutContext");

// src/components/themedLayout/sider/index.tsx
var ThemedSider = /* @__PURE__ */ __name(({
  Title: TitleFromProps,
  render,
  meta,
  activeItemDisabled = false,
  siderItemsAreCollapsed = true
}) => {
  const {
    siderCollapsed,
    setSiderCollapsed,
    mobileSiderOpen,
    setMobileSiderOpen
  } = useThemedLayoutContext();
  const drawerWidth = /* @__PURE__ */ __name(() => {
    if (siderCollapsed)
      return 56;
    return 240;
  }, "drawerWidth");
  const t = (0, import_core3.useTranslate)();
  const Link3 = (0, import_core3.useLink)();
  const translate = (0, import_core3.useTranslate)();
  const { menuItems, selectedKey, defaultOpenKeys } = (0, import_core3.useMenu)({ meta });
  const isExistAuthentication = (0, import_core3.useIsExistAuthentication)();
  const authProvider = (0, import_core3.useActiveAuthProvider)();
  const { warnWhen, setWarnWhen } = (0, import_core3.useWarnAboutChange)();
  const { mutate: mutateLogout } = (0, import_core3.useLogout)();
  const defaultExpandMenuItems = (() => {
    const defaultOpenKeys2 = {};
    if (siderItemsAreCollapsed)
      return defaultOpenKeys2;
    return menuItems.reduce((prev, curr) => {
      const { key } = curr;
      return {
        ...prev,
        [key]: true
      };
    }, {});
  })();
  const [open, setOpen] = (0, import_react4.useState)(
    defaultExpandMenuItems
  );
  import_react4.default.useEffect(() => {
    setOpen((previous) => {
      const previousKeys = Object.keys(previous);
      const previousOpenKeys = previousKeys.filter((key) => previous[key]);
      const uniqueKeys = /* @__PURE__ */ new Set([...previousOpenKeys, ...defaultOpenKeys]);
      const uniqueKeysRecord = Object.fromEntries(
        Array.from(uniqueKeys.values()).map((key) => [key, true])
      );
      return uniqueKeysRecord;
    });
  }, [defaultOpenKeys]);
  const RenderToTitle = TitleFromProps ?? ThemedTitle;
  const handleClick = /* @__PURE__ */ __name((key) => {
    setOpen({ ...open, [key]: !open[key] });
  }, "handleClick");
  const renderTreeView = /* @__PURE__ */ __name((tree, selectedKey2) => {
    return tree.map((item) => {
      const { icon, label, route, name, children, meta: meta2 } = item;
      const isOpen = open[item.key || ""] || false;
      const isSelected = item.key === selectedKey2;
      const isNested = !((meta2 == null ? void 0 : meta2.parent) === void 0);
      if (children.length > 0) {
        return /* @__PURE__ */ import_react4.default.createElement(
          import_core3.CanAccess,
          {
            key: item.key,
            resource: name,
            action: "list",
            params: {
              resource: item
            }
          },
          /* @__PURE__ */ import_react4.default.createElement("div", { key: item.key }, /* @__PURE__ */ import_react4.default.createElement(
            import_Tooltip.default,
            {
              title: label ?? name,
              placement: "right",
              disableHoverListener: !siderCollapsed,
              arrow: true
            },
            /* @__PURE__ */ import_react4.default.createElement(
              import_ListItemButton.default,
              {
                onClick: () => {
                  if (siderCollapsed) {
                    setSiderCollapsed(false);
                    if (!isOpen) {
                      handleClick(item.key || "");
                    }
                  } else {
                    handleClick(item.key || "");
                  }
                },
                sx: {
                  pl: isNested ? 4 : 2,
                  justifyContent: "center"
                }
              },
              /* @__PURE__ */ import_react4.default.createElement(
                import_ListItemIcon.default,
                {
                  sx: {
                    justifyContent: "center",
                    minWidth: "24px",
                    transition: "margin-right 0.3s",
                    marginRight: siderCollapsed ? "0px" : "12px",
                    color: "currentColor"
                  }
                },
                icon ?? /* @__PURE__ */ import_react4.default.createElement(import_ListOutlined.default, null)
              ),
              /* @__PURE__ */ import_react4.default.createElement(
                import_ListItemText.default,
                {
                  primary: label,
                  primaryTypographyProps: {
                    noWrap: true,
                    fontSize: "14px"
                  }
                }
              ),
              isOpen ? /* @__PURE__ */ import_react4.default.createElement(
                import_ExpandLess.default,
                {
                  sx: {
                    color: "text.icon"
                  }
                }
              ) : /* @__PURE__ */ import_react4.default.createElement(
                import_ExpandMore.default,
                {
                  sx: {
                    color: "text.icon"
                  }
                }
              )
            )
          ), !siderCollapsed && /* @__PURE__ */ import_react4.default.createElement(
            import_Collapse.default,
            {
              in: open[item.key || ""],
              timeout: "auto",
              unmountOnExit: true
            },
            /* @__PURE__ */ import_react4.default.createElement(import_List.default, { component: "div", disablePadding: true }, renderTreeView(children, selectedKey2))
          ))
        );
      }
      const linkStyle = activeItemDisabled && isSelected ? { pointerEvents: "none" } : {};
      return /* @__PURE__ */ import_react4.default.createElement(
        import_core3.CanAccess,
        {
          key: item.key,
          resource: name,
          action: "list",
          params: { resource: item }
        },
        /* @__PURE__ */ import_react4.default.createElement(
          import_Tooltip.default,
          {
            title: label ?? name,
            placement: "right",
            disableHoverListener: !siderCollapsed,
            arrow: true
          },
          /* @__PURE__ */ import_react4.default.createElement(
            import_ListItemButton.default,
            {
              component: Link3,
              to: route,
              selected: isSelected,
              style: linkStyle,
              onClick: () => {
                setMobileSiderOpen(false);
              },
              sx: {
                pl: isNested ? 4 : 2,
                py: isNested ? 1.25 : 1,
                justifyContent: "center",
                color: isSelected ? "primary.main" : "text.primary"
              }
            },
            /* @__PURE__ */ import_react4.default.createElement(
              import_ListItemIcon.default,
              {
                sx: {
                  justifyContent: "center",
                  transition: "margin-right 0.3s",
                  marginRight: siderCollapsed ? "0px" : "12px",
                  minWidth: "24px",
                  color: "currentColor"
                }
              },
              icon ?? /* @__PURE__ */ import_react4.default.createElement(import_ListOutlined.default, null)
            ),
            /* @__PURE__ */ import_react4.default.createElement(
              import_ListItemText.default,
              {
                primary: label,
                primaryTypographyProps: {
                  noWrap: true,
                  fontSize: "14px"
                }
              }
            )
          )
        )
      );
    });
  }, "renderTreeView");
  const dashboard = /* @__PURE__ */ import_react4.default.createElement(import_core3.CanAccess, { resource: "dashboard", action: "list" }, /* @__PURE__ */ import_react4.default.createElement(
    import_Tooltip.default,
    {
      title: translate("dashboard.title", "Dashboard"),
      placement: "right",
      disableHoverListener: !siderCollapsed,
      arrow: true
    },
    /* @__PURE__ */ import_react4.default.createElement(
      Link3,
      {
        to: "/",
        style: { textDecoration: "none" },
        onClick: () => {
          setMobileSiderOpen(false);
        }
      },
      /* @__PURE__ */ import_react4.default.createElement(
        import_ListItemButton.default,
        {
          selected: selectedKey === "/",
          sx: {
            pl: 2,
            py: 1,
            justifyContent: "center",
            color: selectedKey === "/" ? "primary.main" : "text.primary"
          }
        },
        /* @__PURE__ */ import_react4.default.createElement(
          import_ListItemIcon.default,
          {
            sx: {
              justifyContent: "center",
              minWidth: "24px",
              transition: "margin-right 0.3s",
              marginRight: siderCollapsed ? "0px" : "12px",
              color: "currentColor",
              fontSize: "14px"
            }
          },
          /* @__PURE__ */ import_react4.default.createElement(import_Dashboard.default, null)
        ),
        /* @__PURE__ */ import_react4.default.createElement(
          import_ListItemText.default,
          {
            primary: translate("dashboard.title", "Dashboard"),
            primaryTypographyProps: {
              noWrap: true,
              fontSize: "14px"
            }
          }
        )
      )
    )
  ));
  const handleLogout = /* @__PURE__ */ __name(() => {
    if (warnWhen) {
      const confirm = window.confirm(
        t(
          "warnWhenUnsavedChanges",
          "Are you sure you want to leave? You have unsaved changes."
        )
      );
      if (confirm) {
        setWarnWhen(false);
        mutateLogout();
      }
    } else {
      mutateLogout();
    }
  }, "handleLogout");
  const logout = isExistAuthentication && /* @__PURE__ */ import_react4.default.createElement(
    import_Tooltip.default,
    {
      title: t("buttons.logout", "Logout"),
      placement: "right",
      disableHoverListener: !siderCollapsed,
      arrow: true
    },
    /* @__PURE__ */ import_react4.default.createElement(
      import_ListItemButton.default,
      {
        key: "logout",
        onClick: () => handleLogout(),
        sx: {
          justifyContent: "center"
        }
      },
      /* @__PURE__ */ import_react4.default.createElement(
        import_ListItemIcon.default,
        {
          sx: {
            justifyContent: "center",
            minWidth: "24px",
            transition: "margin-right 0.3s",
            marginRight: siderCollapsed ? "0px" : "12px",
            color: "currentColor"
          }
        },
        /* @__PURE__ */ import_react4.default.createElement(import_Logout.default, null)
      ),
      /* @__PURE__ */ import_react4.default.createElement(
        import_ListItemText.default,
        {
          primary: t("buttons.logout", "Logout"),
          primaryTypographyProps: {
            noWrap: true,
            fontSize: "14px"
          }
        }
      )
    )
  );
  const items = renderTreeView(menuItems, selectedKey);
  const renderSider = /* @__PURE__ */ __name(() => {
    if (render) {
      return render({
        logout,
        items,
        collapsed: siderCollapsed
      });
    }
    return /* @__PURE__ */ import_react4.default.createElement(import_react4.default.Fragment, null, dashboard, items, logout);
  }, "renderSider");
  const drawer = /* @__PURE__ */ import_react4.default.createElement(
    import_List.default,
    {
      disablePadding: true,
      sx: {
        flexGrow: 1,
        paddingTop: "16px"
      }
    },
    renderSider()
  );
  return /* @__PURE__ */ import_react4.default.createElement(import_react4.default.Fragment, null, /* @__PURE__ */ import_react4.default.createElement(
    import_Box.default,
    {
      sx: {
        width: { xs: drawerWidth() },
        display: {
          xs: "none",
          md: "block"
        },
        transition: "width 0.3s ease"
      }
    }
  ), /* @__PURE__ */ import_react4.default.createElement(
    import_Box.default,
    {
      component: "nav",
      sx: {
        position: "fixed",
        zIndex: 1101,
        width: { sm: drawerWidth() },
        display: "flex"
      }
    },
    /* @__PURE__ */ import_react4.default.createElement(
      import_Drawer.default,
      {
        variant: "temporary",
        elevation: 2,
        open: mobileSiderOpen,
        onClose: () => setMobileSiderOpen(false),
        ModalProps: {
          keepMounted: true
          // Better open performance on mobile.
        },
        sx: {
          display: {
            sm: "block",
            md: "none"
          }
        }
      },
      /* @__PURE__ */ import_react4.default.createElement(
        import_Box.default,
        {
          sx: {
            width: drawerWidth()
          }
        },
        /* @__PURE__ */ import_react4.default.createElement(
          import_Box.default,
          {
            sx: {
              height: 64,
              display: "flex",
              alignItems: "center",
              paddingLeft: "16px",
              fontSize: "14px"
            }
          },
          /* @__PURE__ */ import_react4.default.createElement(RenderToTitle, { collapsed: false })
        ),
        drawer
      )
    ),
    /* @__PURE__ */ import_react4.default.createElement(
      import_Drawer.default,
      {
        variant: "permanent",
        sx: {
          display: { xs: "none", md: "block" },
          "& .MuiDrawer-paper": {
            width: drawerWidth(),
            overflow: "hidden",
            transition: "width 200ms cubic-bezier(0.4, 0, 0.6, 1) 0ms"
          }
        },
        open: true
      },
      /* @__PURE__ */ import_react4.default.createElement(
        import_Paper.default,
        {
          elevation: 0,
          sx: {
            fontSize: "14px",
            width: "100%",
            height: 64,
            display: "flex",
            flexShrink: 0,
            alignItems: "center",
            justifyContent: siderCollapsed ? "center" : "space-between",
            paddingLeft: siderCollapsed ? 0 : "16px",
            paddingRight: siderCollapsed ? 0 : "8px",
            variant: "outlined",
            borderRadius: 0,
            borderBottom: (theme) => `1px solid ${theme.palette.action.focus}`
          }
        },
        /* @__PURE__ */ import_react4.default.createElement(RenderToTitle, { collapsed: siderCollapsed }),
        !siderCollapsed && /* @__PURE__ */ import_react4.default.createElement(import_IconButton.default, { size: "small", onClick: () => setSiderCollapsed(true) }, /* @__PURE__ */ import_react4.default.createElement(import_ChevronLeft.default, null))
      ),
      /* @__PURE__ */ import_react4.default.createElement(
        import_Box.default,
        {
          sx: {
            flexGrow: 1,
            overflowX: "hidden",
            overflowY: "auto"
          }
        },
        drawer
      )
    )
  ));
}, "ThemedSider");

// src/components/themedLayout/header/index.tsx
var import_react6 = __toESM(require("react"));
var import_core4 = require("@refinedev/core");
var import_AppBar = __toESM(require("@mui/material/AppBar"));
var import_Stack = __toESM(require("@mui/material/Stack"));
var import_Toolbar = __toESM(require("@mui/material/Toolbar"));
var import_Typography = __toESM(require("@mui/material/Typography"));
var import_Avatar = __toESM(require("@mui/material/Avatar"));

// src/components/themedLayout/hamburgerMenu/index.tsx
var import_react5 = __toESM(require("react"));
var import_IconButton2 = __toESM(require("@mui/material/IconButton"));
var import_Menu = __toESM(require("@mui/icons-material/Menu"));
var HamburgerIcon = /* @__PURE__ */ __name((props) => /* @__PURE__ */ import_react5.default.createElement(import_IconButton2.default, { color: "inherit", "aria-label": "open drawer", edge: "start", ...props }, /* @__PURE__ */ import_react5.default.createElement(import_Menu.default, null)), "HamburgerIcon");
var HamburgerMenu = /* @__PURE__ */ __name(() => {
  const {
    siderCollapsed,
    setSiderCollapsed,
    mobileSiderOpen,
    setMobileSiderOpen
  } = useThemedLayoutContext();
  return /* @__PURE__ */ import_react5.default.createElement(import_react5.default.Fragment, null, /* @__PURE__ */ import_react5.default.createElement(
    HamburgerIcon,
    {
      onClick: () => setSiderCollapsed(!siderCollapsed),
      sx: {
        mr: 2,
        display: { xs: "none", md: "flex" },
        ...!siderCollapsed && { display: "none" }
      }
    }
  ), /* @__PURE__ */ import_react5.default.createElement(
    HamburgerIcon,
    {
      onClick: () => setMobileSiderOpen(!mobileSiderOpen),
      sx: {
        mr: 2,
        display: { xs: "flex", md: "none" },
        ...mobileSiderOpen && { display: "none" }
      }
    }
  ));
}, "HamburgerMenu");

// src/components/themedLayout/header/index.tsx
var ThemedHeader = /* @__PURE__ */ __name(({
  sticky
}) => {
  const { data: user } = (0, import_core4.useGetIdentity)();
  const prefferedSticky = sticky ?? true;
  return /* @__PURE__ */ import_react6.default.createElement(import_AppBar.default, { position: prefferedSticky ? "sticky" : "relative" }, /* @__PURE__ */ import_react6.default.createElement(import_Toolbar.default, null, /* @__PURE__ */ import_react6.default.createElement(HamburgerMenu, null), /* @__PURE__ */ import_react6.default.createElement(
    import_Stack.default,
    {
      direction: "row",
      width: "100%",
      justifyContent: "flex-end",
      alignItems: "center"
    },
    /* @__PURE__ */ import_react6.default.createElement(
      import_Stack.default,
      {
        direction: "row",
        gap: "16px",
        alignItems: "center",
        justifyContent: "center"
      },
      (user == null ? void 0 : user.name) && /* @__PURE__ */ import_react6.default.createElement(import_Typography.default, { variant: "subtitle2", }, user == null ? void 0 : user.name),
      (user == null ? void 0 : user.avatar) && /* @__PURE__ */ import_react6.default.createElement(import_Avatar.default, { src: user == null ? void 0 : user.avatar, alt: user == null ? void 0 : user.name })
    )
  )));
}, "ThemedHeader");

// src/components/themedLayout/index.tsx
var ThemedLayout = /* @__PURE__ */ __name(({
  Sider,
  Header,
  Title,
  Footer,
  OffLayoutArea,
  children,
  initialSiderCollapsed,
  onSiderCollapsed,
  childrenBoxProps = {},
  containerBoxProps = {}
}) => {
  const SiderToRender = Sider ?? ThemedSider;
  const HeaderToRender = Header ?? ThemedHeader;
  const { sx: childrenSx, ...restChildrenProps } = childrenBoxProps;
  const { sx: containerSx, ...restContainerProps } = containerBoxProps;
  return /* @__PURE__ */ import_react7.default.createElement(
    ThemedLayoutContextProvider,
    {
      initialSiderCollapsed,
      onSiderCollapsed
    },
    /* @__PURE__ */ import_react7.default.createElement(
      import_Box2.default,
      {
        sx: { display: "flex", flexDirection: "row", ...containerSx },
        ...restContainerProps
      },
      /* @__PURE__ */ import_react7.default.createElement(SiderToRender, { Title }),
      /* @__PURE__ */ import_react7.default.createElement(
        import_Box2.default,
        {
          sx: [
            {
              display: "flex",
              flexDirection: "column",
              flex: 1,
              minWidth: "1px",
              minHeight: "1px"
            }
          ]
        },
        /* @__PURE__ */ import_react7.default.createElement(HeaderToRender, null),
        /* @__PURE__ */ import_react7.default.createElement(
          import_Box2.default,
          {
            component: "main",
            sx: {
              p: { xs: 1, md: 2, lg: 3 },
              flexGrow: 1,
              bgcolor: (theme) => theme.palette.background.default,
              ...childrenSx
            },
            ...restChildrenProps
          },
          children
        ),
        Footer && /* @__PURE__ */ import_react7.default.createElement(Footer, null)
      ),
      OffLayoutArea && /* @__PURE__ */ import_react7.default.createElement(OffLayoutArea, null)
    )
  );
}, "ThemedLayout");

// src/components/themedLayout/title/index.tsx
var import_react8 = __toESM(require("react"));
var import_core5 = require("@refinedev/core");
var import_Link = __toESM(require("@mui/material/Link"));
var import_SvgIcon = __toESM(require("@mui/material/SvgIcon"));
var import_Typography2 = __toESM(require("@mui/material/Typography"));
var ThemedTitle = /* @__PURE__ */ __name(({
  collapsed,
  wrapperStyles,
  icon: iconFromProps,
  text: textFromProps
}) => {
  const {
    title: { icon: defaultIcon, text: defaultText } = {}
  } = (0, import_core5.useRefineOptions)();
  const icon = typeof iconFromProps === "undefined" ? defaultIcon : iconFromProps;
  const text = typeof textFromProps === "undefined" ? defaultText : textFromProps;
  const Link3 = (0, import_core5.useLink)();
  return /* @__PURE__ */ import_react8.default.createElement(Link3, { to: "/", style: { textDecoration: "none" } }, /* @__PURE__ */ import_react8.default.createElement(
    import_Link.default,
    {
      underline: "none",
      sx: {
        display: "flex",
        alignItems: "center",
        gap: "12px",
        ...wrapperStyles
      }
    },
    /* @__PURE__ */ import_react8.default.createElement(import_SvgIcon.default, { height: "24px", width: "24px", color: "primary" }, icon),
    !collapsed && /* @__PURE__ */ import_react8.default.createElement(
      import_Typography2.default,
      {
        variant: "h6",
        fontWeight: 700,
        color: "text.primary",
        fontSize: "inherit",
        textOverflow: "ellipsis",
        overflow: "hidden"
      },
      text
    )
  ));
}, "ThemedTitle");

// src/components/pages/error/index.tsx
var import_react9 = __toESM(require("react"));
var import_core6 = require("@refinedev/core");
var import_Stack2 = __toESM(require("@mui/material/Stack"));
var import_Button = __toESM(require("@mui/material/Button"));
var import_Tooltip2 = __toESM(require("@mui/material/Tooltip"));
var import_Typography3 = __toESM(require("@mui/material/Typography"));
var import_Grid2 = __toESM(require("@mui/material/Grid2"));
var import_Info = __toESM(require("@mui/icons-material/Info"));
var ErrorComponent = /* @__PURE__ */ __name(() => {
  const [errorMessage, setErrorMessage] = (0, import_react9.useState)();
  const go = (0, import_core6.useGo)();
  const { resource, action } = (0, import_core6.useResourceParams)();
  const translate = (0, import_core6.useTranslate)();
  (0, import_react9.useEffect)(() => {
    if (resource && action) {
      setErrorMessage(
        translate(
          "pages.error.info",
          {
            action,
            resource: resource == null ? void 0 : resource.name
          },
          `You may have forgotten to add the "${action}" component to "${resource == null ? void 0 : resource.name}" resource.`
        )
      );
    }
  }, [action, resource]);
  return /* @__PURE__ */ import_react9.default.createElement(import_Grid2.default, { display: "flex", justifyContent: "center", alignItems: "center", mt: 20 }, /* @__PURE__ */ import_react9.default.createElement(import_Grid2.default, { container: true, direction: "column", display: "flex", alignItems: "center" }, /* @__PURE__ */ import_react9.default.createElement(import_Typography3.default, { variant: "h1" }, "404"), /* @__PURE__ */ import_react9.default.createElement(import_Stack2.default, { direction: "row", spacing: "2" }, /* @__PURE__ */ import_react9.default.createElement(import_Typography3.default, null, translate(
    "pages.error.404",
    "Sorry, the page you visited does not exist."
  )), errorMessage && /* @__PURE__ */ import_react9.default.createElement(import_Tooltip2.default, { title: errorMessage }, /* @__PURE__ */ import_react9.default.createElement(import_Info.default, { }))), /* @__PURE__ */ import_react9.default.createElement(
    import_Button.default,
    {
      onClick: () => {
        go({ to: "/" });
      }
    },
    translate("pages.error.backHome", "Back Home")
  )));
}, "ErrorComponent");

// src/components/pages/auth/index.tsx
var import_react10 = __toESM(require("react"));

// src/components/pages/auth/components/login/index.tsx
var React8 = __toESM(require("react"));
var import_react_hook_form = require("@refinedev/react-hook-form");
var import_react_hook_form2 = require("react-hook-form");
var import_Button2 = __toESM(require("@mui/material/Button"));
var import_TextField = __toESM(require("@mui/material/TextField"));
var import_Box3 = __toESM(require("@mui/material/Box"));
var import_Typography4 = __toESM(require("@mui/material/Typography"));
var import_Container = __toESM(require("@mui/material/Container"));
var import_Card = __toESM(require("@mui/material/Card"));
var import_CardContent = __toESM(require("@mui/material/CardContent"));
var import_Checkbox = __toESM(require("@mui/material/Checkbox"));
var import_FormControlLabel = __toESM(require("@mui/material/FormControlLabel"));
var import_Divider = __toESM(require("@mui/material/Divider"));
var import_Link2 = __toESM(require("@mui/material/Link"));
var import_Stack3 = __toESM(require("@mui/material/Stack"));
var import_core7 = require("@refinedev/core");

// src/components/pages/auth/components/styles.ts
var layoutStyles = {};
var titleStyles = {
  textAlign: "center",
  fontSize: "24px",
  marginBottom: "24px",
  overflowWrap: "break-word",
  hyphens: "manual",
  textOverflow: "unset",
  whiteSpace: "pre-wrap"
};

// src/components/pages/auth/components/login/index.tsx
var LoginPage = /* @__PURE__ */ __name(({
  providers,
  registerLink,
  forgotPasswordLink,
  rememberMe,
  contentProps,
  wrapperProps,
  renderContent,
  formProps,
  title,
  hideForm,
  mutationVariables
}) => {
  var _a;
  const { onSubmit, ...useFormProps } = formProps || {};
  const methods = (0, import_react_hook_form.useForm)({
    ...useFormProps
  });
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = methods;
  const { mutate: login, isPending } = (0, import_core7.useLogin)();
  const translate = (0, import_core7.useTranslate)();
  const Link3 = (0, import_core7.useLink)();
  const PageTitle = title === false ? null : /* @__PURE__ */ React8.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React8.createElement(
      ThemedTitle,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ React8.createElement(React8.Fragment, null, /* @__PURE__ */ React8.createElement(import_Stack3.default, { spacing: 1 }, providers.map((provider) => {
        return /* @__PURE__ */ React8.createElement(
          import_Button2.default,
          {
            key: provider.name,
            variant: "outlined",
            fullWidth: true,
            sx: {
              color: "primary.light",
              borderColor: "primary.light",
              textTransform: "none"
            },
            onClick: () => login({ ...mutationVariables, providerName: provider.name }),
            startIcon: provider.icon
          },
          provider.label
        );
      })), !hideForm && /* @__PURE__ */ React8.createElement(
        import_Divider.default,
        {
          sx: {
            fontSize: "12px",
            marginY: "16px"
          }
        },
        translate("pages.login.divider", "or")
      ));
    }
    return null;
  }, "renderProviders");
  const Content = /* @__PURE__ */ React8.createElement(import_Card.default, { ...contentProps ?? {} }, /* @__PURE__ */ React8.createElement(import_CardContent.default, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React8.createElement(
    import_Typography4.default,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.login.title", "Sign in to your account")
  ), renderProviders(), !hideForm && /* @__PURE__ */ React8.createElement(
    import_Box3.default,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return login({ ...mutationVariables, ...data });
      })
    },
    /* @__PURE__ */ React8.createElement(
      import_TextField.default,
      {
        ...register("email", {
          required: translate(
            "pages.login.errors.requiredEmail",
            "Email is required"
          )
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.login.fields.email", "Email"),
        error: !!errors.email,
        name: "email",
        type: "email",
        autoComplete: "email",
        sx: {
          mt: 0
        }
      }
    ),
    /* @__PURE__ */ React8.createElement(
      import_TextField.default,
      {
        ...register("password", {
          required: translate(
            "pages.login.errors.requiredPassword",
            "Password is required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate("pages.login.fields.password", "Password"),
        helperText: (_a = errors == null ? void 0 : errors.password) == null ? void 0 : _a.message,
        error: !!errors.password,
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React8.createElement(
      import_Box3.default,
      {
        component: "div",
        sx: {
          mt: "24px",
          display: "flex",
          justifyContent: "space-between",
          alignItems: "center"
        }
      },
      rememberMe ?? /* @__PURE__ */ React8.createElement(
        import_FormControlLabel.default,
        {
          sx: {
            span: {
              fontSize: "14px",
              color: "text.secondary"
            }
          },
          color: "secondary",
          control: /* @__PURE__ */ React8.createElement(
            import_Checkbox.default,
            {
              size: "small",
              id: "remember",
              ...register("remember")
            }
          ),
          label: translate(
            "pages.login.buttons.rememberMe",
            "Remember me"
          )
        }
      ),
      forgotPasswordLink ?? /* @__PURE__ */ React8.createElement(
        import_Link2.default,
        {
          variant: "body2",
          color: "primary",
          fontSize: "12px",
          component: Link3,
          underline: "none",
          to: "/forgot-password"
        },
        translate(
          "pages.login.buttons.forgotPassword",
          "Forgot password?"
        )
      )
    ),
    /* @__PURE__ */ React8.createElement(
      import_Button2.default,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        disabled: isPending,
        sx: { mt: "24px" }
      },
      translate("pages.login.signin", "Sign in")
    )
  ), registerLink ?? /* @__PURE__ */ React8.createElement(
    import_Box3.default,
    {
      sx: {
        mt: "24px",
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
      }
    },
    /* @__PURE__ */ React8.createElement(
      import_Typography4.default,
      {
        textAlign: "center",
        variant: "body2",
        component: "span",
        fontSize: "12px"
      },
      translate(
        "pages.login.buttons.noAccount",
        "Don\u2019t have an account?"
      )
    ),
    /* @__PURE__ */ React8.createElement(
      import_Link2.default,
      {
        ml: "4px",
        fontSize: "12px",
        variant: "body2",
        color: "primary",
        component: Link3,
        underline: "none",
        to: "/register",
        fontWeight: "bold"
      },
      translate("pages.login.signup", "Sign up")
    )
  )));
  return /* @__PURE__ */ React8.createElement(import_react_hook_form2.FormProvider, { ...methods }, /* @__PURE__ */ React8.createElement(import_Box3.default, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React8.createElement(
    import_Container.default,
    {
      component: "main",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: hideForm ? "flex-start" : "center",
        alignItems: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    /* @__PURE__ */ React8.createElement(
      import_Box3.default,
      {
        sx: {
          width: "100%",
          maxWidth: "400px",
          display: "flex",
          flexDirection: "column",
          paddingTop: hideForm ? "15dvh" : 0
        }
      },
      renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React8.createElement(React8.Fragment, null, PageTitle, Content)
    )
  )));
}, "LoginPage");

// src/components/pages/auth/components/register/index.tsx
var React9 = __toESM(require("react"));
var import_react_hook_form3 = require("@refinedev/react-hook-form");
var import_Button3 = __toESM(require("@mui/material/Button"));
var import_TextField2 = __toESM(require("@mui/material/TextField"));
var import_Box4 = __toESM(require("@mui/material/Box"));
var import_Typography5 = __toESM(require("@mui/material/Typography"));
var import_Container2 = __toESM(require("@mui/material/Container"));
var import_Card2 = __toESM(require("@mui/material/Card"));
var import_CardContent2 = __toESM(require("@mui/material/CardContent"));
var import_Divider2 = __toESM(require("@mui/material/Divider"));
var import_Stack4 = __toESM(require("@mui/material/Stack"));
var import_Link3 = __toESM(require("@mui/material/Link"));
var import_core8 = require("@refinedev/core");
var RegisterPage = /* @__PURE__ */ __name(({
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  providers,
  formProps,
  title,
  hideForm,
  mutationVariables
}) => {
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = (0, import_react_hook_form3.useForm)({
    ...useFormProps
  });
  const { mutate: registerMutate, isPending } = (0, import_core8.useRegister)();
  const translate = (0, import_core8.useTranslate)();
  const Link3 = (0, import_core8.useLink)();
  const PageTitle = title === false ? null : /* @__PURE__ */ React9.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React9.createElement(
      ThemedTitle,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const renderProviders = /* @__PURE__ */ __name(() => {
    if (providers && providers.length > 0) {
      return /* @__PURE__ */ React9.createElement(React9.Fragment, null, /* @__PURE__ */ React9.createElement(import_Stack4.default, { spacing: 1 }, providers.map((provider) => {
        return /* @__PURE__ */ React9.createElement(
          import_Button3.default,
          {
            key: provider.name,
            color: "secondary",
            fullWidth: true,
            variant: "outlined",
            sx: {
              color: "primary.light",
              borderColor: "primary.light",
              textTransform: "none"
            },
            onClick: () => registerMutate({
              ...mutationVariables,
              providerName: provider.name
            }),
            startIcon: provider.icon
          },
          provider.label
        );
      })), !hideForm && /* @__PURE__ */ React9.createElement(
        import_Divider2.default,
        {
          sx: {
            fontSize: "12px",
            marginY: "16px"
          }
        },
        translate(
          "pages.register.divider",
          translate("pages.login.divider", "or")
        )
      ));
    }
    return null;
  }, "renderProviders");
  const Content = /* @__PURE__ */ React9.createElement(import_Card2.default, { ...contentProps ?? {} }, /* @__PURE__ */ React9.createElement(import_CardContent2.default, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React9.createElement(
    import_Typography5.default,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.register.title", "Sign up for your account")
  ), renderProviders(), !hideForm && /* @__PURE__ */ React9.createElement(
    import_Box4.default,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return registerMutate({ ...mutationVariables, ...data });
      })
    },
    /* @__PURE__ */ React9.createElement(
      import_TextField2.default,
      {
        ...register("email", {
          required: translate(
            "pages.register.errors.requiredEmail",
            "Email is required"
          ),
          pattern: {
            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
            message: translate(
              "pages.register.errors.validEmail",
              "Invalid email address"
            )
          }
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.register.email", "Email"),
        error: !!errors.email,
        helperText: errors["email"] ? errors["email"].message : "",
        name: "email",
        autoComplete: "email",
        sx: {
          mt: 0
        }
      }
    ),
    /* @__PURE__ */ React9.createElement(
      import_TextField2.default,
      {
        ...register("password", {
          required: translate(
            "pages.register.errors.requiredPassword",
            "Password is required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate("pages.register.fields.password", "Password"),
        helperText: errors["password"] ? errors["password"].message : "",
        error: !!errors.password,
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React9.createElement(
      import_Button3.default,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        disabled: isPending,
        sx: {
          mt: "24px"
        }
      },
      translate("pages.register.signup", "Sign up")
    )
  ), loginLink ?? /* @__PURE__ */ React9.createElement(
    import_Box4.default,
    {
      display: "flex",
      justifyContent: "flex-end",
      alignItems: "center",
      sx: {
        mt: "24px",
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
      }
    },
    /* @__PURE__ */ React9.createElement(import_Typography5.default, { variant: "body2", component: "span", fontSize: "12px" }, translate(
      "pages.register.buttons.haveAccount",
      translate(
        "pages.login.buttons.haveAccount",
        "Have an account?"
      )
    )),
    /* @__PURE__ */ React9.createElement(
      import_Link3.default,
      {
        ml: "4px",
        variant: "body2",
        color: "primary",
        component: Link3,
        underline: "none",
        to: "/login",
        fontSize: "12px",
        fontWeight: "bold"
      },
      translate(
        "pages.register.signin",
        translate("pages.login.signin", "Sign in")
      )
    )
  )));
  return /* @__PURE__ */ React9.createElement(import_Box4.default, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React9.createElement(
    import_Container2.default,
    {
      component: "main",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: hideForm ? "flex-start" : "center",
        alignItems: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    /* @__PURE__ */ React9.createElement(
      import_Box4.default,
      {
        sx: {
          width: "100%",
          maxWidth: "400px",
          display: "flex",
          flexDirection: "column",
          paddingTop: hideForm ? "15dvh" : 0
        }
      },
      renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React9.createElement(React9.Fragment, null, PageTitle, Content)
    )
  ));
}, "RegisterPage");

// src/components/pages/auth/components/forgotPassword/index.tsx
var import_Button4 = __toESM(require("@mui/material/Button"));
var import_TextField3 = __toESM(require("@mui/material/TextField"));
var import_Box5 = __toESM(require("@mui/material/Box"));
var import_Typography6 = __toESM(require("@mui/material/Typography"));
var import_Container3 = __toESM(require("@mui/material/Container"));
var import_Card3 = __toESM(require("@mui/material/Card"));
var import_CardContent3 = __toESM(require("@mui/material/CardContent"));
var import_Link4 = __toESM(require("@mui/material/Link"));
var import_react_hook_form4 = require("@refinedev/react-hook-form");
var React10 = __toESM(require("react"));
var import_core9 = require("@refinedev/core");
var ForgotPasswordPage = /* @__PURE__ */ __name(({
  loginLink,
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title,
  mutationVariables
}) => {
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    handleSubmit,
    formState: { errors }
  } = (0, import_react_hook_form4.useForm)({
    ...useFormProps
  });
  const { mutate, isPending } = (0, import_core9.useForgotPassword)();
  const translate = (0, import_core9.useTranslate)();
  const Link3 = (0, import_core9.useLink)();
  const PageTitle = title === false ? null : /* @__PURE__ */ React10.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React10.createElement(
      ThemedTitle,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const Content = /* @__PURE__ */ React10.createElement(import_Card3.default, { ...contentProps ?? {} }, /* @__PURE__ */ React10.createElement(import_CardContent3.default, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React10.createElement(
    import_Typography6.default,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.forgotPassword.title", "Forgot your password?")
  ), /* @__PURE__ */ React10.createElement(
    import_Box5.default,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return mutate({ ...mutationVariables, ...data });
      })
    },
    /* @__PURE__ */ React10.createElement(
      import_TextField3.default,
      {
        ...register("email", {
          required: translate(
            "pages.forgotPassword.errors.requiredEmail",
            "Email is required"
          ),
          pattern: {
            value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i,
            message: translate(
              "pages.register.errors.validEmail",
              "Invalid email address"
            )
          }
        }),
        id: "email",
        margin: "normal",
        fullWidth: true,
        label: translate("pages.forgotPassword.fields.email", "Email"),
        name: "email",
        type: "email",
        error: !!errors.email,
        autoComplete: "email",
        sx: {
          m: 0
        }
      }
    ),
    loginLink ?? /* @__PURE__ */ React10.createElement(import_Box5.default, { textAlign: "right", sx: { mt: "24px" } }, /* @__PURE__ */ React10.createElement(import_Typography6.default, { variant: "body2", component: "span", fontSize: "12px" }, translate(
      "pages.forgotPassword.buttons.haveAccount",
      translate(
        "pages.register.buttons.haveAccount",
        "Have an account? "
      )
    )), " ", /* @__PURE__ */ React10.createElement(
      import_Link4.default,
      {
        variant: "body2",
        component: Link3,
        underline: "none",
        to: "/login",
        fontWeight: "bold",
        fontSize: "12px",
        color: "primary.light"
      },
      translate(
        "pages.forgotPassword.signin",
        translate("pages.login.signin", "Sign in")
      )
    )),
    /* @__PURE__ */ React10.createElement(
      import_Button4.default,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        sx: { mt: "24px" },
        disabled: isPending
      },
      translate(
        "pages.forgotPassword.buttons.submit",
        "Send reset instructions"
      )
    )
  )));
  return /* @__PURE__ */ React10.createElement(React10.Fragment, null, /* @__PURE__ */ React10.createElement(import_Box5.default, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React10.createElement(
    import_Container3.default,
    {
      component: "main",
      maxWidth: "xs",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React10.createElement(React10.Fragment, null, PageTitle, Content)
  )));
}, "ForgotPasswordPage");

// src/components/pages/auth/components/updatePassword/index.tsx
var React11 = __toESM(require("react"));
var import_react_hook_form5 = require("@refinedev/react-hook-form");
var import_Button5 = __toESM(require("@mui/material/Button"));
var import_TextField4 = __toESM(require("@mui/material/TextField"));
var import_Box6 = __toESM(require("@mui/material/Box"));
var import_Typography7 = __toESM(require("@mui/material/Typography"));
var import_Container4 = __toESM(require("@mui/material/Container"));
var import_Card4 = __toESM(require("@mui/material/Card"));
var import_CardContent4 = __toESM(require("@mui/material/CardContent"));
var import_core10 = require("@refinedev/core");
var UpdatePasswordPage = /* @__PURE__ */ __name(({
  wrapperProps,
  contentProps,
  renderContent,
  formProps,
  title = void 0,
  mutationVariables
}) => {
  var _a, _b;
  const { onSubmit, ...useFormProps } = formProps || {};
  const {
    register,
    watch,
    handleSubmit,
    formState: { errors }
  } = (0, import_react_hook_form5.useForm)({
    ...useFormProps
  });
  const { mutate: update, isPending } = (0, import_core10.useUpdatePassword)();
  const translate = (0, import_core10.useTranslate)();
  const PageTitle = title === false ? null : /* @__PURE__ */ React11.createElement(
    "div",
    {
      style: {
        display: "flex",
        justifyContent: "center",
        marginBottom: "32px",
        fontSize: "20px"
      }
    },
    title ?? /* @__PURE__ */ React11.createElement(
      ThemedTitle,
      {
        collapsed: false,
        wrapperStyles: {
          gap: "8px"
        }
      }
    )
  );
  const Content = /* @__PURE__ */ React11.createElement(import_Card4.default, { ...contentProps ?? {} }, /* @__PURE__ */ React11.createElement(import_CardContent4.default, { sx: { p: "32px", "&:last-child": { pb: "32px" } } }, /* @__PURE__ */ React11.createElement(
    import_Typography7.default,
    {
      component: "h1",
      variant: "h5",
      align: "center",
      style: titleStyles,
      color: "primary",
      fontWeight: 700
    },
    translate("pages.updatePassword.title", "Set New Password")
  ), /* @__PURE__ */ React11.createElement(
    import_Box6.default,
    {
      component: "form",
      onSubmit: handleSubmit((data) => {
        if (onSubmit) {
          return onSubmit(data);
        }
        return update({ ...mutationVariables, ...data });
      })
    },
    /* @__PURE__ */ React11.createElement(
      import_TextField4.default,
      {
        ...register("password", {
          required: translate(
            "pages.updatePassword.errors.requiredPassword",
            "Password required"
          )
        }),
        id: "password",
        margin: "normal",
        fullWidth: true,
        name: "password",
        label: translate(
          "pages.updatePassword.fields.password",
          "New Password"
        ),
        helperText: (_a = errors == null ? void 0 : errors.password) == null ? void 0 : _a.message,
        error: !!(errors == null ? void 0 : errors.password),
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-password",
        sx: {
          m: 0
        }
      }
    ),
    /* @__PURE__ */ React11.createElement(
      import_TextField4.default,
      {
        ...register("confirmPassword", {
          required: translate(
            "pages.updatePassword.errors.requiredConfirmPassword",
            "Confirm Password is required"
          ),
          validate: (value) => {
            if (watch("password") !== value) {
              return translate(
                "pages.updatePassword.errors.confirmPasswordNotMatch",
                "Passwords do not match"
              );
            }
            return true;
          }
        }),
        id: "confirmPassword",
        margin: "normal",
        fullWidth: true,
        name: "confirmPassword",
        label: translate(
          "pages.updatePassword.fields.confirmPassword",
          "Confirm New Password"
        ),
        helperText: (_b = errors == null ? void 0 : errors.confirmPassword) == null ? void 0 : _b.message,
        error: !!(errors == null ? void 0 : errors.confirmPassword),
        type: "password",
        placeholder: "\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF\u25CF",
        autoComplete: "current-confirm-password",
        sx: {
          mb: 0
        }
      }
    ),
    /* @__PURE__ */ React11.createElement(
      import_Button5.default,
      {
        type: "submit",
        fullWidth: true,
        variant: "contained",
        sx: {
          mt: "24px"
        },
        disabled: isPending
      },
      translate("pages.updatePassword.buttons.submit", "Update")
    )
  )));
  return /* @__PURE__ */ React11.createElement(React11.Fragment, null, /* @__PURE__ */ React11.createElement(import_Box6.default, { component: "div", style: layoutStyles, ...wrapperProps ?? {} }, /* @__PURE__ */ React11.createElement(
    import_Container4.default,
    {
      component: "main",
      maxWidth: "xs",
      sx: {
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        minHeight: "100dvh",
        padding: "16px",
        width: "100%",
        maxWidth: "400px"
      }
    },
    renderContent ? renderContent(Content, PageTitle) : /* @__PURE__ */ React11.createElement(React11.Fragment, null, PageTitle, Content)
  )));
}, "UpdatePasswordPage");

// src/components/pages/auth/index.tsx
var AuthPage = /* @__PURE__ */ __name((props) => {
  const { type } = props;
  const renderView = /* @__PURE__ */ __name(() => {
    switch (type) {
      case "register":
        return /* @__PURE__ */ import_react10.default.createElement(RegisterPage, { ...props });
      case "forgotPassword":
        return /* @__PURE__ */ import_react10.default.createElement(ForgotPasswordPage, { ...props });
      case "updatePassword":
        return /* @__PURE__ */ import_react10.default.createElement(UpdatePasswordPage, { ...props });
      default:
        return /* @__PURE__ */ import_react10.default.createElement(LoginPage, { ...props });
    }
  }, "renderView");
  return /* @__PURE__ */ import_react10.default.createElement(import_react10.default.Fragment, null, renderView());
}, "AuthPage");

// src/components/crud/create/index.tsx
var import_react11 = __toESM(require("react"));
var import_core11 = require("@refinedev/core");
var import_Card5 = __toESM(require("@mui/material/Card"));
var import_CardHeader = __toESM(require("@mui/material/CardHeader"));
var import_CardActions = __toESM(require("@mui/material/CardActions"));
var import_CardContent5 = __toESM(require("@mui/material/CardContent"));
var import_IconButton3 = __toESM(require("@mui/material/IconButton"));
var import_Typography8 = __toESM(require("@mui/material/Typography"));
var import_Box7 = __toESM(require("@mui/material/Box"));
var import_CircularProgress = __toESM(require("@mui/material/CircularProgress"));
var import_system = require("@mui/system");
var import_ArrowBack = __toESM(require("@mui/icons-material/ArrowBack"));
var import_ui_types = require("@refinedev/ui-types");
var Create = /* @__PURE__ */ __name(({
  title,
  children,
  saveButtonProps: saveButtonPropsFromProps,
  resource: resourceFromProps,
  isLoading = false,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps
}) => {
  var _a;
  const translate = (0, import_core11.useTranslate)();
  const back = (0, import_core11.useBack)();
  const getUserFriendlyName = (0, import_core11.useUserFriendlyName)();
  const { resource, action, identifier } = (0, import_core11.useResourceParams)({
    resource: resourceFromProps
  });
  const breadcrumb = breadcrumbFromProps;
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react11.default.createElement(import_react11.default.Fragment, null, breadcrumb) : /* @__PURE__ */ import_react11.default.createElement(Breadcrumb, null);
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps
  };
  const defaultFooterButtons = /* @__PURE__ */ import_react11.default.createElement(SaveButton, { ...saveButtonProps });
  return /* @__PURE__ */ import_react11.default.createElement(
    import_Card5.default,
    {
      ...wrapperProps ?? {},
      sx: {
        position: "relative",
        ...wrapperProps == null ? void 0 : wrapperProps.sx
      }
    },
    isLoading && /* @__PURE__ */ import_react11.default.createElement(
      import_Box7.default,
      {
        sx: {
          position: "absolute",
          inset: 0,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          zIndex: (theme) => theme.zIndex.drawer + 1,
          // this is needed to support custom themes, dark mode etc.
          bgcolor: (theme) => (0, import_system.alpha)(theme.palette.background.paper, 0.4)
        }
      },
      /* @__PURE__ */ import_react11.default.createElement(import_CircularProgress.default, null)
    ),
    breadcrumbComponent,
    /* @__PURE__ */ import_react11.default.createElement(
      import_CardHeader.default,
      {
        sx: {
          display: "flex",
          flexWrap: "wrap",
          ".MuiCardHeader-action": {
            margin: 0,
            alignSelf: "center"
          }
        },
        title: title ?? /* @__PURE__ */ import_react11.default.createElement(
          import_Typography8.default,
          {
            variant: "h5",
            className: import_ui_types.RefinePageHeaderClassNames.Title
          },
          translate(
            `${identifier}.titles.create`,
            `Create ${getUserFriendlyName(
              ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier,
              "singular"
            )}`
          )
        ),
        avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ import_react11.default.createElement(
          import_IconButton3.default,
          {
            onClick: action !== "list" || typeof action !== "undefined" ? back : void 0
          },
          /* @__PURE__ */ import_react11.default.createElement(import_ArrowBack.default, null)
        ),
        action: headerButtons ? /* @__PURE__ */ import_react11.default.createElement(import_Box7.default, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
          defaultButtons: null
        }) : headerButtons : null) : void 0,
        ...headerProps ?? {}
      }
    ),
    /* @__PURE__ */ import_react11.default.createElement(import_CardContent5.default, { ...contentProps ?? {} }, children),
    /* @__PURE__ */ import_react11.default.createElement(
      import_CardActions.default,
      {
        sx: {
          display: "flex",
          justifyContent: "flex-end",
          gap: "16px",
          padding: "16px"
        },
        ...footerButtonProps ?? {}
      },
      footerButtons ? typeof footerButtons === "function" ? footerButtons({
        defaultButtons: defaultFooterButtons,
        saveButtonProps
      }) : footerButtons : defaultFooterButtons
    )
  );
}, "Create");

// src/components/crud/edit/index.tsx
var import_react12 = __toESM(require("react"));
var import_core12 = require("@refinedev/core");
var import_Card6 = __toESM(require("@mui/material/Card"));
var import_CardHeader2 = __toESM(require("@mui/material/CardHeader"));
var import_IconButton4 = __toESM(require("@mui/material/IconButton"));
var import_CardContent6 = __toESM(require("@mui/material/CardContent"));
var import_CardActions2 = __toESM(require("@mui/material/CardActions"));
var import_Typography9 = __toESM(require("@mui/material/Typography"));
var import_Box8 = __toESM(require("@mui/material/Box"));
var import_CircularProgress2 = __toESM(require("@mui/material/CircularProgress"));
var import_system2 = require("@mui/system");
var import_ArrowBack2 = __toESM(require("@mui/icons-material/ArrowBack"));
var import_ui_types2 = require("@refinedev/ui-types");
var Edit = /* @__PURE__ */ __name(({
  title,
  saveButtonProps: saveButtonPropsFromProps,
  mutationMode: mutationModeProp,
  recordItemId,
  children,
  deleteButtonProps: deleteButtonPropsFromProps,
  canDelete,
  resource: resourceFromProps,
  isLoading = false,
  breadcrumb: breadcrumbFromProps,
  dataProviderName,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps,
  autoSaveProps
}) => {
  var _a, _b;
  const translate = (0, import_core12.useTranslate)();
  const { mutationMode: mutationModeContext } = (0, import_core12.useMutationMode)();
  const mutationMode = mutationModeProp ?? mutationModeContext;
  const back = (0, import_core12.useBack)();
  const go = (0, import_core12.useGo)();
  const getUserFriendlyName = (0, import_core12.useUserFriendlyName)();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = (0, import_core12.useResourceParams)({ resource: resourceFromProps, id: recordItemId });
  const goListPath = (0, import_core12.useToPath)({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const isDeleteButtonVisible = canDelete ?? (((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) || deleteButtonPropsFromProps);
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react12.default.createElement(import_react12.default.Fragment, null, breadcrumb) : /* @__PURE__ */ import_react12.default.createElement(Breadcrumb, null);
  const listButtonProps = hasList ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    recordItemId: id,
    dataProviderName
  };
  const defaultHeaderButtons = /* @__PURE__ */ import_react12.default.createElement(import_Box8.default, { display: "flex", flexDirection: "row", alignItems: "center" }, autoSaveProps && /* @__PURE__ */ import_react12.default.createElement(AutoSaveIndicator, { ...autoSaveProps }), hasList && /* @__PURE__ */ import_react12.default.createElement(ListButton, { ...listButtonProps }), /* @__PURE__ */ import_react12.default.createElement(RefreshButton, { ...refreshButtonProps }));
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier,
    mutationMode,
    variant: "outlined",
    onSuccess: () => {
      go({ to: goListPath });
    },
    recordItemId: id,
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const saveButtonProps = {
    ...isLoading ? { disabled: true } : {},
    ...saveButtonPropsFromProps
  };
  const defaultFooterButtons = /* @__PURE__ */ import_react12.default.createElement(import_react12.default.Fragment, null, isDeleteButtonVisible && /* @__PURE__ */ import_react12.default.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ import_react12.default.createElement(SaveButton, { ...saveButtonProps }));
  return /* @__PURE__ */ import_react12.default.createElement(
    import_Card6.default,
    {
      ...wrapperProps ?? {},
      sx: {
        position: "relative",
        ...wrapperProps == null ? void 0 : wrapperProps.sx
      }
    },
    isLoading && /* @__PURE__ */ import_react12.default.createElement(
      import_Box8.default,
      {
        sx: {
          position: "absolute",
          inset: 0,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          zIndex: (theme) => theme.zIndex.drawer + 1,
          // this is needed to support custom themes, dark mode etc.
          bgcolor: (theme) => (0, import_system2.alpha)(theme.palette.background.paper, 0.4)
        }
      },
      /* @__PURE__ */ import_react12.default.createElement(import_CircularProgress2.default, null)
    ),
    breadcrumbComponent,
    /* @__PURE__ */ import_react12.default.createElement(
      import_CardHeader2.default,
      {
        sx: {
          display: "flex",
          flexWrap: "wrap",
          ".MuiCardHeader-action": {
            margin: 0,
            alignSelf: "center"
          }
        },
        title: title ?? /* @__PURE__ */ import_react12.default.createElement(
          import_Typography9.default,
          {
            variant: "h5",
            className: import_ui_types2.RefinePageHeaderClassNames.Title
          },
          translate(
            `${identifier}.titles.edit`,
            `Edit ${getUserFriendlyName(
              ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.label) ?? identifier,
              "singular"
            )}`
          )
        ),
        avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ import_react12.default.createElement(
          import_IconButton4.default,
          {
            onClick: action !== "list" && typeof action !== "undefined" ? back : void 0
          },
          /* @__PURE__ */ import_react12.default.createElement(import_ArrowBack2.default, null)
        ),
        action: /* @__PURE__ */ import_react12.default.createElement(import_Box8.default, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
          defaultButtons: defaultHeaderButtons,
          listButtonProps,
          refreshButtonProps
        }) : headerButtons : defaultHeaderButtons),
        ...headerProps ?? {}
      }
    ),
    /* @__PURE__ */ import_react12.default.createElement(import_CardContent6.default, { ...contentProps ?? {} }, children),
    /* @__PURE__ */ import_react12.default.createElement(
      import_CardActions2.default,
      {
        sx: {
          display: "flex",
          justifyContent: "flex-end",
          gap: "16px",
          padding: "16px"
        },
        ...footerButtonProps ?? {}
      },
      footerButtons ? typeof footerButtons === "function" ? footerButtons({
        defaultButtons: defaultFooterButtons,
        deleteButtonProps,
        saveButtonProps
      }) : footerButtons : defaultFooterButtons
    )
  );
}, "Edit");

// src/components/crud/list/index.tsx
var import_react13 = __toESM(require("react"));
var import_core13 = require("@refinedev/core");
var import_Card7 = __toESM(require("@mui/material/Card"));
var import_CardHeader3 = __toESM(require("@mui/material/CardHeader"));
var import_CardContent7 = __toESM(require("@mui/material/CardContent"));
var import_Typography10 = __toESM(require("@mui/material/Typography"));
var import_Box9 = __toESM(require("@mui/material/Box"));
var import_ui_types3 = require("@refinedev/ui-types");
var List2 = /* @__PURE__ */ __name(({
  title,
  canCreate,
  children,
  createButtonProps: createButtonPropsFromProps,
  resource: resourceFromProps,
  breadcrumb: breadcrumbFromProps,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons
}) => {
  var _a;
  const translate = (0, import_core13.useTranslate)();
  const getUserFriendlyName = (0, import_core13.useUserFriendlyName)();
  const { resource, identifier } = (0, import_core13.useResourceParams)({
    resource: resourceFromProps
  });
  const isCreateButtonVisible = canCreate ?? (!!(resource == null ? void 0 : resource.create) || createButtonPropsFromProps);
  const breadcrumb = breadcrumbFromProps;
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react13.default.createElement(import_react13.default.Fragment, null, breadcrumb) : /* @__PURE__ */ import_react13.default.createElement(Breadcrumb, null);
  const createButtonProps = isCreateButtonVisible ? {
    resource: identifier,
    ...createButtonPropsFromProps
  } : void 0;
  const defaultHeaderButtons = isCreateButtonVisible ? /* @__PURE__ */ import_react13.default.createElement(CreateButton, { ...createButtonProps }) : null;
  return /* @__PURE__ */ import_react13.default.createElement(import_Card7.default, { ...wrapperProps ?? {} }, breadcrumbComponent, /* @__PURE__ */ import_react13.default.createElement(
    import_CardHeader3.default,
    {
      sx: {
        display: "flex",
        flexWrap: "wrap",
        ".MuiCardHeader-action": {
          margin: 0,
          alignSelf: "center"
        }
      },
      title: title ?? /* @__PURE__ */ import_react13.default.createElement(
        import_Typography10.default,
        {
          variant: "h5",
          className: import_ui_types3.RefinePageHeaderClassNames.Title
        },
        translate(
          `${identifier}.titles.list`,
          getUserFriendlyName(
            ((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.label) ?? identifier,
            "plural"
          )
        )
      ),
      action: /* @__PURE__ */ import_react13.default.createElement(import_Box9.default, { display: "flex", gap: "16px", ...headerButtonProps }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
        defaultButtons: defaultHeaderButtons,
        createButtonProps
      }) : headerButtons : defaultHeaderButtons),
      ...headerProps ?? {}
    }
  ), /* @__PURE__ */ import_react13.default.createElement(import_CardContent7.default, { ...contentProps ?? {} }, children));
}, "List");

// src/components/crud/show/index.tsx
var import_react14 = __toESM(require("react"));
var import_core14 = require("@refinedev/core");
var import_Card8 = __toESM(require("@mui/material/Card"));
var import_CardActions3 = __toESM(require("@mui/material/CardActions"));
var import_CardContent8 = __toESM(require("@mui/material/CardContent"));
var import_CardHeader4 = __toESM(require("@mui/material/CardHeader"));
var import_IconButton5 = __toESM(require("@mui/material/IconButton"));
var import_Box10 = __toESM(require("@mui/material/Box"));
var import_Typography11 = __toESM(require("@mui/material/Typography"));
var import_CircularProgress3 = __toESM(require("@mui/material/CircularProgress"));
var import_system3 = require("@mui/system");
var import_ArrowBack3 = __toESM(require("@mui/icons-material/ArrowBack"));
var import_ui_types4 = require("@refinedev/ui-types");
var Show = /* @__PURE__ */ __name(({
  title,
  canEdit,
  canDelete,
  deleteButtonProps: deleteButtonPropsFromProps,
  isLoading = false,
  children,
  resource: resourceFromProps,
  recordItemId,
  breadcrumb: breadcrumbFromProps,
  dataProviderName,
  wrapperProps,
  headerProps,
  contentProps,
  headerButtonProps,
  headerButtons,
  footerButtonProps,
  footerButtons,
  goBack: goBackFromProps
}) => {
  var _a, _b, _c;
  const translate = (0, import_core14.useTranslate)();
  const back = (0, import_core14.useBack)();
  const go = (0, import_core14.useGo)();
  const getUserFriendlyName = (0, import_core14.useUserFriendlyName)();
  const {
    resource,
    action,
    id: idFromParams,
    identifier
  } = (0, import_core14.useResourceParams)({ resource: resourceFromProps, id: recordItemId });
  const goListPath = (0, import_core14.useToPath)({
    resource,
    action: "list"
  });
  const id = recordItemId ?? idFromParams;
  const breadcrumb = breadcrumbFromProps;
  const hasList = (resource == null ? void 0 : resource.list) && !recordItemId;
  const hasDelete = canDelete ?? (((_a = resource == null ? void 0 : resource.meta) == null ? void 0 : _a.canDelete) || deleteButtonPropsFromProps);
  const isDeleteButtonVisible = hasDelete && typeof id !== "undefined";
  const isEditButtonVisible = canEdit ?? ((_b = resource == null ? void 0 : resource.meta) == null ? void 0 : _b.canEdit) ?? !!(resource == null ? void 0 : resource.edit);
  const breadcrumbComponent = typeof breadcrumb !== "undefined" ? /* @__PURE__ */ import_react14.default.createElement(import_react14.default.Fragment, null, breadcrumb) : /* @__PURE__ */ import_react14.default.createElement(Breadcrumb, null);
  const listButtonProps = hasList ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier ?? (resource == null ? void 0 : resource.name)
  } : void 0;
  const editButtonProps = isEditButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier ?? (resource == null ? void 0 : resource.name),
    recordItemId: id
  } : void 0;
  const deleteButtonProps = isDeleteButtonVisible ? {
    ...isLoading ? { disabled: true } : {},
    resource: identifier ?? (resource == null ? void 0 : resource.name),
    recordItemId: id,
    onSuccess: () => {
      go({ to: goListPath });
    },
    dataProviderName,
    ...deleteButtonPropsFromProps
  } : void 0;
  const refreshButtonProps = {
    ...isLoading ? { disabled: true } : {},
    resource: identifier ?? (resource == null ? void 0 : resource.name),
    recordItemId: id,
    dataProviderName
  };
  const defaultHeaderButtons = /* @__PURE__ */ import_react14.default.createElement(import_react14.default.Fragment, null, hasList && /* @__PURE__ */ import_react14.default.createElement(ListButton, { ...listButtonProps }), isEditButtonVisible && /* @__PURE__ */ import_react14.default.createElement(EditButton, { ...editButtonProps }), isDeleteButtonVisible && /* @__PURE__ */ import_react14.default.createElement(DeleteButton, { ...deleteButtonProps }), /* @__PURE__ */ import_react14.default.createElement(RefreshButton, { ...refreshButtonProps }));
  return /* @__PURE__ */ import_react14.default.createElement(
    import_Card8.default,
    {
      ...wrapperProps ?? {},
      sx: {
        position: "relative",
        ...wrapperProps == null ? void 0 : wrapperProps.sx
      }
    },
    isLoading && /* @__PURE__ */ import_react14.default.createElement(
      import_Box10.default,
      {
        sx: {
          position: "absolute",
          inset: 0,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
          zIndex: (theme) => theme.zIndex.drawer + 1,
          // this is needed to support custom themes, dark mode etc.
          bgcolor: (theme) => (0, import_system3.alpha)(theme.palette.background.paper, 0.4)
        }
      },
      /* @__PURE__ */ import_react14.default.createElement(import_CircularProgress3.default, null)
    ),
    breadcrumbComponent,
    /* @__PURE__ */ import_react14.default.createElement(
      import_CardHeader4.default,
      {
        sx: {
          display: "flex",
          flexWrap: "wrap",
          ".MuiCardHeader-action": {
            margin: 0,
            alignSelf: "center"
          }
        },
        title: title ?? /* @__PURE__ */ import_react14.default.createElement(
          import_Typography11.default,
          {
            variant: "h5",
            className: import_ui_types4.RefinePageHeaderClassNames.Title
          },
          translate(
            `${identifier}.titles.show`,
            `Show ${getUserFriendlyName(
              ((_c = resource == null ? void 0 : resource.meta) == null ? void 0 : _c.label) ?? identifier,
              "singular"
            )}`
          )
        ),
        avatar: typeof goBackFromProps !== "undefined" ? goBackFromProps : /* @__PURE__ */ import_react14.default.createElement(
          import_IconButton5.default,
          {
            onClick: action !== "list" && typeof action !== "undefined" ? back : void 0
          },
          /* @__PURE__ */ import_react14.default.createElement(import_ArrowBack3.default, null)
        ),
        action: /* @__PURE__ */ import_react14.default.createElement(import_Box10.default, { display: "flex", gap: "16px", ...headerButtonProps ?? {} }, headerButtons ? typeof headerButtons === "function" ? headerButtons({
          defaultButtons: defaultHeaderButtons,
          deleteButtonProps,
          editButtonProps,
          listButtonProps,
          refreshButtonProps
        }) : headerButtons : defaultHeaderButtons),
        ...headerProps ?? {}
      }
    ),
    /* @__PURE__ */ import_react14.default.createElement(import_CardContent8.default, { ...contentProps ?? {} }, children),
    /* @__PURE__ */ import_react14.default.createElement(import_CardActions3.default, { sx: { padding: "16px" }, ...footerButtonProps ?? {} }, footerButtons ? typeof footerButtons === "function" ? footerButtons({ defaultButtons: null }) : footerButtons : null)
  );
}, "Show");

// src/components/buttons/create/index.tsx
var import_react15 = __toESM(require("react"));
var import_core15 = require("@refinedev/core");
var import_ui_types5 = require("@refinedev/ui-types");
var import_Button6 = __toESM(require("@mui/material/Button"));
var import_AddBoxOutlined = __toESM(require("@mui/icons-material/AddBoxOutlined"));
var CreateButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...props
}) => {
  const { to, label, title, disabled, hidden, LinkComponent } = (0, import_core15.useCreateButton)(
    {
      resource: resourceNameFromProps,
      meta,
      accessControl
    }
  );
  const isDisabled = disabled || props.disabled;
  const isHidden = hidden || props.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = props;
  return /* @__PURE__ */ import_react15.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react15.default.createElement(
      import_Button6.default,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ import_react15.default.createElement(import_AddBoxOutlined.default, { ...svgIconProps }),
        title,
        variant: "contained",
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types5.RefineButtonClassNames.CreateButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react15.default.createElement(import_AddBoxOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "CreateButton");

// src/components/buttons/edit/index.tsx
var import_react16 = __toESM(require("react"));
var import_core16 = require("@refinedev/core");
var import_ui_types6 = require("@refinedev/ui-types");
var import_Button7 = __toESM(require("@mui/material/Button"));
var import_EditOutlined = __toESM(require("@mui/icons-material/EditOutlined"));
var EditButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core16.useEditButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    accessControl,
    meta
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react16.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react16.default.createElement(
      import_Button7.default,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ import_react16.default.createElement(import_EditOutlined.default, { sx: { selfAlign: "center" }, ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types6.RefineButtonClassNames.EditButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react16.default.createElement(import_EditOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "EditButton");

// src/components/buttons/delete/index.tsx
var import_react17 = __toESM(require("react"));
var import_core17 = require("@refinedev/core");
var import_ui_types7 = require("@refinedev/ui-types");
var import_Button8 = __toESM(require("@mui/material/Button"));
var import_Dialog = __toESM(require("@mui/material/Dialog"));
var import_DialogActions = __toESM(require("@mui/material/DialogActions"));
var import_DialogTitle = __toESM(require("@mui/material/DialogTitle"));
var import_LoadingButton = __toESM(require("@mui/lab/LoadingButton"));
var import_DeleteOutline = __toESM(require("@mui/icons-material/DeleteOutline"));
var DeleteButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  onSuccess,
  mutationMode,
  children,
  successNotification,
  errorNotification,
  hideText = false,
  accessControl,
  meta,
  dataProviderName,
  confirmTitle,
  confirmOkText,
  confirmCancelText,
  svgIconProps,
  invalidates,
  ...rest
}) => {
  const {
    onConfirm,
    title,
    label,
    hidden,
    disabled,
    loading,
    confirmTitle: defaultConfirmTitle,
    confirmOkLabel,
    cancelLabel
  } = (0, import_core17.useDeleteButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    dataProviderName,
    mutationMode,
    accessControl,
    invalidates,
    onSuccess,
    meta,
    successNotification,
    errorNotification
  });
  const [open, setOpen] = import_react17.default.useState(false);
  const { sx, ...restProps } = rest;
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  return /* @__PURE__ */ import_react17.default.createElement("div", null, /* @__PURE__ */ import_react17.default.createElement(
    import_LoadingButton.default,
    {
      color: "error",
      onClick: () => setOpen(true),
      disabled: isDisabled,
      loading,
      startIcon: !hideText && /* @__PURE__ */ import_react17.default.createElement(import_DeleteOutline.default, { ...svgIconProps }),
      title,
      sx: { minWidth: 0, ...sx },
      loadingPosition: hideText ? "center" : "start",
      
      className: import_ui_types7.RefineButtonClassNames.DeleteButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react17.default.createElement(import_DeleteOutline.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  ), /* @__PURE__ */ import_react17.default.createElement(
    import_Dialog.default,
    {
      open,
      onClose: () => setOpen(false),
      "aria-labelledby": "alert-dialog-title",
      "aria-describedby": "alert-dialog-description"
    },
    /* @__PURE__ */ import_react17.default.createElement(import_DialogTitle.default, { id: "alert-dialog-title" }, confirmTitle ?? defaultConfirmTitle),
    /* @__PURE__ */ import_react17.default.createElement(import_DialogActions.default, { sx: { justifyContent: "center" } }, /* @__PURE__ */ import_react17.default.createElement(import_Button8.default, { onClick: () => setOpen(false) }, confirmCancelText ?? cancelLabel), /* @__PURE__ */ import_react17.default.createElement(
      import_Button8.default,
      {
        color: "error",
        onClick: () => {
          onConfirm();
          setOpen(false);
        },
        autoFocus: true
      },
      confirmOkText ?? confirmOkLabel
    ))
  ));
}, "DeleteButton");

// src/components/buttons/refresh/index.tsx
var import_react18 = __toESM(require("react"));
var import_core18 = require("@refinedev/core");
var import_ui_types8 = require("@refinedev/ui-types");
var import_LoadingButton2 = __toESM(require("@mui/lab/LoadingButton"));
var import_RefreshOutlined = __toESM(require("@mui/icons-material/RefreshOutlined"));
var RefreshButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  dataProviderName,
  svgIconProps,
  children,
  onClick,
  ...rest
}) => {
  const {
    onClick: onRefresh,
    loading,
    label
  } = (0, import_core18.useRefreshButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    dataProviderName
  });
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react18.default.createElement(
    import_LoadingButton2.default,
    {
      startIcon: !hideText && /* @__PURE__ */ import_react18.default.createElement(import_RefreshOutlined.default, { ...svgIconProps }),
      loading,
      loadingPosition: hideText ? "center" : "start",
      onClick: onClick ? onClick : onRefresh,
      sx: { minWidth: 0, ...sx },
      
      className: import_ui_types8.RefineButtonClassNames.RefreshButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react18.default.createElement(import_RefreshOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "RefreshButton");

// src/components/buttons/show/index.tsx
var import_react19 = __toESM(require("react"));
var import_core19 = require("@refinedev/core");
var import_ui_types9 = require("@refinedev/ui-types");
var import_Button9 = __toESM(require("@mui/material/Button"));
var import_VisibilityOutlined = __toESM(require("@mui/icons-material/VisibilityOutlined"));
var ShowButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core19.useShowButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    accessControl,
    meta
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react19.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react19.default.createElement(
      import_Button9.default,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ import_react19.default.createElement(import_VisibilityOutlined.default, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types9.RefineButtonClassNames.ShowButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react19.default.createElement(import_VisibilityOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "ShowButton");

// src/components/buttons/list/index.tsx
var import_react20 = __toESM(require("react"));
var import_core20 = require("@refinedev/core");
var import_ui_types10 = require("@refinedev/ui-types");
var import_Button10 = __toESM(require("@mui/material/Button"));
var import_ListOutlined2 = __toESM(require("@mui/icons-material/ListOutlined"));
var ListButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core20.useListButton)({
    resource: resourceNameFromProps,
    meta,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react20.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react20.default.createElement(
      import_Button10.default,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ import_react20.default.createElement(import_ListOutlined2.default, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types10.RefineButtonClassNames.ListButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react20.default.createElement(import_ListOutlined2.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "ListButton");

// src/components/buttons/export/index.tsx
var import_react21 = __toESM(require("react"));
var import_core21 = require("@refinedev/core");
var import_ui_types11 = require("@refinedev/ui-types");
var import_LoadingButton3 = __toESM(require("@mui/lab/LoadingButton"));
var import_ImportExportOutlined = __toESM(require("@mui/icons-material/ImportExportOutlined"));
var ExportButton = /* @__PURE__ */ __name(({
  hideText = false,
  children,
  loading = false,
  svgIconProps,
  ...rest
}) => {
  const { label } = (0, import_core21.useExportButton)();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react21.default.createElement(
    import_LoadingButton3.default,
    {
      ...rest,
      loading,
      startIcon: !hideText && /* @__PURE__ */ import_react21.default.createElement(import_ImportExportOutlined.default, { ...svgIconProps }),
      loadingPosition: hideText ? "center" : "start",
      sx: { minWidth: 0, ...sx },
      
      className: import_ui_types11.RefineButtonClassNames.ExportButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react21.default.createElement(import_ImportExportOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "ExportButton");

// src/components/buttons/save/index.tsx
var import_react22 = __toESM(require("react"));
var import_core22 = require("@refinedev/core");
var import_ui_types12 = require("@refinedev/ui-types");
var import_LoadingButton4 = __toESM(require("@mui/lab/LoadingButton"));
var import_SaveOutlined = __toESM(require("@mui/icons-material/SaveOutlined"));
var SaveButton = /* @__PURE__ */ __name(({
  hideText = false,
  svgIconProps,
  children,
  ...rest
}) => {
  const { label } = (0, import_core22.useSaveButton)();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react22.default.createElement(
    import_LoadingButton4.default,
    {
      startIcon: !hideText && /* @__PURE__ */ import_react22.default.createElement(import_SaveOutlined.default, { ...svgIconProps }),
      sx: { minWidth: 0, ...sx },
      variant: "contained",
      
      className: import_ui_types12.RefineButtonClassNames.SaveButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react22.default.createElement(import_SaveOutlined.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  );
}, "SaveButton");

// src/components/buttons/clone/index.tsx
var import_react23 = __toESM(require("react"));
var import_core23 = require("@refinedev/core");
var import_ui_types13 = require("@refinedev/ui-types");
var import_Button11 = __toESM(require("@mui/material/Button"));
var import_AddBoxOutlined2 = __toESM(require("@mui/icons-material/AddBoxOutlined"));
var CloneButton = /* @__PURE__ */ __name(({
  resource: resourceNameFromProps,
  recordItemId,
  hideText = false,
  accessControl,
  svgIconProps,
  meta,
  children,
  onClick,
  ...rest
}) => {
  const { to, label, title, hidden, disabled, LinkComponent } = (0, import_core23.useCloneButton)({
    resource: resourceNameFromProps,
    id: recordItemId,
    meta,
    accessControl
  });
  const isDisabled = disabled || rest.disabled;
  const isHidden = hidden || rest.hidden;
  if (isHidden)
    return null;
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react23.default.createElement(
    LinkComponent,
    {
      to,
      replace: false,
      onClick: (e) => {
        if (isDisabled) {
          e.preventDefault();
          return;
        }
        if (onClick) {
          e.preventDefault();
          onClick(e);
        }
      },
      style: { textDecoration: "none" }
    },
    /* @__PURE__ */ import_react23.default.createElement(
      import_Button11.default,
      {
        disabled: isDisabled,
        startIcon: !hideText && /* @__PURE__ */ import_react23.default.createElement(import_AddBoxOutlined2.default, { ...svgIconProps }),
        title,
        sx: { minWidth: 0, ...sx },
        
        className: import_ui_types13.RefineButtonClassNames.CloneButton,
        ...restProps
      },
      hideText ? /* @__PURE__ */ import_react23.default.createElement(import_AddBoxOutlined2.default, { fontSize: "small", ...svgIconProps }) : children ?? label
    )
  );
}, "CloneButton");

// src/components/buttons/import/index.tsx
var import_react24 = __toESM(require("react"));
var import_core24 = require("@refinedev/core");
var import_ui_types14 = require("@refinedev/ui-types");
var import_LoadingButton5 = __toESM(require("@mui/lab/LoadingButton"));
var import_ImportExportOutlined2 = __toESM(require("@mui/icons-material/ImportExportOutlined"));
var ImportButton = /* @__PURE__ */ __name(({
  inputProps,
  hideText = false,
  loading = false,
  svgIconProps,
  children,
  ...rest
}) => {
  const { label } = (0, import_core24.useImportButton)();
  const { sx, ...restProps } = rest;
  return /* @__PURE__ */ import_react24.default.createElement("label", { htmlFor: "contained-button-file" }, /* @__PURE__ */ import_react24.default.createElement("input", { ...inputProps, id: "contained-button-file", multiple: true, hidden: true }), /* @__PURE__ */ import_react24.default.createElement(
    import_LoadingButton5.default,
    {
      component: "span",
      startIcon: !hideText && /* @__PURE__ */ import_react24.default.createElement(import_ImportExportOutlined2.default, { ...svgIconProps }),
      loadingPosition: hideText ? "center" : "start",
      loading,
      sx: { minWidth: 0, ...sx },
      
      className: import_ui_types14.RefineButtonClassNames.ImportButton,
      ...restProps
    },
    hideText ? /* @__PURE__ */ import_react24.default.createElement(import_ImportExportOutlined2.default, { fontSize: "small", ...svgIconProps }) : children ?? label
  ));
}, "ImportButton");

// src/components/circularDeterminate/index.tsx
var import_react25 = __toESM(require("react"));
var import_Box11 = __toESM(require("@mui/material/Box"));
var import_CircularProgress4 = __toESM(require("@mui/material/CircularProgress"));
var import_Typography12 = __toESM(require("@mui/material/Typography"));
var CircularDeterminate = /* @__PURE__ */ __name(({
  undoableTimeout,
  message
}) => {
  const [progress, setProgress] = (0, import_react25.useState)(100);
  const [timeCount, setTimeCount] = (0, import_react25.useState)(undoableTimeout);
  (0, import_react25.useEffect)(() => {
    const increaseProgress = 100 / undoableTimeout;
    const timer = setInterval(() => {
      setTimeCount((prevProgress) => prevProgress - 1);
      setProgress((prevProgress) => prevProgress - increaseProgress);
    }, 1e3);
    if (timeCount === 0) {
      clearInterval(timer);
    }
    return () => {
      clearInterval(timer);
    };
  }, [timeCount]);
  return /* @__PURE__ */ import_react25.default.createElement(import_react25.default.Fragment, null, /* @__PURE__ */ import_react25.default.createElement(import_Box11.default, { sx: { position: "relative", display: "inline-flex" } }, /* @__PURE__ */ import_react25.default.createElement(
    import_CircularProgress4.default,
    {
      color: "inherit",
      variant: "determinate",
      value: progress
    }
  ), /* @__PURE__ */ import_react25.default.createElement(
    import_Box11.default,
    {
      sx: {
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
        position: "absolute",
        display: "flex",
        alignItems: "center",
        justifyContent: "center"
      }
    },
    /* @__PURE__ */ import_react25.default.createElement(import_Typography12.default, { component: "div" }, timeCount)
  )), /* @__PURE__ */ import_react25.default.createElement(
    import_Box11.default,
    {
      sx: {
        marginLeft: "10px",
        maxWidth: { xs: "150px", md: "100%" }
      }
    },
    /* @__PURE__ */ import_react25.default.createElement(import_Typography12.default, { variant: "subtitle2" }, message)
  ));
}, "CircularDeterminate");

// src/components/fields/text/index.tsx
var import_react26 = __toESM(require("react"));
var import_Typography13 = __toESM(require("@mui/material/Typography"));
var TextField5 = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react26.default.createElement(import_Typography13.default, { variant: "body2", ...rest }, value);
}, "TextField");

// src/components/fields/tag/index.tsx
var import_react27 = __toESM(require("react"));
var import_Chip = __toESM(require("@mui/material/Chip"));
var TagField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react27.default.createElement(import_Chip.default, { label: value == null ? void 0 : value.toString(), ...rest });
}, "TagField");

// src/components/fields/email/index.tsx
var import_react28 = __toESM(require("react"));
var import_Typography14 = __toESM(require("@mui/material/Typography"));
var import_Link5 = __toESM(require("@mui/material/Link"));
var EmailField = /* @__PURE__ */ __name(({ value, ...rest }) => {
  return /* @__PURE__ */ import_react28.default.createElement(import_Typography14.default, { variant: "body2" }, /* @__PURE__ */ import_react28.default.createElement(import_Link5.default, { href: `mailto:${value}`, ...rest }, value));
}, "EmailField");

// src/components/fields/boolean/index.tsx
var import_react29 = __toESM(require("react"));
var import_Tooltip3 = __toESM(require("@mui/material/Tooltip"));
var import_CheckOutlined = __toESM(require("@mui/icons-material/CheckOutlined"));
var import_CloseOutlined = __toESM(require("@mui/icons-material/CloseOutlined"));
var BooleanField = /* @__PURE__ */ __name(({
  value,
  valueLabelTrue = "true",
  valueLabelFalse = "false",
  trueIcon,
  falseIcon,
  svgIconProps,
  ...rest
}) => {
  return /* @__PURE__ */ import_react29.default.createElement(import_Tooltip3.default, { title: value ? valueLabelTrue : valueLabelFalse, ...rest }, value ? /* @__PURE__ */ import_react29.default.createElement("span", null, trueIcon ?? /* @__PURE__ */ import_react29.default.createElement(import_CheckOutlined.default, { ...svgIconProps })) : /* @__PURE__ */ import_react29.default.createElement("span", null, falseIcon ?? /* @__PURE__ */ import_react29.default.createElement(import_CloseOutlined.default, { ...svgIconProps })));
}, "BooleanField");

// src/components/fields/date/index.tsx
var import_react30 = __toESM(require("react"));
var import_dayjs = __toESM(require("dayjs"));
var import_localizedFormat = __toESM(require("dayjs/plugin/localizedFormat"));
var import_Typography15 = __toESM(require("@mui/material/Typography"));
import_dayjs.default.extend(import_localizedFormat.default);
var defaultLocale = import_dayjs.default.locale();
var DateField = /* @__PURE__ */ __name(({
  value,
  locales,
  format: dateFormat = "L",
  ...rest
}) => {
  return /* @__PURE__ */ import_react30.default.createElement(import_Typography15.default, { variant: "body2", ...rest }, value ? (0, import_dayjs.default)(value).locale(locales || defaultLocale).format(dateFormat) : "");
}, "DateField");

// src/components/fields/file/index.tsx
var import_react31 = __toESM(require("react"));
var FileField = /* @__PURE__ */ __name(({
  title,
  src,
  ...rest
}) => {
  return /* @__PURE__ */ import_react31.default.createElement(UrlField, { value: src, title, ...rest }, title ?? src);
}, "FileField");

// src/components/fields/url/index.tsx
var import_react32 = __toESM(require("react"));
var import_Link6 = __toESM(require("@mui/material/Link"));
var import_Typography16 = __toESM(require("@mui/material/Typography"));
var UrlField = /* @__PURE__ */ __name(({
  children,
  value,
  ...rest
}) => {
  return /* @__PURE__ */ import_react32.default.createElement(import_Typography16.default, { variant: "body2" }, /* @__PURE__ */ import_react32.default.createElement(import_Link6.default, { href: value, ...rest }, children ?? value));
}, "UrlField");

// src/components/fields/number/index.tsx
var import_react33 = __toESM(require("react"));
var import_Typography17 = __toESM(require("@mui/material/Typography"));
function toLocaleStringSupportsOptions() {
  return !!(typeof Intl === "object" && Intl && typeof Intl.NumberFormat === "function");
}
__name(toLocaleStringSupportsOptions, "toLocaleStringSupportsOptions");
var NumberField = /* @__PURE__ */ __name(({
  value,
  locale,
  options,
  ...rest
}) => {
  const number = Number(value);
  return /* @__PURE__ */ import_react33.default.createElement(import_Typography17.default, { variant: "body2", ...rest }, toLocaleStringSupportsOptions() ? number.toLocaleString(locale, options) : number);
}, "NumberField");

// src/components/fields/markdown/index.tsx
var import_react34 = __toESM(require("react"));
var import_react_markdown = __toESM(require("react-markdown"));
var import_remark_gfm = __toESM(require("remark-gfm"));
var MarkdownField = /* @__PURE__ */ __name(({ value = "" }) => {
  return /* @__PURE__ */ import_react34.default.createElement(
    import_react_markdown.default,
    {
      remarkPlugins: [import_remark_gfm.default]
    },
    value
  );
}, "MarkdownField");

// src/components/breadcrumb/index.tsx
var import_react35 = __toESM(require("react"));
var import_core25 = require("@refinedev/core");
var import_Breadcrumbs = __toESM(require("@mui/material/Breadcrumbs"));
var import_Typography18 = __toESM(require("@mui/material/Typography"));
var import_Grid22 = __toESM(require("@mui/material/Grid2"));
var import_HomeOutlined = __toESM(require("@mui/icons-material/HomeOutlined"));
var Breadcrumb = /* @__PURE__ */ __name(({
  breadcrumbProps,
  showHome = true,
  hideIcons = false,
  meta,
  minItems = 2
}) => {
  var _a, _b;
  const { breadcrumbs } = (0, import_core25.useBreadcrumb)({ meta });
  const Link3 = (0, import_core25.useLink)();
  const { resources } = (0, import_core25.useResourceParams)();
  const rootRouteResource = (0, import_core25.matchResourceFromRoute)("/", resources);
  if (breadcrumbs.length < minItems)
    return null;
  const LinkRouter = /* @__PURE__ */ __name((props) => {
    const { to, children, ...restProps } = props;
    return /* @__PURE__ */ import_react35.default.createElement(Link3, { to: to || "" }, /* @__PURE__ */ import_react35.default.createElement("span", { ...restProps }, children));
  }, "LinkRouter");
  return /* @__PURE__ */ import_react35.default.createElement(
    import_Breadcrumbs.default,
    {
      "aria-label": "breadcrumb",
      sx: {
        padding: 2,
        ...(breadcrumbProps == null ? void 0 : breadcrumbProps.sx) ?? {}
      },
      ...breadcrumbProps
    },
    showHome && rootRouteResource.found && /* @__PURE__ */ import_react35.default.createElement(
      LinkRouter,
      {
        underline: "hover",
        sx: {
          display: "flex",
          alignItems: "center"
        },
        color: "inherit",
        to: "/"
      },
      ((_b = (_a = rootRouteResource == null ? void 0 : rootRouteResource.resource) == null ? void 0 : _a.meta) == null ? void 0 : _b.icon) ?? /* @__PURE__ */ import_react35.default.createElement(
        import_HomeOutlined.default,
        {
          sx: {
            fontSize: "18px"
          }
        }
      )
    ),
    breadcrumbs.map(({ label, icon, href }) => {
      return /* @__PURE__ */ import_react35.default.createElement(
        import_Grid22.default,
        {
          key: label,
          sx: {
            display: "flex",
            alignItems: "center",
            "& .MuiSvgIcon-root": {
              fontSize: "16px"
            }
          }
        },
        !hideIcons && icon,
        href ? /* @__PURE__ */ import_react35.default.createElement(
          LinkRouter,
          {
            underline: "hover",
            sx: {
              display: "flex",
              alignItems: "center",
              fontSize: "14px",
              marginLeft: 0.5
            },
            color: "inherit",
            to: href,
            variant: "subtitle1"
          },
          label
        ) : /* @__PURE__ */ import_react35.default.createElement(import_Typography18.default, { fontSize: "14px" }, label)
      );
    })
  );
}, "Breadcrumb");

// src/components/autoSaveIndicator/index.tsx
var import_react36 = __toESM(require("react"));
var import_core26 = require("@refinedev/core");
var import_Typography19 = __toESM(require("@mui/material/Typography"));
var import_MoreHorizOutlined = __toESM(require("@mui/icons-material/MoreHorizOutlined"));
var import_SyncOutlined = __toESM(require("@mui/icons-material/SyncOutlined"));
var import_TaskAltOutlined = __toESM(require("@mui/icons-material/TaskAltOutlined"));
var import_ErrorOutlineOutlined = __toESM(require("@mui/icons-material/ErrorOutlineOutlined"));
var AutoSaveIndicator = /* @__PURE__ */ __name(({
  status,
  elements: {
    success = /* @__PURE__ */ import_react36.default.createElement(
      Message,
      {
        translationKey: "autoSave.success",
        defaultMessage: "saved",
        icon: /* @__PURE__ */ import_react36.default.createElement(import_TaskAltOutlined.default, { fontSize: "small" })
      }
    ),
    error = /* @__PURE__ */ import_react36.default.createElement(
      Message,
      {
        translationKey: "autoSave.error",
        defaultMessage: "auto save failure",
        icon: /* @__PURE__ */ import_react36.default.createElement(import_ErrorOutlineOutlined.default, { fontSize: "small" })
      }
    ),
    loading = /* @__PURE__ */ import_react36.default.createElement(
      Message,
      {
        translationKey: "autoSave.loading",
        defaultMessage: "saving...",
        icon: /* @__PURE__ */ import_react36.default.createElement(import_SyncOutlined.default, { fontSize: "small" })
      }
    ),
    idle = /* @__PURE__ */ import_react36.default.createElement(
      Message,
      {
        translationKey: "autoSave.idle",
        defaultMessage: "waiting for changes",
        icon: /* @__PURE__ */ import_react36.default.createElement(import_MoreHorizOutlined.default, { fontSize: "small" })
      }
    )
  } = {}
}) => {
  return /* @__PURE__ */ import_react36.default.createElement(
    import_core26.AutoSaveIndicator,
    {
      status,
      elements: {
        success,
        error,
        loading,
        idle
      }
    }
  );
}, "AutoSaveIndicator");
var Message = /* @__PURE__ */ __name(({
  translationKey,
  defaultMessage,
  icon
}) => {
  const translate = (0, import_core26.useTranslate)();
  return /* @__PURE__ */ import_react36.default.createElement(
    import_Typography19.default,
    {
      color: "gray",
      fontSize: "0.8rem",
      position: "relative",
      display: "flex",
      alignItems: "center",
      flexWrap: "wrap",
      marginRight: ".3rem"
    },
    translate(translationKey, defaultMessage),
    /* @__PURE__ */ import_react36.default.createElement("span", { style: { position: "relative", top: "3px", marginLeft: "3px" } }, icon)
  );
}, "Message");

// src/theme/index.ts
var import_styles5 = require("@mui/material/styles");

// src/theme/palette/lightPalette.ts
var lightPalette = {
  mode: "light",
  primary: {
    main: "#67be23",
    contrastText: "#fff"
  },
  secondary: {
    main: "#2A132E",
    contrastText: "#fff"
  },
  background: {
    default: "#f0f0f0",
    paper: "#ffffff"
  },
  success: {
    main: "#67be23",
    contrastText: "#fff"
  },
  error: {
    main: "#fa541c",
    contrastText: "#fff"
  },
  warning: {
    main: "#fa8c16",
    contrastText: "#fff"
  },
  info: {
    main: "#0b82f0",
    contrastText: "#fff"
  },
  divider: "rgba(0,0,0,0)",
  text: {
    primary: "#626262",
    secondary: "#9f9f9f",
    disabled: "#c1c1c1"
  }
};

// src/theme/palette/darkPalette.ts
var darkPalette = {
  mode: "dark",
  primary: {
    main: "#67be23",
    contrastText: "#fff"
  },
  secondary: {
    main: "#2A132E",
    contrastText: "#fff"
  },
  background: {
    default: "#212121",
    paper: "#242424"
  },
  success: {
    main: "#67be23",
    contrastText: "#fff"
  },
  error: {
    main: "#ee2a1e",
    contrastText: "#fff"
  },
  warning: {
    main: "#fa8c16",
    contrastText: "#fff"
  },
  info: {
    main: "#1890ff",
    contrastText: "#fff"
  },
  divider: "rgba(0,0,0,0)",
  text: {
    primary: "#fff",
    secondary: "rgba(255,255,255,0.7)",
    disabled: "#d1d1d1"
  }
};

// src/theme/typography.ts
var typography = {
  fontFamily: [
    "Montserrat",
    "-apple-system",
    "BlinkMacSystemFont",
    '"Segoe UI"',
    "Roboto",
    '"Helvetica Neue"',
    "Arial",
    "sans-serif",
    '"Apple Color Emoji"',
    '"Segoe UI Emoji"',
    '"Segoe UI Symbol"'
  ].join(",")
};

// src/theme/palette/refinePalette.ts
var RefinePalettes = {
  Blue: {
    mode: "light",
    primary: {
      main: "#1976D2",
      light: "#4791db",
      dark: "#115293"
    }
  },
  BlueDark: {
    mode: "dark",
    primary: {
      main: "#67b7f7",
      light: "#85c5f8",
      dark: "#4880ac"
    }
  },
  Purple: {
    mode: "light",
    primary: {
      main: "#7B1FA2",
      light: "#954bb4",
      dark: "#561571"
    }
  },
  PurpleDark: {
    mode: "dark",
    primary: {
      main: "#AB47BC",
      light: "#bb6bc9",
      dark: "#773183"
    }
  },
  Magenta: {
    mode: "light",
    primary: {
      main: "#C2185B",
      light: "#ce467b",
      dark: "#87103f"
    }
  },
  MagentaDark: {
    mode: "dark",
    primary: {
      main: "#EC407A",
      light: "#ef6694",
      dark: "#a52c55"
    }
  },
  Red: {
    mode: "light",
    primary: {
      main: "#D32F2F",
      light: "#db5858",
      dark: "#932020"
    }
  },
  RedDark: {
    mode: "dark",
    primary: {
      main: "#EF5350",
      light: "#f27573",
      dark: "#a73a38"
    }
  },
  Orange: {
    mode: "light",
    primary: {
      main: "#F57C00",
      light: "#f79633",
      dark: "#ab5600"
    }
  },
  OrangeDark: {
    mode: "dark",
    primary: {
      main: "#FFA726",
      light: "#ffb851",
      dark: "#b2741a"
    }
  },
  Yellow: {
    mode: "light",
    primary: {
      main: "#FFA000",
      light: "#ffb333",
      dark: "#b27000"
    }
  },
  YellowDark: {
    mode: "dark",
    primary: {
      main: "#FFCA28",
      light: "#ffd453",
      dark: "#E87040"
    }
  },
  Green: {
    mode: "light",
    primary: {
      main: "#689F38",
      light: "#86b25f",
      dark: "#486f27"
    }
  },
  GreenDark: {
    mode: "dark",
    primary: {
      main: "#9CCC65",
      light: "#afd683",
      dark: "#6d8e46"
    }
  }
};

// src/theme/index.ts
var commonThemeProperties = {
  shape: {
    borderRadius: 6
  },
  typography: {
    ...typography
  }
};
var LightTheme = (0, import_styles5.createTheme)({
  ...commonThemeProperties,
  palette: lightPalette,
  components: {
    MuiAppBar: {
      styleOverrides: {
        colorDefault: {
          backgroundColor: "#fff"
        }
      }
    },
    MuiPaper: {
      styleOverrides: {
        root: {
          backgroundImage: "linear-gradient(rgba(255, 255, 255, 0.01), rgba(255, 255, 255, 0.01))"
        }
      }
    },
    MuiTypography: {
      styleOverrides: {
        h5: {
          fontWeight: 800,
          lineHeight: "2rem"
        }
      }
    }
  }
});
var DarkTheme = (0, import_styles5.createTheme)({
  ...commonThemeProperties,
  palette: darkPalette,
  components: {
    MuiPaper: {
      styleOverrides: {
        root: {
          backgroundImage: "linear-gradient(rgba(255, 255, 255, 0.025), rgba(255, 255, 255, 0.025))"
        }
      }
    },
    MuiAppBar: {
      defaultProps: {
        color: "transparent"
      }
    },
    MuiTypography: {
      styleOverrides: {
        h5: {
          fontWeight: 800,
          lineHeight: "2rem"
        }
      }
    }
  }
});
var RefineThemes = Object.keys(RefinePalettes).reduce((acc, key) => {
  const paletteName = key;
  return {
    ...acc,
    [key]: (0, import_styles5.createTheme)({
      palette: {
        ...RefinePalettes[paletteName]
      },
      components: {
        MuiButton: {
          styleOverrides: {
            root: ({ ownerState }) => ({
              ...ownerState.variant === "contained" && ownerState.color === "primary" && {
                color: "#fff"
              }
            })
          }
        }
      }
    })
  };
}, {});

// src/providers/notificationProvider/index.tsx
var import_react37 = __toESM(require("react"));
var import_notistack = require("notistack");
var import_Box12 = __toESM(require("@mui/material/Box"));
var import_IconButton6 = __toESM(require("@mui/material/IconButton"));
var import_Typography20 = __toESM(require("@mui/material/Typography"));
var import_UndoOutlined = __toESM(require("@mui/icons-material/UndoOutlined"));
var useNotificationProvider = /* @__PURE__ */ __name(() => {
  const { closeSnackbar, enqueueSnackbar } = (0, import_notistack.useSnackbar)();
  const notificationProvider = {
    open: ({
      message,
      type,
      undoableTimeout,
      key,
      cancelMutation,
      description
    }) => {
      if (type === "progress") {
        const action = /* @__PURE__ */ __name((key2) => /* @__PURE__ */ import_react37.default.createElement(
          import_IconButton6.default,
          {
            onClick: () => {
              cancelMutation == null ? void 0 : cancelMutation();
              closeSnackbar(key2);
            },
            color: "inherit"
          },
          /* @__PURE__ */ import_react37.default.createElement(import_UndoOutlined.default, null)
        ), "action");
        enqueueSnackbar(
          /* @__PURE__ */ import_react37.default.createElement(import_react37.default.Fragment, null, /* @__PURE__ */ import_react37.default.createElement(
            CircularDeterminate,
            {
              undoableTimeout: undoableTimeout ?? 0,
              message
            }
          )),
          {
            action,
            preventDuplicate: true,
            key,
            autoHideDuration: (undoableTimeout ?? 0) * 1e3
          }
        );
      } else {
        enqueueSnackbar(
          /* @__PURE__ */ import_react37.default.createElement(import_Box12.default, null, /* @__PURE__ */ import_react37.default.createElement(import_Typography20.default, { variant: "subtitle2", component: "h6" }, description), /* @__PURE__ */ import_react37.default.createElement(import_Typography20.default, { variant: "caption", component: "p" }, message)),
          {
            key,
            variant: type
          }
        );
      }
    },
    close: (key) => {
      closeSnackbar(key);
    }
  };
  return notificationProvider;
}, "useNotificationProvider");

// src/providers/refineSnackbarProvider/index.tsx
var import_react38 = __toESM(require("react"));
var import_styles6 = require("@mui/material/styles");
var import_notistack2 = require("notistack");
var SnackbarProviderWithDefaultValues = /* @__PURE__ */ __name(({
  anchorOrigin = {
    vertical: "top",
    horizontal: "right"
  },
  disableWindowBlurListener = true,
  ...rest
}) => {
  return /* @__PURE__ */ import_react38.default.createElement(
    import_notistack2.SnackbarProvider,
    {
      anchorOrigin,
      disableWindowBlurListener,
      ...rest
    }
  );
}, "SnackbarProviderWithDefaultValues");
var RefineSnackbarProvider = (0, import_styles6.styled)(SnackbarProviderWithDefaultValues)`
&.SnackbarItem-contentRoot {
    background-color: ${(props) => props.theme.palette.background.default};
    color: ${(props) => props.theme.palette.primary.main};
}
&.SnackbarItem-variantSuccess {
    background-color: ${(props) => props.theme.palette.success.main};
    color: ${(props) => props.theme.palette.success.contrastText};
}
&.SnackbarItem-variantError {
    background-color: ${(props) => props.theme.palette.error.main};
    color: ${(props) => props.theme.palette.error.contrastText};
}
&.SnackbarItem-variantInfo {
    background-color: ${(props) => props.theme.palette.info.main};
    color: ${(props) => props.theme.palette.info.contrastText};
}
&.SnackbarItem-variantWarning {
    background-color: ${(props) => props.theme.palette.warning.main};
    color: ${(props) => props.theme.palette.warning.contrastText};
}
`;
